<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Genetic Groups | Selecting SNPs for SALT array</title>
  <meta name="description" content="3 Genetic Groups | Selecting SNPs for SALT array" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Genetic Groups | Selecting SNPs for SALT array" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Genetic Groups | Selecting SNPs for SALT array" />
  
  
  

<meta name="author" content="Joseph Matt" />


<meta name="date" content="2023-12-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="input-data.html"/>
<link rel="next" href="select-candidate-snps.html"/>
<script src="book_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="book_files/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="book_files/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="book_files/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="book_files/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#objective-organize-individual-and-snp-data"><i class="fa fa-check"></i><b>2.1</b> Objective: Organize individual and SNP data</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#individual-sample-information"><i class="fa fa-check"></i><b>2.2</b> Individual Sample Information</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="input-data.html"><a href="input-data.html#basic-sample-information"><i class="fa fa-check"></i><b>2.2.1</b> Basic sample information</a></li>
<li class="chapter" data-level="2.2.2" data-path="input-data.html"><a href="input-data.html#quality-control-info-from-thermo-fisher"><i class="fa fa-check"></i><b>2.2.2</b> Quality control info from Thermo Fisher</a></li>
<li class="chapter" data-level="2.2.3" data-path="input-data.html"><a href="input-data.html#list-of-oysters-that-passed-the-genotyping-according-to-thermofisher"><i class="fa fa-check"></i><b>2.2.3</b> List of oysters that passed the genotyping according to ThermoFisher</a></li>
<li class="chapter" data-level="2.2.4" data-path="input-data.html"><a href="input-data.html#merge-files"><i class="fa fa-check"></i><b>2.2.4</b> Merge Files</a></li>
<li class="chapter" data-level="2.2.5" data-path="input-data.html"><a href="input-data.html#tidy-up-latlong-data-and-site-codes"><i class="fa fa-check"></i><b>2.2.5</b> Tidy Up Lat/Long Data and Site Codes</a></li>
<li class="chapter" data-level="2.2.6" data-path="input-data.html"><a href="input-data.html#organize-oysters-into-metagroup-categories"><i class="fa fa-check"></i><b>2.2.6</b> Organize oysters into “metagroup” categories:</a></li>
<li class="chapter" data-level="2.2.7" data-path="input-data.html"><a href="input-data.html#map-sites-where-samples-were-collected"><i class="fa fa-check"></i><b>2.2.7</b> Map sites where samples were collected</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#snp-information"><i class="fa fa-check"></i><b>2.3</b> SNP Information</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="input-data.html"><a href="input-data.html#general-information-on-snps"><i class="fa fa-check"></i><b>2.3.1</b> General Information on SNPs</a></li>
<li class="chapter" data-level="2.3.2" data-path="input-data.html"><a href="input-data.html#recommended-snps"><i class="fa fa-check"></i><b>2.3.2</b> Recommended SNPs</a></li>
<li class="chapter" data-level="2.3.3" data-path="input-data.html"><a href="input-data.html#quality-control-info"><i class="fa fa-check"></i><b>2.3.3</b> Quality control info</a></li>
<li class="chapter" data-level="2.3.4" data-path="input-data.html"><a href="input-data.html#merge-files-1"><i class="fa fa-check"></i><b>2.3.4</b> Merge Files</a></li>
<li class="chapter" data-level="2.3.5" data-path="input-data.html"><a href="input-data.html#formatting-for-hierfstat-and-adegenet"><i class="fa fa-check"></i><b>2.3.5</b> Formatting for hierfstat and adegenet</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#save-files"><i class="fa fa-check"></i><b>2.4</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genetic-groups.html"><a href="genetic-groups.html"><i class="fa fa-check"></i><b>3</b> Genetic Groups</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genetic-groups.html"><a href="genetic-groups.html#objective-assign-individuals-to-genetic-groups"><i class="fa fa-check"></i><b>3.1</b> Objective: Assign individuals to genetic groups</a></li>
<li class="chapter" data-level="3.2" data-path="genetic-groups.html"><a href="genetic-groups.html#data-subsetting"><i class="fa fa-check"></i><b>3.2</b> Data Subsetting</a></li>
<li class="chapter" data-level="3.3" data-path="genetic-groups.html"><a href="genetic-groups.html#kmeans---all-wild-individuals"><i class="fa fa-check"></i><b>3.3</b> Kmeans - All Wild Individuals</a></li>
<li class="chapter" data-level="3.4" data-path="genetic-groups.html"><a href="genetic-groups.html#pca---k-groups"><i class="fa fa-check"></i><b>3.4</b> PCA - K groups</a></li>
<li class="chapter" data-level="3.5" data-path="genetic-groups.html"><a href="genetic-groups.html#kmeans-and-pca-with-four-groups"><i class="fa fa-check"></i><b>3.5</b> Kmeans and PCA with four groups</a></li>
<li class="chapter" data-level="3.6" data-path="genetic-groups.html"><a href="genetic-groups.html#pca---four-k-groups"><i class="fa fa-check"></i><b>3.6</b> PCA - Four K groups</a></li>
<li class="chapter" data-level="3.7" data-path="genetic-groups.html"><a href="genetic-groups.html#assign-genetic-groups"><i class="fa fa-check"></i><b>3.7</b> Assign Genetic Groups</a></li>
<li class="chapter" data-level="3.8" data-path="genetic-groups.html"><a href="genetic-groups.html#summary"><i class="fa fa-check"></i><b>3.8</b> Summary</a></li>
<li class="chapter" data-level="3.9" data-path="genetic-groups.html"><a href="genetic-groups.html#save-files-1"><i class="fa fa-check"></i><b>3.9</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html"><i class="fa fa-check"></i><b>4</b> Select Candidate SNPs</a>
<ul>
<li class="chapter" data-level="4.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#objective-identify-candidate-snps-by"><i class="fa fa-check"></i><b>4.1</b> Objective: Identify candidate SNPs by</a></li>
<li class="chapter" data-level="4.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#read-in-data"><i class="fa fa-check"></i><b>4.2</b> Read in Data</a></li>
<li class="chapter" data-level="4.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#format-data"><i class="fa fa-check"></i><b>4.3</b> Format Data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genind-object-from-105k-snps"><i class="fa fa-check"></i><b>4.3.1</b> Make genind object from 105K SNPs</a></li>
<li class="chapter" data-level="4.3.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-a-dataframe-specifying-genetic-group-and-metapop."><i class="fa fa-check"></i><b>4.3.2</b> Make a dataframe specifying genetic group and “metapop.”</a></li>
<li class="chapter" data-level="4.3.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#assign-populations-to-genind-object"><i class="fa fa-check"></i><b>4.3.3</b> Assign populations to genind object</a></li>
<li class="chapter" data-level="4.3.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#format-data-into-the-hierfstat-format-for-analysis-of-heterozygosity-and-maf"><i class="fa fa-check"></i><b>4.3.4</b> Format data into the hierfstat format for analysis of heterozygosity and MAF</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#hwe-filtering"><i class="fa fa-check"></i><b>4.4</b> HWE filtering</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#filtering-out-the-loci-with-ho-0.5"><i class="fa fa-check"></i><b>4.4.1</b> Filtering out the loci with Ho &gt; 0.5</a></li>
<li class="chapter" data-level="4.4.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#re-analyze-the-data-for-heterozygosities"><i class="fa fa-check"></i><b>4.4.2</b> Re-analyze the data for heterozygosities</a></li>
<li class="chapter" data-level="4.4.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genind-object-without-the-loci-that-have-heterozygosities-over-0.5"><i class="fa fa-check"></i><b>4.4.3</b> Make genind object without the loci that have heterozygosities over 0.5</a></li>
<li class="chapter" data-level="4.4.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genotypes-data-frame-without-the-loci-that-have-heterozygosities-over-0.5"><i class="fa fa-check"></i><b>4.4.4</b> Make genotypes data frame without the loci that have heterozygosities over 0.5</a></li>
<li class="chapter" data-level="4.4.5" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#hw-test"><i class="fa fa-check"></i><b>4.4.5</b> HW test</a></li>
<li class="chapter" data-level="4.4.6" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-loci-that-fail-hwe-test-p-0.05"><i class="fa fa-check"></i><b>4.4.6</b> Select loci that fail HWE test (p &lt; 0.05)</a></li>
<li class="chapter" data-level="4.4.7" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#remove-loci-out-of-hwe-from-the-genind-object-and-genotypes-data-frame"><i class="fa fa-check"></i><b>4.4.7</b> Remove loci out of HWE from the genind object and genotypes data frame</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-population-diagnostic-loci"><i class="fa fa-check"></i><b>4.5</b> Select population diagnostic loci</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#south-texas-vs.-gulf"><i class="fa fa-check"></i><b>4.5.1</b> South Texas vs. Gulf</a></li>
<li class="chapter" data-level="4.5.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-25-snps-that-distinguish-stx-from-gulf"><i class="fa fa-check"></i><b>4.5.2</b> Select 25 SNPs that distinguish stx from gulf</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#ngom-vs.-fl"><i class="fa fa-check"></i><b>4.6</b> nGOM vs. FL</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-25-snps-with-highest-fst-for-differentiating-nfl-vs.-gfl-vs.-ngom"><i class="fa fa-check"></i><b>4.6.1</b> Select 25 SNPs with highest Fst for differentiating nfl vs. gfl vs. ngom</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#maf"><i class="fa fa-check"></i><b>4.7</b> MAF</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#calculate-minor-allele-frequency-for-each-locus-in-each-genetic-group"><i class="fa fa-check"></i><b>4.7.1</b> Calculate Minor Allele Frequency for each locus in each genetic group</a></li>
<li class="chapter" data-level="4.7.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#re-calculate-after-removing-loci-with-maf-of-1"><i class="fa fa-check"></i><b>4.7.2</b> Re-calculate after removing loci with MAF of 1</a></li>
<li class="chapter" data-level="4.7.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#identify-loci-with-high-maf-in-all-groups"><i class="fa fa-check"></i><b>4.7.3</b> Identify loci with high MAF in all groups</a></li>
<li class="chapter" data-level="4.7.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-maf-loci-and-exmaine-histograms-of-loci"><i class="fa fa-check"></i><b>4.7.4</b> Select MAF loci and exmaine histograms of loci</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#save-files-2"><i class="fa fa-check"></i><b>4.8</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html"><i class="fa fa-check"></i><b>5</b> Select SNPs for Panel</a>
<ul>
<li class="chapter" data-level="5.1" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#read-in-all-data-on-snps-and-relevant-data-from-section-3"><i class="fa fa-check"></i><b>5.1</b> Read in all data on SNPs and relevant data from Section 3</a></li>
<li class="chapter" data-level="5.2" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#filter-the-snp-data-file-to-only-include-maf-snps-selected-in-section3"><i class="fa fa-check"></i><b>5.2</b> Filter the SNP data file to only include MAF SNPS selected in Section3</a></li>
<li class="chapter" data-level="5.3" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#specify-number-of-markers-from-each-chromosome-you-will-select"><i class="fa fa-check"></i><b>5.3</b> Specify number of markers from each chromosome you will select</a></li>
<li class="chapter" data-level="5.4" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#use-greedy-algorithm-to-select-5200-snps-for-5k-array-200-extras"><i class="fa fa-check"></i><b>5.4</b> Use greedy algorithm to select 5200 snps for 5K array (200 extras)</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#make-snp_tbl"><i class="fa fa-check"></i><b>5.4.1</b> Make snp_tbl</a></li>
<li class="chapter" data-level="5.4.2" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#check-and-make-sure-the-algorithm-gave-sensible-results"><i class="fa fa-check"></i><b>5.4.2</b> Check and make sure the algorithm gave sensible results</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#add-pop-diagnostic-snps-to-maf-snps-for-final-list"><i class="fa fa-check"></i><b>5.5</b> Add Pop Diagnostic SNPs to MAF SNPs for final list</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="summary-plots-of-snps-for-panel.html"><a href="summary-plots-of-snps-for-panel.html"><i class="fa fa-check"></i><b>6</b> Summary Plots of SNPs for Panel</a>
<ul>
<li class="chapter" data-level="6.1" data-path="summary-plots-of-snps-for-panel.html"><a href="summary-plots-of-snps-for-panel.html#distribution-of-snps"><i class="fa fa-check"></i><b>6.1</b> Distribution of SNPs</a></li>
<li class="chapter" data-level="6.2" data-path="summary-plots-of-snps-for-panel.html"><a href="summary-plots-of-snps-for-panel.html#maf-per-population"><i class="fa fa-check"></i><b>6.2</b> MAF per population</a></li>
<li class="chapter" data-level="6.3" data-path="summary-plots-of-snps-for-panel.html"><a href="summary-plots-of-snps-for-panel.html#pca-with-selected-snps"><i class="fa fa-check"></i><b>6.3</b> PCA with selected SNPs</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Selecting SNPs for SALT array</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genetic-groups" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Genetic Groups<a href="genetic-groups.html#genetic-groups" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="objective-assign-individuals-to-genetic-groups" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Objective: Assign individuals to genetic groups<a href="genetic-groups.html#objective-assign-individuals-to-genetic-groups" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Previous research has suggested that there are four different genetic clusters in the gulf: STX, nGOM, nFL (northern Florida), and genFL (general Florida).</p>
<p>The objective of this section is to use the genotype data from the 600K SNP array to assign oysters to genetic groups.</p>
</div>
<div id="data-subsetting" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Data Subsetting<a href="genetic-groups.html#data-subsetting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Read in SNP data – combined_filtered data set</p>
<ul>
<li><p>row names are individuals</p></li>
<li><p>columns are loci</p></li>
<li><p>cells are genotypes (11,12,21,22)</p></li>
</ul>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="genetic-groups.html#cb43-1" aria-hidden="true" tabindex="-1"></a>combined_filtered <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/derived/combined_filtered.txt&quot;</span>,<span class="at">header=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>Read in Individual data – sample_info_full data set</p>
<ul>
<li><p>first column is individuals</p></li>
<li><p>52 other columns of descriptive information on individuals</p></li>
</ul>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="genetic-groups.html#cb44-1" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/derived/sample_info_full.txt&quot;</span>,<span class="at">header=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>Produce list of individuals that are from selected lines and take the individuals from the selected lines out of the SNP data set</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="genetic-groups.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Produce a list of individuals that are from selected lines (O&#39;Boy or Auburn selected lines)</span></span>
<span id="cb45-2"><a href="genetic-groups.html#cb45-2" aria-hidden="true" tabindex="-1"></a>selected <span class="ot">&lt;-</span> sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb45-3"><a href="genetic-groups.html#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Collection_data <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Auburn selected line 1&quot;</span>, <span class="st">&quot;Auburn selected line 2&quot;</span>, <span class="st">&quot;O&#39;Boy selected line&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-4"><a href="genetic-groups.html#cb45-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Sample)</span>
<span id="cb45-5"><a href="genetic-groups.html#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="genetic-groups.html#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Take the individuals from selected lines out of the SNP data set</span></span>
<span id="cb45-7"><a href="genetic-groups.html#cb45-7" aria-hidden="true" tabindex="-1"></a>genotypes <span class="ot">&lt;-</span> combined_filtered <span class="sc">%&gt;%</span></span>
<span id="cb45-8"><a href="genetic-groups.html#cb45-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">row.names</span>(.) <span class="sc">%in%</span> selected)</span></code></pre></div>
<p>Subset the SNP data set by selecting a random 20K SNPs</p>
<ul>
<li>should speed up analysis and be just as effective for genetic group assignment</li>
</ul>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="genetic-groups.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">54</span>)</span>
<span id="cb46-2"><a href="genetic-groups.html#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="genetic-groups.html#cb46-3" aria-hidden="true" tabindex="-1"></a>combined_filtered_20K <span class="ot">&lt;-</span> genotypes <span class="sc">%&gt;%</span></span>
<span id="cb46-4"><a href="genetic-groups.html#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">sample</span>(<span class="fu">seq_len</span>(<span class="fu">ncol</span>(.)), <span class="at">size =</span> <span class="dv">20000</span>,<span class="at">replace=</span><span class="cn">FALSE</span>))</span></code></pre></div>
</div>
<div id="kmeans---all-wild-individuals" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Kmeans - All Wild Individuals<a href="genetic-groups.html#kmeans---all-wild-individuals" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Prior groups need to be defined for DAPC</p>
<ul>
<li><p>identify groups initially using k-means clustering</p></li>
<li><p>k-means is a clustering algorithm which finds a given number of groups that maximizes variation among groups</p></li>
<li><p>k-means is run sequentially with increasing number of ks, and solutions can be compared using BIC</p></li>
<li><p>ideally, lowest BIC is best clustering solution</p></li>
<li><p>k-means can be done much faster on the PCA data than the raw data</p></li>
<li><p>transform the data using PCA with the find.clusters function, asks the user to specify the number of retained PCS interactively</p></li>
</ul>
<p>First, Convert SNP data to a genind object</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="genetic-groups.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co">#convert SNP data to genind</span></span>
<span id="cb47-2"><a href="genetic-groups.html#cb47-2" aria-hidden="true" tabindex="-1"></a>gen1 <span class="ot">&lt;-</span> <span class="fu">df2genind</span>(combined_filtered_20K, <span class="at">ncode =</span> <span class="dv">1</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>, <span class="at">NA.char =</span> <span class="cn">NA</span>, <span class="at">ploidy =</span> <span class="dv">2</span>)</span>
<span id="cb47-3"><a href="genetic-groups.html#cb47-3" aria-hidden="true" tabindex="-1"></a>gen1</span></code></pre></div>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 380 individuals; 20,000 loci; 38,841 alleles; size: 66.8 Mb
## 
##  // Basic content
##    @tab:  380 x 38841 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 1-2)
##    @loc.fac: locus factor for the 38841 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 2-2)
##    @type:  codom
##    @call: df2genind(X = combined_filtered_20K, sep = &quot;&quot;, ncode = 1, NA.char = NA, 
##     ploidy = 2)
## 
##  // Optional content
##    - empty -</code></pre>
<p>Use find.clusters to identify groups via k-means clustering</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="genetic-groups.html#cb49-1" aria-hidden="true" tabindex="-1"></a>grp <span class="ot">&lt;-</span> <span class="fu">find.clusters</span>(gen1, <span class="at">n.pca =</span> <span class="dv">500</span>, <span class="at">choose =</span> <span class="cn">FALSE</span>, <span class="at">max.n.clust =</span> <span class="dv">5</span>) <span class="co"># retain 500 PCs, retain 3 clusters</span></span>
<span id="cb49-2"><a href="genetic-groups.html#cb49-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-3"><a href="genetic-groups.html#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(grp<span class="sc">$</span>Kstat, <span class="at">type=</span><span class="st">&quot;o&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;number of clusters (K)&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;BIC&quot;</span>,</span>
<span id="cb49-4"><a href="genetic-groups.html#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="at">col=</span><span class="st">&quot;blue&quot;</span>, <span class="at">main=</span><span class="st">&quot;Detection based on BIC&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
<p>Looks like there are three or four groups</p>
<p>Choose three clusters</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="genetic-groups.html#cb50-1" aria-hidden="true" tabindex="-1"></a>grp <span class="ot">&lt;-</span> <span class="fu">find.clusters</span>(gen1, <span class="at">n.pca =</span> <span class="dv">500</span>, <span class="at">n.clust =</span> <span class="dv">3</span>)</span>
<span id="cb50-2"><a href="genetic-groups.html#cb50-2" aria-hidden="true" tabindex="-1"></a>grp<span class="sc">$</span>size</span></code></pre></div>
<pre><code>## [1]  62  98 220</code></pre>
<p>Use the k-means clustering and the sample metadata to create a tibble that contains the 380 individuals that are wild collected and 54 columns of descriptive information, including the kmeans group assignment “k_group”</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="genetic-groups.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Returns data frame of individuals and their k-means assignment</span></span>
<span id="cb52-2"><a href="genetic-groups.html#cb52-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(grp<span class="sc">$</span>grp,<span class="at">rownames=</span><span class="cn">NA</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-3"><a href="genetic-groups.html#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(.,<span class="at">var=</span><span class="st">&quot;Sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-4"><a href="genetic-groups.html#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">k_group =</span> value)</span></code></pre></div>
<p>Make a data set that only has wild individuals</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="genetic-groups.html#cb53-1" aria-hidden="true" tabindex="-1"></a>sample_info_wild <span class="ot">&lt;-</span> sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb53-2"><a href="genetic-groups.html#cb53-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> Collection_data <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Auburn selected line 1&quot;</span>, <span class="st">&quot;Auburn selected line 2&quot;</span>, <span class="st">&quot;O&#39;Boy selected line&quot;</span>))</span></code></pre></div>
<p>Add a column of k_means group to data on individuals</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="genetic-groups.html#cb54-1" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> sample_info_wild <span class="sc">%&gt;%</span></span>
<span id="cb54-2"><a href="genetic-groups.html#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., x, </span>
<span id="cb54-3"><a href="genetic-groups.html#cb54-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">&quot;Sample&quot;</span>)</span></code></pre></div>
</div>
<div id="pca---k-groups" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> PCA - K groups<a href="genetic-groups.html#pca---k-groups" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Use PCAs to confirm there are 3 groups as identified by k means</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="genetic-groups.html#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co">#identify each individual to each k group in the genind</span></span>
<span id="cb55-2"><a href="genetic-groups.html#cb55-2" aria-hidden="true" tabindex="-1"></a>kg<span class="ot">&lt;-</span>sample_data <span class="sc">%&gt;%</span></span>
<span id="cb55-3"><a href="genetic-groups.html#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Sample,k_group)</span>
<span id="cb55-4"><a href="genetic-groups.html#cb55-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-5"><a href="genetic-groups.html#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign strata to genind</span></span>
<span id="cb55-6"><a href="genetic-groups.html#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="fu">strata</span>(gen1) <span class="ot">&lt;-</span> kg[<span class="fu">match</span>(<span class="fu">indNames</span>(gen1),sample_data<span class="sc">$</span>Sample),]</span>
<span id="cb55-7"><a href="genetic-groups.html#cb55-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-8"><a href="genetic-groups.html#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="co"># set pop to GG/STX</span></span>
<span id="cb55-9"><a href="genetic-groups.html#cb55-9" aria-hidden="true" tabindex="-1"></a><span class="fu">setPop</span>(gen1) <span class="ot">&lt;-</span> <span class="er">~</span>k_group</span></code></pre></div>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="genetic-groups.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Check PCA eigenvalues</span></span>
<span id="cb56-2"><a href="genetic-groups.html#cb56-2" aria-hidden="true" tabindex="-1"></a>x<span class="ot">&lt;-</span> <span class="fu">scaleGen</span>(gen1, <span class="at">NA.method =</span> <span class="st">&quot;mean&quot;</span>)</span>
<span id="cb56-3"><a href="genetic-groups.html#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="genetic-groups.html#cb56-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot a PCA </span></span>
<span id="cb56-5"><a href="genetic-groups.html#cb56-5" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">dudi.pca</span>(x,<span class="at">cent=</span><span class="cn">FALSE</span>,<span class="at">scale=</span><span class="cn">FALSE</span>,<span class="at">scannf=</span><span class="cn">FALSE</span>,<span class="at">nf=</span><span class="dv">4</span>)</span>
<span id="cb56-6"><a href="genetic-groups.html#cb56-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-7"><a href="genetic-groups.html#cb56-7" aria-hidden="true" tabindex="-1"></a><span class="co">#plot1</span></span>
<span id="cb56-8"><a href="genetic-groups.html#cb56-8" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(pca<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>],<span class="at">main=</span><span class="st">&quot;All_Samples_PCA_eigenvalues&quot;</span>, <span class="at">col=</span><span class="fu">heat.colors</span>(<span class="dv">50</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-45-1.png" width="672" /></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="genetic-groups.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co">#extract PCs for individuals</span></span>
<span id="cb57-2"><a href="genetic-groups.html#cb57-2" aria-hidden="true" tabindex="-1"></a>PC_ind <span class="ot">&lt;-</span> <span class="fu">PC.ind</span>(pca)</span>
<span id="cb57-3"><a href="genetic-groups.html#cb57-3" aria-hidden="true" tabindex="-1"></a>eig <span class="ot">&lt;-</span> <span class="fu">eigenvalues</span>(pca)</span>
<span id="cb57-4"><a href="genetic-groups.html#cb57-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-5"><a href="genetic-groups.html#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="co">#associate metadata with individuals </span></span>
<span id="cb57-6"><a href="genetic-groups.html#cb57-6" aria-hidden="true" tabindex="-1"></a>PC_ind <span class="ot">&lt;-</span> PC_ind <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Sample=</span>Sample_ID) <span class="sc">%&gt;%</span></span>
<span id="cb57-7"><a href="genetic-groups.html#cb57-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,sample_data,<span class="at">by=</span><span class="st">&quot;Sample&quot;</span>)</span>
<span id="cb57-8"><a href="genetic-groups.html#cb57-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-9"><a href="genetic-groups.html#cb57-9" aria-hidden="true" tabindex="-1"></a>pctot12 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(PC_ind, <span class="fu">aes</span>(<span class="at">x =</span> Axis1, <span class="at">y =</span> Axis2, <span class="at">color =</span> k_group)) <span class="sc">+</span></span>
<span id="cb57-10"><a href="genetic-groups.html#cb57-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geom_text(size=3)+</span></span>
<span id="cb57-11"><a href="genetic-groups.html#cb57-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb57-12"><a href="genetic-groups.html#cb57-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste</span>(<span class="st">&quot;PC1:&quot;</span>, <span class="fu">round</span>(eig[<span class="dv">1</span>, <span class="dv">3</span>], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">&quot;%&quot;</span>), </span>
<span id="cb57-13"><a href="genetic-groups.html#cb57-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">paste</span>(<span class="st">&quot;PC2:&quot;</span>, <span class="fu">round</span>(eig[<span class="dv">2</span>, <span class="dv">3</span>], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">&quot;%&quot;</span>))<span class="sc">+</span></span>
<span id="cb57-14"><a href="genetic-groups.html#cb57-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_hue</span>(<span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb57-15"><a href="genetic-groups.html#cb57-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;All_Samples_Axis2vs.Axis1&quot;</span>)</span>
<span id="cb57-16"><a href="genetic-groups.html#cb57-16" aria-hidden="true" tabindex="-1"></a>pctot12</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-46"></span>
<img src="_main_files/figure-html/unnamed-chunk-46-1.png" alt="All Individuals PC1 vs. PC2" width="672" />
<p class="caption">
Figure 3.1: All Individuals PC1 vs. PC2
</p>
</div>
<p>Re-run PCA without k_group 1 (very different from rest)</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="genetic-groups.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co">#remove k_group 1</span></span>
<span id="cb58-2"><a href="genetic-groups.html#cb58-2" aria-hidden="true" tabindex="-1"></a>gen2 <span class="ot">&lt;-</span> gen1[<span class="sc">!</span>gen1<span class="sc">@</span>strata<span class="sc">$</span>k_group <span class="sc">%in%</span> <span class="dv">1</span>]</span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="genetic-groups.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Check PCA eigenvalues</span></span>
<span id="cb59-2"><a href="genetic-groups.html#cb59-2" aria-hidden="true" tabindex="-1"></a>x<span class="ot">&lt;-</span> <span class="fu">scaleGen</span>(gen2, <span class="at">NA.method =</span> <span class="st">&quot;mean&quot;</span>)</span>
<span id="cb59-3"><a href="genetic-groups.html#cb59-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-4"><a href="genetic-groups.html#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot a PCA </span></span>
<span id="cb59-5"><a href="genetic-groups.html#cb59-5" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">dudi.pca</span>(x,<span class="at">cent=</span><span class="cn">FALSE</span>,<span class="at">scale=</span><span class="cn">FALSE</span>,<span class="at">scannf=</span><span class="cn">FALSE</span>,<span class="at">nf=</span><span class="dv">4</span>)</span>
<span id="cb59-6"><a href="genetic-groups.html#cb59-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-7"><a href="genetic-groups.html#cb59-7" aria-hidden="true" tabindex="-1"></a><span class="co">#plot1</span></span>
<span id="cb59-8"><a href="genetic-groups.html#cb59-8" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(pca<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>],<span class="at">main=</span><span class="st">&quot;All_Samples_PCA_eigenvalues&quot;</span>, <span class="at">col=</span><span class="fu">heat.colors</span>(<span class="dv">50</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-48-1.png" width="672" /></p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="genetic-groups.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co">#extract PCs for individuals</span></span>
<span id="cb60-2"><a href="genetic-groups.html#cb60-2" aria-hidden="true" tabindex="-1"></a>PC_ind <span class="ot">&lt;-</span> <span class="fu">PC.ind</span>(pca)</span>
<span id="cb60-3"><a href="genetic-groups.html#cb60-3" aria-hidden="true" tabindex="-1"></a>eig <span class="ot">&lt;-</span> <span class="fu">eigenvalues</span>(pca)</span>
<span id="cb60-4"><a href="genetic-groups.html#cb60-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-5"><a href="genetic-groups.html#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="co">#associate metadata with individuals </span></span>
<span id="cb60-6"><a href="genetic-groups.html#cb60-6" aria-hidden="true" tabindex="-1"></a>PC_ind <span class="ot">&lt;-</span> PC_ind <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Sample=</span>Sample_ID) <span class="sc">%&gt;%</span></span>
<span id="cb60-7"><a href="genetic-groups.html#cb60-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,sample_data,<span class="at">by=</span><span class="st">&quot;Sample&quot;</span>)</span>
<span id="cb60-8"><a href="genetic-groups.html#cb60-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-9"><a href="genetic-groups.html#cb60-9" aria-hidden="true" tabindex="-1"></a>pctot12 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(PC_ind, <span class="fu">aes</span>(<span class="at">x =</span> Axis1, <span class="at">y =</span> Axis2, <span class="at">color =</span> k_group)) <span class="sc">+</span></span>
<span id="cb60-10"><a href="genetic-groups.html#cb60-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geom_text(size=3)+</span></span>
<span id="cb60-11"><a href="genetic-groups.html#cb60-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb60-12"><a href="genetic-groups.html#cb60-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste</span>(<span class="st">&quot;PC1:&quot;</span>, <span class="fu">round</span>(eig[<span class="dv">1</span>, <span class="dv">3</span>], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">&quot;%&quot;</span>), </span>
<span id="cb60-13"><a href="genetic-groups.html#cb60-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">paste</span>(<span class="st">&quot;PC2:&quot;</span>, <span class="fu">round</span>(eig[<span class="dv">2</span>, <span class="dv">3</span>], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">&quot;%&quot;</span>))<span class="sc">+</span></span>
<span id="cb60-14"><a href="genetic-groups.html#cb60-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_hue</span>(<span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb60-15"><a href="genetic-groups.html#cb60-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;All_Samples_Axis2vs.Axis1&quot;</span>)</span>
<span id="cb60-16"><a href="genetic-groups.html#cb60-16" aria-hidden="true" tabindex="-1"></a>pctot12</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-49"></span>
<img src="_main_files/figure-html/unnamed-chunk-49-1.png" alt="All Individuals PC1 vs. PC2" width="672" />
<p class="caption">
Figure 3.2: All Individuals PC1 vs. PC2
</p>
</div>
<p>Looks like at least three groups in the PCA above</p>
<p>Worth evaluating the PCA if choosing four groups</p>
</div>
<div id="kmeans-and-pca-with-four-groups" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Kmeans and PCA with four groups<a href="genetic-groups.html#kmeans-and-pca-with-four-groups" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="genetic-groups.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co">#choose four clusters</span></span>
<span id="cb61-2"><a href="genetic-groups.html#cb61-2" aria-hidden="true" tabindex="-1"></a>grp <span class="ot">&lt;-</span> <span class="fu">find.clusters</span>(gen1, <span class="at">n.pca =</span> <span class="dv">500</span>, <span class="at">n.clust =</span> <span class="dv">4</span>)</span>
<span id="cb61-3"><a href="genetic-groups.html#cb61-3" aria-hidden="true" tabindex="-1"></a>grp<span class="sc">$</span>size</span></code></pre></div>
<pre><code>## [1]  62  33  68 217</code></pre>
<p>Use the k-means clustering and the sample metadata to create a tibble that contains the 380 individuals that are wild collected and 54 columns of descriptive information, including the kmeans group assignment “k_group”</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="genetic-groups.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Returns data frame of individuals and their k-means assignment</span></span>
<span id="cb63-2"><a href="genetic-groups.html#cb63-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(grp<span class="sc">$</span>grp,<span class="at">rownames=</span><span class="cn">NA</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-3"><a href="genetic-groups.html#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(.,<span class="at">var=</span><span class="st">&quot;Sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-4"><a href="genetic-groups.html#cb63-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">k_group =</span> value)</span></code></pre></div>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="genetic-groups.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># add a column of k_means group to data on individuals</span></span>
<span id="cb64-2"><a href="genetic-groups.html#cb64-2" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> sample_info_wild <span class="sc">%&gt;%</span></span>
<span id="cb64-3"><a href="genetic-groups.html#cb64-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., x, </span>
<span id="cb64-4"><a href="genetic-groups.html#cb64-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">&quot;Sample&quot;</span>)</span></code></pre></div>
</div>
<div id="pca---four-k-groups" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> PCA - Four K groups<a href="genetic-groups.html#pca---four-k-groups" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Use PCAs to confirm there are 4 groups</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="genetic-groups.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co">#identify each individual to each k group in the genind</span></span>
<span id="cb65-2"><a href="genetic-groups.html#cb65-2" aria-hidden="true" tabindex="-1"></a>kg<span class="ot">&lt;-</span>sample_data <span class="sc">%&gt;%</span></span>
<span id="cb65-3"><a href="genetic-groups.html#cb65-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Sample,k_group)</span>
<span id="cb65-4"><a href="genetic-groups.html#cb65-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-5"><a href="genetic-groups.html#cb65-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign strata to genind</span></span>
<span id="cb65-6"><a href="genetic-groups.html#cb65-6" aria-hidden="true" tabindex="-1"></a><span class="fu">strata</span>(gen1) <span class="ot">&lt;-</span> kg[<span class="fu">match</span>(<span class="fu">indNames</span>(gen1),sample_data<span class="sc">$</span>Sample),]</span>
<span id="cb65-7"><a href="genetic-groups.html#cb65-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-8"><a href="genetic-groups.html#cb65-8" aria-hidden="true" tabindex="-1"></a><span class="co"># set pop to GG/STX</span></span>
<span id="cb65-9"><a href="genetic-groups.html#cb65-9" aria-hidden="true" tabindex="-1"></a><span class="fu">setPop</span>(gen1) <span class="ot">&lt;-</span> <span class="er">~</span>k_group</span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="genetic-groups.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Check PCA eigenvalues</span></span>
<span id="cb66-2"><a href="genetic-groups.html#cb66-2" aria-hidden="true" tabindex="-1"></a>x<span class="ot">&lt;-</span> <span class="fu">scaleGen</span>(gen1, <span class="at">NA.method =</span> <span class="st">&quot;mean&quot;</span>)</span>
<span id="cb66-3"><a href="genetic-groups.html#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="genetic-groups.html#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot a PCA </span></span>
<span id="cb66-5"><a href="genetic-groups.html#cb66-5" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">dudi.pca</span>(x,<span class="at">cent=</span><span class="cn">FALSE</span>,<span class="at">scale=</span><span class="cn">FALSE</span>,<span class="at">scannf=</span><span class="cn">FALSE</span>,<span class="at">nf=</span><span class="dv">4</span>)</span>
<span id="cb66-6"><a href="genetic-groups.html#cb66-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-7"><a href="genetic-groups.html#cb66-7" aria-hidden="true" tabindex="-1"></a><span class="co">#plot1</span></span>
<span id="cb66-8"><a href="genetic-groups.html#cb66-8" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(pca<span class="sc">$</span>eig[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>],<span class="at">main=</span><span class="st">&quot;All_Samples_PCA_eigenvalues&quot;</span>, <span class="at">col=</span><span class="fu">heat.colors</span>(<span class="dv">50</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-54-1.png" width="672" /></p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="genetic-groups.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co">#extract PCs for individuals</span></span>
<span id="cb67-2"><a href="genetic-groups.html#cb67-2" aria-hidden="true" tabindex="-1"></a>PC_ind <span class="ot">&lt;-</span> <span class="fu">PC.ind</span>(pca)</span>
<span id="cb67-3"><a href="genetic-groups.html#cb67-3" aria-hidden="true" tabindex="-1"></a>eig <span class="ot">&lt;-</span> <span class="fu">eigenvalues</span>(pca)</span>
<span id="cb67-4"><a href="genetic-groups.html#cb67-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-5"><a href="genetic-groups.html#cb67-5" aria-hidden="true" tabindex="-1"></a><span class="co">#associate metadata with individuals </span></span>
<span id="cb67-6"><a href="genetic-groups.html#cb67-6" aria-hidden="true" tabindex="-1"></a>PC_ind <span class="ot">&lt;-</span> PC_ind <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Sample=</span>Sample_ID) <span class="sc">%&gt;%</span></span>
<span id="cb67-7"><a href="genetic-groups.html#cb67-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,sample_data,<span class="at">by=</span><span class="st">&quot;Sample&quot;</span>)</span>
<span id="cb67-8"><a href="genetic-groups.html#cb67-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-9"><a href="genetic-groups.html#cb67-9" aria-hidden="true" tabindex="-1"></a>pctot12 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(PC_ind, <span class="fu">aes</span>(<span class="at">x =</span> Axis1, <span class="at">y =</span> Axis2, <span class="at">color =</span> k_group)) <span class="sc">+</span></span>
<span id="cb67-10"><a href="genetic-groups.html#cb67-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geom_text(size=3)+</span></span>
<span id="cb67-11"><a href="genetic-groups.html#cb67-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb67-12"><a href="genetic-groups.html#cb67-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste</span>(<span class="st">&quot;PC1:&quot;</span>, <span class="fu">round</span>(eig[<span class="dv">1</span>, <span class="dv">3</span>], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">&quot;%&quot;</span>), </span>
<span id="cb67-13"><a href="genetic-groups.html#cb67-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">paste</span>(<span class="st">&quot;PC2:&quot;</span>, <span class="fu">round</span>(eig[<span class="dv">2</span>, <span class="dv">3</span>], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">&quot;%&quot;</span>))<span class="sc">+</span></span>
<span id="cb67-14"><a href="genetic-groups.html#cb67-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_hue</span>(<span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb67-15"><a href="genetic-groups.html#cb67-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;All_Samples_Axis2vs.Axis1&quot;</span>)</span>
<span id="cb67-16"><a href="genetic-groups.html#cb67-16" aria-hidden="true" tabindex="-1"></a>pctot12</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-55"></span>
<img src="_main_files/figure-html/unnamed-chunk-55-1.png" alt="All Individuals PC1 vs. PC2" width="672" />
<p class="caption">
Figure 3.3: All Individuals PC1 vs. PC2
</p>
</div>
<p>I would say there are four groups based on the K means + PCA. Proceed with these four.</p>
</div>
<div id="assign-genetic-groups" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> Assign Genetic Groups<a href="genetic-groups.html#assign-genetic-groups" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, investigate each k_group and give it a common name based on geographic origin of the majority of the individuals</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="genetic-groups.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co">#K_group1</span></span>
<span id="cb68-2"><a href="genetic-groups.html#cb68-2" aria-hidden="true" tabindex="-1"></a>sample_data <span class="sc">%&gt;%</span></span>
<span id="cb68-3"><a href="genetic-groups.html#cb68-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(k_group <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb68-4"><a href="genetic-groups.html#cb68-4" aria-hidden="true" tabindex="-1"></a><span class="fu">count</span>(SampleOrigin,Geographic_state,Geographic_location, Geographic_code)</span></code></pre></div>
<pre><code>##   SampleOrigin Geographic_state Geographic_location Geographic_code  n
## 1  South-Texas            Texas  Corpus Christi Bay            CCTX  6
## 2  South-Texas            Texas      Port Mansfield            PMTX 17
## 3  South-Texas            Texas  Upper Laguna Madre            ULTX 39</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="genetic-groups.html#cb70-1" aria-hidden="true" tabindex="-1"></a>MainStates <span class="ot">&lt;-</span> <span class="fu">map_data</span>(<span class="st">&quot;state&quot;</span>)</span>
<span id="cb70-2"><a href="genetic-groups.html#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="genetic-groups.html#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb70-4"><a href="genetic-groups.html#cb70-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;lightgray&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb70-5"><a href="genetic-groups.html#cb70-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb70-6"><a href="genetic-groups.html#cb70-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb70-7"><a href="genetic-groups.html#cb70-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb70-8"><a href="genetic-groups.html#cb70-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>)) <span class="sc">+</span> </span>
<span id="cb70-9"><a href="genetic-groups.html#cb70-9" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">data =</span> sample_data <span class="sc">%&gt;%</span></span>
<span id="cb70-10"><a href="genetic-groups.html#cb70-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">latitude2 =</span> <span class="fu">round</span>(latitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb70-11"><a href="genetic-groups.html#cb70-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">longitude2 =</span> <span class="fu">round</span>(longitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb70-12"><a href="genetic-groups.html#cb70-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(k_group,Geographic_code,longitude2,latitude2) <span class="sc">%&gt;%</span></span>
<span id="cb70-13"><a href="genetic-groups.html#cb70-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">&#39;longitude2&#39;</span>, <span class="st">&#39;latitude2&#39;</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb70-14"><a href="genetic-groups.html#cb70-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(k_group <span class="sc">==</span> <span class="dv">1</span>),</span>
<span id="cb70-15"><a href="genetic-groups.html#cb70-15" aria-hidden="true" tabindex="-1"></a>    <span class="co">#counts for proportional representation</span></span>
<span id="cb70-16"><a href="genetic-groups.html#cb70-16" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> latitude2, <span class="at">x =</span> longitude2, <span class="at">group=</span>Geographic_code,<span class="at">size =</span> n), <span class="at">colour=</span> <span class="st">&quot;forestgreen&quot;</span>,</span>
<span id="cb70-17"><a href="genetic-groups.html#cb70-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">stroke=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-57-1.png" width="672" />
k_group 1 is south texas</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="genetic-groups.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co">#k group 2</span></span>
<span id="cb71-2"><a href="genetic-groups.html#cb71-2" aria-hidden="true" tabindex="-1"></a>sample_data <span class="sc">%&gt;%</span></span>
<span id="cb71-3"><a href="genetic-groups.html#cb71-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(k_group <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb71-4"><a href="genetic-groups.html#cb71-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(SampleOrigin,Geographic_state,Geographic_location, Geographic_code)</span></code></pre></div>
<pre><code>##   SampleOrigin Geographic_state Geographic_location Geographic_code  n
## 1   Broodstock          Alabama      Alligator Lake            ALAL  2
## 2   Broodstock          Florida    Alligator Harbor            AHFL  9
## 3   Broodstock          Florida          Oyster Bay            OBFL 10
## 4   Broodstock          Florida       Pensacola Bay            PBFL 12</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="genetic-groups.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb73-2"><a href="genetic-groups.html#cb73-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;lightgray&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb73-3"><a href="genetic-groups.html#cb73-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb73-4"><a href="genetic-groups.html#cb73-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb73-5"><a href="genetic-groups.html#cb73-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb73-6"><a href="genetic-groups.html#cb73-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>)) <span class="sc">+</span> </span>
<span id="cb73-7"><a href="genetic-groups.html#cb73-7" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">data =</span> sample_data <span class="sc">%&gt;%</span></span>
<span id="cb73-8"><a href="genetic-groups.html#cb73-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">latitude2 =</span> <span class="fu">round</span>(latitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb73-9"><a href="genetic-groups.html#cb73-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">longitude2 =</span> <span class="fu">round</span>(longitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb73-10"><a href="genetic-groups.html#cb73-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(k_group,Geographic_code,longitude2,latitude2) <span class="sc">%&gt;%</span></span>
<span id="cb73-11"><a href="genetic-groups.html#cb73-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">&#39;longitude2&#39;</span>, <span class="st">&#39;latitude2&#39;</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb73-12"><a href="genetic-groups.html#cb73-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(k_group <span class="sc">==</span> <span class="dv">2</span>),</span>
<span id="cb73-13"><a href="genetic-groups.html#cb73-13" aria-hidden="true" tabindex="-1"></a>    <span class="co">#counts for proportional representation</span></span>
<span id="cb73-14"><a href="genetic-groups.html#cb73-14" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> latitude2, <span class="at">x =</span> longitude2, <span class="at">group=</span>Geographic_code,<span class="at">size =</span> n), <span class="at">colour=</span> <span class="st">&quot;forestgreen&quot;</span>,</span>
<span id="cb73-15"><a href="genetic-groups.html#cb73-15" aria-hidden="true" tabindex="-1"></a>             <span class="at">stroke=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-59-1.png" width="672" />
k_group 2 is north florida (nfl)</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="genetic-groups.html#cb74-1" aria-hidden="true" tabindex="-1"></a>sample_data <span class="sc">%&gt;%</span></span>
<span id="cb74-2"><a href="genetic-groups.html#cb74-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(k_group <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb74-3"><a href="genetic-groups.html#cb74-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(SampleOrigin,Geographic_state,Geographic_location,Geographic_code)</span></code></pre></div>
<pre><code>##        SampleOrigin Geographic_state             Geographic_location Geographic_code  n
## 1  Backwater-Bay-FL          Florida                   Backwater Bay           BWBFL 24
## 2        Broodstock          Florida                Alligator Harbor            AHFL  1
## 3        Broodstock          Florida                   Corrigan Reef            CRFL 10
## 4        Broodstock          Florida                    Seahorse Key            SKFL 10
## 5        Broodstock          Florida                  Suwannee River            LRFL 10
## 6     South-Florida          Florida Boca Ciega Bay Aquatic Preserve            BCFL  3
## 7     South-Florida          Florida            Caloosahatchee River            CAFL  4
## 8     South-Florida          Florida                   Madelaine Key            MKFL  2
## 9     South-Florida          Florida                       Tampa Bay            TBFL  2
## 10      South-Texas            Texas                  Port Mansfield            PMTX  2</code></pre>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="genetic-groups.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb76-2"><a href="genetic-groups.html#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;lightgray&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb76-3"><a href="genetic-groups.html#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb76-4"><a href="genetic-groups.html#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb76-5"><a href="genetic-groups.html#cb76-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb76-6"><a href="genetic-groups.html#cb76-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>)) <span class="sc">+</span> </span>
<span id="cb76-7"><a href="genetic-groups.html#cb76-7" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">data =</span> sample_data <span class="sc">%&gt;%</span></span>
<span id="cb76-8"><a href="genetic-groups.html#cb76-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">latitude2 =</span> <span class="fu">round</span>(latitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb76-9"><a href="genetic-groups.html#cb76-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">longitude2 =</span> <span class="fu">round</span>(longitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb76-10"><a href="genetic-groups.html#cb76-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(k_group,Geographic_code,longitude2,latitude2) <span class="sc">%&gt;%</span></span>
<span id="cb76-11"><a href="genetic-groups.html#cb76-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">&#39;longitude2&#39;</span>, <span class="st">&#39;latitude2&#39;</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb76-12"><a href="genetic-groups.html#cb76-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(k_group <span class="sc">==</span> <span class="dv">3</span>),</span>
<span id="cb76-13"><a href="genetic-groups.html#cb76-13" aria-hidden="true" tabindex="-1"></a>    <span class="co">#counts for proportional representation</span></span>
<span id="cb76-14"><a href="genetic-groups.html#cb76-14" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> latitude2, <span class="at">x =</span> longitude2, <span class="at">group=</span>Geographic_code,<span class="at">size =</span> n), <span class="at">colour=</span> <span class="st">&quot;forestgreen&quot;</span>,</span>
<span id="cb76-15"><a href="genetic-groups.html#cb76-15" aria-hidden="true" tabindex="-1"></a>             <span class="at">stroke=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-61-1.png" width="672" />
k_group 3 is general florida (gfl)</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="genetic-groups.html#cb77-1" aria-hidden="true" tabindex="-1"></a>sample_data <span class="sc">%&gt;%</span></span>
<span id="cb77-2"><a href="genetic-groups.html#cb77-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(k_group <span class="sc">==</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb77-3"><a href="genetic-groups.html#cb77-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(SampleOrigin,Geographic_state,Geographic_location,Geographic_code)</span></code></pre></div>
<pre><code>##     SampleOrigin Geographic_state Geographic_location Geographic_code  n
## 1     Broodstock          Alabama      Alligator Lake            ALAL 16
## 2     Broodstock          Alabama         Cedar Point            CPAL 20
## 3     Broodstock          Florida       Pensacola Bay            PBFL  2
## 4     Broodstock        Louisiana        Caillou Lake            SLLA 22
## 5     Broodstock        Louisiana      Calcasieu Lake            LCLA 24
## 6     Broodstock      Mississippi    Pascagoula River            PSMS 27
## 7     Broodstock            Texas  East Galveston Bay            EGTX 15
## 8     Broodstock            Texas  East Matagorda Bay            EMTX 18
## 9     Broodstock            Texas         Sabine Pass            LSTX 23
## 10    Broodstock            Texas     San Antonio Bay            SATX 11
## 11    Broodstock            Texas  West Galveston Bay            WGTX 10
## 12    Broodstock            Texas  West Matagorda Bay            WMTX 16
## 13 South-Florida          Florida           Tampa Bay            TBFL  2
## 14   South-Texas            Texas  Corpus Christi Bay            CCTX  8
## 15   South-Texas            Texas  Upper Laguna Madre            ULTX  3</code></pre>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="genetic-groups.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb79-2"><a href="genetic-groups.html#cb79-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;lightgray&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb79-3"><a href="genetic-groups.html#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb79-4"><a href="genetic-groups.html#cb79-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb79-5"><a href="genetic-groups.html#cb79-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb79-6"><a href="genetic-groups.html#cb79-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>)) <span class="sc">+</span> </span>
<span id="cb79-7"><a href="genetic-groups.html#cb79-7" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">data =</span> sample_data <span class="sc">%&gt;%</span></span>
<span id="cb79-8"><a href="genetic-groups.html#cb79-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">latitude2 =</span> <span class="fu">round</span>(latitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb79-9"><a href="genetic-groups.html#cb79-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">longitude2 =</span> <span class="fu">round</span>(longitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb79-10"><a href="genetic-groups.html#cb79-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(k_group,Geographic_code,longitude2,latitude2) <span class="sc">%&gt;%</span></span>
<span id="cb79-11"><a href="genetic-groups.html#cb79-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">&#39;longitude2&#39;</span>, <span class="st">&#39;latitude2&#39;</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb79-12"><a href="genetic-groups.html#cb79-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(k_group <span class="sc">==</span> <span class="dv">4</span>),</span>
<span id="cb79-13"><a href="genetic-groups.html#cb79-13" aria-hidden="true" tabindex="-1"></a>    <span class="co">#counts for proportional representation</span></span>
<span id="cb79-14"><a href="genetic-groups.html#cb79-14" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> latitude2, <span class="at">x =</span> longitude2, <span class="at">group=</span>Geographic_code,<span class="at">size =</span> n), <span class="at">colour=</span> <span class="st">&quot;forestgreen&quot;</span>,</span>
<span id="cb79-15"><a href="genetic-groups.html#cb79-15" aria-hidden="true" tabindex="-1"></a>             <span class="at">stroke=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-63-1.png" width="672" />
k_group 4 is northern gulf of mexico (ngom)</p>
<p>Add column in data file that specifies each “k_group” with “genetic group”</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="genetic-groups.html#cb80-1" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> sample_data <span class="sc">%&gt;%</span></span>
<span id="cb80-2"><a href="genetic-groups.html#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mutate</span>(<span class="at">gen_group =</span> <span class="fu">case_when</span>(</span>
<span id="cb80-3"><a href="genetic-groups.html#cb80-3" aria-hidden="true" tabindex="-1"></a>    k_group <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;stx&quot;</span>,</span>
<span id="cb80-4"><a href="genetic-groups.html#cb80-4" aria-hidden="true" tabindex="-1"></a>    k_group <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;nfl&quot;</span>,</span>
<span id="cb80-5"><a href="genetic-groups.html#cb80-5" aria-hidden="true" tabindex="-1"></a>    k_group <span class="sc">==</span> <span class="dv">3</span> <span class="sc">~</span> <span class="st">&quot;gfl&quot;</span>,</span>
<span id="cb80-6"><a href="genetic-groups.html#cb80-6" aria-hidden="true" tabindex="-1"></a>    k_group <span class="sc">==</span> <span class="dv">4</span> <span class="sc">~</span> <span class="st">&quot;ngom&quot;</span>))</span></code></pre></div>
</div>
<div id="summary" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Summary<a href="genetic-groups.html#summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The script above takes all of the individuals in the array data and assigns them to genetic groups via kmeans clustering using 20K randomly selected snps.
The data is in the “sample_data” tibble. The genetic groups are defined as below:</p>
<p>stx = south texas
ngom = northern gulf of mexico
nfl = northern florida
sfl = southern florida</p>
<p>The table below (“summary”) summarizes the data</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="genetic-groups.html#cb81-1" aria-hidden="true" tabindex="-1"></a>summary<span class="ot">&lt;-</span> sample_data <span class="sc">%&gt;%</span></span>
<span id="cb81-2"><a href="genetic-groups.html#cb81-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Sample,sample_id,Geographic_code,Geographic_state,gen_group) <span class="sc">%&gt;%</span></span>
<span id="cb81-3"><a href="genetic-groups.html#cb81-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Geographic_state,gen_group) <span class="sc">%&gt;%</span></span>
<span id="cb81-4"><a href="genetic-groups.html#cb81-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>() <span class="sc">%&gt;%</span></span>
<span id="cb81-5"><a href="genetic-groups.html#cb81-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">sample_location =</span> Geographic_state, <span class="at">genetic_group =</span> gen_group)</span></code></pre></div>
</div>
<div id="save-files-1" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> Save files<a href="genetic-groups.html#save-files-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="genetic-groups.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co">#save file of individual data without selected individuals and with genetic groups assigned through k_means</span></span>
<span id="cb82-2"><a href="genetic-groups.html#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(sample_data, <span class="at">file =</span> <span class="st">&quot;data//derived//sample_data&quot;</span>)</span>
<span id="cb82-3"><a href="genetic-groups.html#cb82-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-4"><a href="genetic-groups.html#cb82-4" aria-hidden="true" tabindex="-1"></a><span class="co">#save file of SNP data without selected individuals</span></span>
<span id="cb82-5"><a href="genetic-groups.html#cb82-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(genotypes, <span class="at">file =</span> <span class="st">&quot;data//derived//genotypes&quot;</span>)</span>
<span id="cb82-6"><a href="genetic-groups.html#cb82-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-7"><a href="genetic-groups.html#cb82-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(combined_filtered_20K, <span class="at">file =</span> <span class="st">&quot;data//derived//combined_filtered_20K&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="input-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="select-candidate-snps.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="book_files/gitbook-2.6.7/js/app.min.js"></script>
<script src="book_files/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="book_files/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"toolbar": {
"position": "static"
}
});
});
</script>

</body>

</html>
