<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Selecting 5K SNPs for SALT array</title>
  <meta name="description" content="This is a series of markdown files demonstrating how SNPs were selected for the SALT oyster array. It is formatted as a series of chapters. Each chapter builds on the previous with the ultimate product of ~5K candidate SNPs to send to Paragon for producing a 5K SNP array for the SALT consortium." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Selecting 5K SNPs for SALT array" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a series of markdown files demonstrating how SNPs were selected for the SALT oyster array. It is formatted as a series of chapters. Each chapter builds on the previous with the ultimate product of ~5K candidate SNPs to send to Paragon for producing a 5K SNP array for the SALT consortium." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Selecting 5K SNPs for SALT array" />
  
  <meta name="twitter:description" content="This is a series of markdown files demonstrating how SNPs were selected for the SALT oyster array. It is formatted as a series of chapters. Each chapter builds on the previous with the ultimate product of ~5K candidate SNPs to send to Paragon for producing a 5K SNP array for the SALT consortium." />
  

<meta name="author" content="Joseph Matt" />


<meta name="date" content="2023-12-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="input-data.html"/>
<script src="book_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="book_files/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="book_files/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="book_files/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="book_files/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>1</b> Input Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="input-data.html"><a href="input-data.html#objective-organize-individual-and-snp-data"><i class="fa fa-check"></i><b>1.1</b> Objective: Organize individual and SNP data</a></li>
<li class="chapter" data-level="1.2" data-path="input-data.html"><a href="input-data.html#individual-sample-information"><i class="fa fa-check"></i><b>1.2</b> Individual Sample Information</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="input-data.html"><a href="input-data.html#basic-sample-information"><i class="fa fa-check"></i><b>1.2.1</b> Basic sample information</a></li>
<li class="chapter" data-level="1.2.2" data-path="input-data.html"><a href="input-data.html#quality-control-info-from-thermo-fisher"><i class="fa fa-check"></i><b>1.2.2</b> Quality control info from Thermo Fisher</a></li>
<li class="chapter" data-level="1.2.3" data-path="input-data.html"><a href="input-data.html#list-of-oysters-that-passed-the-genotyping-according-to-thermofisher"><i class="fa fa-check"></i><b>1.2.3</b> List of oysters that passed the genotyping according to ThermoFisher</a></li>
<li class="chapter" data-level="1.2.4" data-path="input-data.html"><a href="input-data.html#merge-files"><i class="fa fa-check"></i><b>1.2.4</b> Merge Files</a></li>
<li class="chapter" data-level="1.2.5" data-path="input-data.html"><a href="input-data.html#tidy-up-latlong-data-and-site-codes"><i class="fa fa-check"></i><b>1.2.5</b> Tidy Up Lat/Long Data and Site Codes</a></li>
<li class="chapter" data-level="1.2.6" data-path="input-data.html"><a href="input-data.html#organize-oysters-into-metagroup-categories"><i class="fa fa-check"></i><b>1.2.6</b> Organize oysters into “metagroup” categories:</a></li>
<li class="chapter" data-level="1.2.7" data-path="input-data.html"><a href="input-data.html#map-sites-where-samples-were-collected"><i class="fa fa-check"></i><b>1.2.7</b> Map sites where samples were collected</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="input-data.html"><a href="input-data.html#snp-information"><i class="fa fa-check"></i><b>1.3</b> SNP Information</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="input-data.html"><a href="input-data.html#general-information-on-snps"><i class="fa fa-check"></i><b>1.3.1</b> General Information on SNPs</a></li>
<li class="chapter" data-level="1.3.2" data-path="input-data.html"><a href="input-data.html#recommended-snps"><i class="fa fa-check"></i><b>1.3.2</b> Recommended SNPs</a></li>
<li class="chapter" data-level="1.3.3" data-path="input-data.html"><a href="input-data.html#quality-control-info"><i class="fa fa-check"></i><b>1.3.3</b> Quality control info</a></li>
<li class="chapter" data-level="1.3.4" data-path="input-data.html"><a href="input-data.html#merge-files-1"><i class="fa fa-check"></i><b>1.3.4</b> Merge Files</a></li>
<li class="chapter" data-level="1.3.5" data-path="input-data.html"><a href="input-data.html#formatting-for-hierfstat-and-adegenet"><i class="fa fa-check"></i><b>1.3.5</b> Formatting for hierfstat and adegenet</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="input-data.html"><a href="input-data.html#save-files"><i class="fa fa-check"></i><b>1.4</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="genetic-groups.html"><a href="genetic-groups.html"><i class="fa fa-check"></i><b>2</b> Genetic Groups</a>
<ul>
<li class="chapter" data-level="2.1" data-path="genetic-groups.html"><a href="genetic-groups.html#objective-assign-individuals-to-genetic-groups"><i class="fa fa-check"></i><b>2.1</b> Objective: Assign individuals to genetic groups</a></li>
<li class="chapter" data-level="2.2" data-path="genetic-groups.html"><a href="genetic-groups.html#data-subsetting"><i class="fa fa-check"></i><b>2.2</b> Data Subsetting</a></li>
<li class="chapter" data-level="2.3" data-path="genetic-groups.html"><a href="genetic-groups.html#kmeans---all-wild-individuals"><i class="fa fa-check"></i><b>2.3</b> Kmeans - All Wild Individuals</a></li>
<li class="chapter" data-level="2.4" data-path="genetic-groups.html"><a href="genetic-groups.html#pca---k-groups"><i class="fa fa-check"></i><b>2.4</b> PCA - K groups</a></li>
<li class="chapter" data-level="2.5" data-path="genetic-groups.html"><a href="genetic-groups.html#kmeans-and-pca-with-four-groups"><i class="fa fa-check"></i><b>2.5</b> Kmeans and PCA with four groups</a></li>
<li class="chapter" data-level="2.6" data-path="genetic-groups.html"><a href="genetic-groups.html#pca---four-k-groups"><i class="fa fa-check"></i><b>2.6</b> PCA - Four K groups</a></li>
<li class="chapter" data-level="2.7" data-path="genetic-groups.html"><a href="genetic-groups.html#assign-genetic-groups"><i class="fa fa-check"></i><b>2.7</b> Assign Genetic Groups</a></li>
<li class="chapter" data-level="2.8" data-path="genetic-groups.html"><a href="genetic-groups.html#summary"><i class="fa fa-check"></i><b>2.8</b> Summary</a></li>
<li class="chapter" data-level="2.9" data-path="genetic-groups.html"><a href="genetic-groups.html#save-files-1"><i class="fa fa-check"></i><b>2.9</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html"><i class="fa fa-check"></i><b>3</b> Select Candidate SNPs</a>
<ul>
<li class="chapter" data-level="3.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#objective-identify-candidate-snps-by"><i class="fa fa-check"></i><b>3.1</b> Objective: Identify candidate SNPs by</a></li>
<li class="chapter" data-level="3.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#read-in-data"><i class="fa fa-check"></i><b>3.2</b> Read in Data</a></li>
<li class="chapter" data-level="3.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#format-data"><i class="fa fa-check"></i><b>3.3</b> Format Data</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genind-object-from-105k-snps"><i class="fa fa-check"></i><b>3.3.1</b> Make genind object from 105K SNPs</a></li>
<li class="chapter" data-level="3.3.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-a-dataframe-specifying-genetic-group-and-metapop."><i class="fa fa-check"></i><b>3.3.2</b> Make a dataframe specifying genetic group and “metapop.”</a></li>
<li class="chapter" data-level="3.3.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#assign-populations-to-genind-object"><i class="fa fa-check"></i><b>3.3.3</b> Assign populations to genind object</a></li>
<li class="chapter" data-level="3.3.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#format-data-into-the-hierfstat-format-for-analysis-of-heterozygosity-and-maf"><i class="fa fa-check"></i><b>3.3.4</b> Format data into the hierfstat format for analysis of heterozygosity and MAF</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#hwe-filtering"><i class="fa fa-check"></i><b>3.4</b> HWE filtering</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#filtering-out-the-loci-with-ho-0.5"><i class="fa fa-check"></i><b>3.4.1</b> Filtering out the loci with Ho &gt; 0.5</a></li>
<li class="chapter" data-level="3.4.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#re-analyze-the-data-for-heterozygosities"><i class="fa fa-check"></i><b>3.4.2</b> Re-analyze the data for heterozygosities</a></li>
<li class="chapter" data-level="3.4.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genind-object-without-the-loci-that-have-heterozygosities-over-0.5"><i class="fa fa-check"></i><b>3.4.3</b> Make genind object without the loci that have heterozygosities over 0.5</a></li>
<li class="chapter" data-level="3.4.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genotypes-data-frame-without-the-loci-that-have-heterozygosities-over-0.5"><i class="fa fa-check"></i><b>3.4.4</b> Make genotypes data frame without the loci that have heterozygosities over 0.5</a></li>
<li class="chapter" data-level="3.4.5" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#hw-test"><i class="fa fa-check"></i><b>3.4.5</b> HW test</a></li>
<li class="chapter" data-level="3.4.6" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-loci-that-fail-hwe-test-p-0.05"><i class="fa fa-check"></i><b>3.4.6</b> Select loci that fail HWE test (p &lt; 0.05)</a></li>
<li class="chapter" data-level="3.4.7" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#remove-loci-out-of-hwe-from-the-genind-object-and-genotypes-data-frame"><i class="fa fa-check"></i><b>3.4.7</b> Remove loci out of HWE from the genind object and genotypes data frame</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-population-diagnostic-loci"><i class="fa fa-check"></i><b>3.5</b> Select population diagnostic loci</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#south-texas-vs.-gulf"><i class="fa fa-check"></i><b>3.5.1</b> South Texas vs. Gulf</a></li>
<li class="chapter" data-level="3.5.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-25-snps-that-distinguish-stx-from-gulf"><i class="fa fa-check"></i><b>3.5.2</b> Select 25 SNPs that distinguish stx from gulf</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#ngom-vs.-fl"><i class="fa fa-check"></i><b>3.6</b> nGOM vs. FL</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-25-snps-with-highest-fst-for-differentiating-nfl-vs.-gfl-vs.-ngom"><i class="fa fa-check"></i><b>3.6.1</b> Select 25 SNPs with highest Fst for differentiating nfl vs. gfl vs. ngom</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#maf"><i class="fa fa-check"></i><b>3.7</b> MAF</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#calculate-minor-allele-frequency-for-each-locus-in-each-genetic-group"><i class="fa fa-check"></i><b>3.7.1</b> Calculate Minor Allele Frequency for each locus in each genetic group</a></li>
<li class="chapter" data-level="3.7.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#re-calculate-after-removing-loci-with-maf-of-1"><i class="fa fa-check"></i><b>3.7.2</b> Re-calculate after removing loci with MAF of 1</a></li>
<li class="chapter" data-level="3.7.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#identify-loci-with-high-maf-in-all-groups"><i class="fa fa-check"></i><b>3.7.3</b> Identify loci with high MAF in all groups</a></li>
<li class="chapter" data-level="3.7.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-maf-loci-and-exmaine-histograms-of-loci"><i class="fa fa-check"></i><b>3.7.4</b> Select MAF loci and exmaine histograms of loci</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#save-files-2"><i class="fa fa-check"></i><b>3.8</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html"><i class="fa fa-check"></i><b>4</b> Select SNPs for Panel</a>
<ul>
<li class="chapter" data-level="4.1" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#read-in-all-data-on-snps-and-relevant-data-from-section-3"><i class="fa fa-check"></i><b>4.1</b> Read in all data on SNPs and relevant data from Section 3</a></li>
<li class="chapter" data-level="4.2" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#filter-the-snp-data-file-to-only-include-maf-snps-selected-in-section3"><i class="fa fa-check"></i><b>4.2</b> Filter the SNP data file to only include MAF SNPS selected in Section3</a></li>
<li class="chapter" data-level="4.3" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#specify-number-of-markers-from-each-chromosome-you-will-select"><i class="fa fa-check"></i><b>4.3</b> Specify number of markers from each chromosome you will select</a></li>
<li class="chapter" data-level="4.4" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#use-greedy-algorithm-to-select-5200-snps-for-5k-array-200-extras"><i class="fa fa-check"></i><b>4.4</b> Use greedy algorithm to select 5200 snps for 5K array (200 extras)</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#make-snp_tbl"><i class="fa fa-check"></i><b>4.4.1</b> Make snp_tbl</a></li>
<li class="chapter" data-level="4.4.2" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#check-and-make-sure-the-algorithm-gave-sensible-results"><i class="fa fa-check"></i><b>4.4.2</b> Check and make sure the algorithm gave sensible results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="add-pop-diagnostic-snps-to-maf-snps-for-final-list.html"><a href="add-pop-diagnostic-snps-to-maf-snps-for-final-list.html"><i class="fa fa-check"></i><b>5</b> Add Pop Diagnostic SNPs to MAF SNPs for final list</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Selecting 5K SNPs for SALT array</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Selecting 5K SNPs for SALT array</h1>
<p class="author"><em>Joseph Matt</em></p>
<p class="date"><em>2023-12-11</em></p>
</div>
<div id="preface" class="section level1 unnumbered hasAnchor">
<h1>Preface<a href="index.html#preface" class="anchor-section" aria-label="Anchor link to header"></a></h1>

</div>
            </section>

          </div>
        </div>
      </div>

<a href="input-data.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="book_files/gitbook-2.6.7/js/app.min.js"></script>
<script src="book_files/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="book_files/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"toolbar": {
"position": "static"
}
});
});
</script>

</body>

</html>
