<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Input Data | Selecting SNPs for SALT array</title>
  <meta name="description" content="2 Input Data | Selecting SNPs for SALT array" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Input Data | Selecting SNPs for SALT array" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Input Data | Selecting SNPs for SALT array" />
  
  
  

<meta name="author" content="Joseph Matt" />


<meta name="date" content="2023-12-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="genetic-groups.html"/>
<script src="book_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="book_files/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="book_files/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="book_files/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="book_files/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#objective-organize-individual-and-snp-data"><i class="fa fa-check"></i><b>2.1</b> Objective: Organize individual and SNP data</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#individual-sample-information"><i class="fa fa-check"></i><b>2.2</b> Individual Sample Information</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="input-data.html"><a href="input-data.html#basic-sample-information"><i class="fa fa-check"></i><b>2.2.1</b> Basic sample information</a></li>
<li class="chapter" data-level="2.2.2" data-path="input-data.html"><a href="input-data.html#quality-control-info-from-thermo-fisher"><i class="fa fa-check"></i><b>2.2.2</b> Quality control info from Thermo Fisher</a></li>
<li class="chapter" data-level="2.2.3" data-path="input-data.html"><a href="input-data.html#list-of-oysters-that-passed-the-genotyping-according-to-thermofisher"><i class="fa fa-check"></i><b>2.2.3</b> List of oysters that passed the genotyping according to ThermoFisher</a></li>
<li class="chapter" data-level="2.2.4" data-path="input-data.html"><a href="input-data.html#merge-files"><i class="fa fa-check"></i><b>2.2.4</b> Merge Files</a></li>
<li class="chapter" data-level="2.2.5" data-path="input-data.html"><a href="input-data.html#tidy-up-latlong-data-and-site-codes"><i class="fa fa-check"></i><b>2.2.5</b> Tidy Up Lat/Long Data and Site Codes</a></li>
<li class="chapter" data-level="2.2.6" data-path="input-data.html"><a href="input-data.html#organize-oysters-into-metagroup-categories"><i class="fa fa-check"></i><b>2.2.6</b> Organize oysters into “metagroup” categories:</a></li>
<li class="chapter" data-level="2.2.7" data-path="input-data.html"><a href="input-data.html#map-sites-where-samples-were-collected"><i class="fa fa-check"></i><b>2.2.7</b> Map sites where samples were collected</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#snp-information"><i class="fa fa-check"></i><b>2.3</b> SNP Information</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="input-data.html"><a href="input-data.html#general-information-on-snps"><i class="fa fa-check"></i><b>2.3.1</b> General Information on SNPs</a></li>
<li class="chapter" data-level="2.3.2" data-path="input-data.html"><a href="input-data.html#recommended-snps"><i class="fa fa-check"></i><b>2.3.2</b> Recommended SNPs</a></li>
<li class="chapter" data-level="2.3.3" data-path="input-data.html"><a href="input-data.html#quality-control-info"><i class="fa fa-check"></i><b>2.3.3</b> Quality control info</a></li>
<li class="chapter" data-level="2.3.4" data-path="input-data.html"><a href="input-data.html#merge-files-1"><i class="fa fa-check"></i><b>2.3.4</b> Merge Files</a></li>
<li class="chapter" data-level="2.3.5" data-path="input-data.html"><a href="input-data.html#formatting-for-hierfstat-and-adegenet"><i class="fa fa-check"></i><b>2.3.5</b> Formatting for hierfstat and adegenet</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#save-files"><i class="fa fa-check"></i><b>2.4</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genetic-groups.html"><a href="genetic-groups.html"><i class="fa fa-check"></i><b>3</b> Genetic Groups</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genetic-groups.html"><a href="genetic-groups.html#objective-assign-individuals-to-genetic-groups"><i class="fa fa-check"></i><b>3.1</b> Objective: Assign individuals to genetic groups</a></li>
<li class="chapter" data-level="3.2" data-path="genetic-groups.html"><a href="genetic-groups.html#data-subsetting"><i class="fa fa-check"></i><b>3.2</b> Data Subsetting</a></li>
<li class="chapter" data-level="3.3" data-path="genetic-groups.html"><a href="genetic-groups.html#kmeans---all-wild-individuals"><i class="fa fa-check"></i><b>3.3</b> Kmeans - All Wild Individuals</a></li>
<li class="chapter" data-level="3.4" data-path="genetic-groups.html"><a href="genetic-groups.html#pca---k-groups"><i class="fa fa-check"></i><b>3.4</b> PCA - K groups</a></li>
<li class="chapter" data-level="3.5" data-path="genetic-groups.html"><a href="genetic-groups.html#kmeans-and-pca-with-four-groups"><i class="fa fa-check"></i><b>3.5</b> Kmeans and PCA with four groups</a></li>
<li class="chapter" data-level="3.6" data-path="genetic-groups.html"><a href="genetic-groups.html#pca---four-k-groups"><i class="fa fa-check"></i><b>3.6</b> PCA - Four K groups</a></li>
<li class="chapter" data-level="3.7" data-path="genetic-groups.html"><a href="genetic-groups.html#assign-genetic-groups"><i class="fa fa-check"></i><b>3.7</b> Assign Genetic Groups</a></li>
<li class="chapter" data-level="3.8" data-path="genetic-groups.html"><a href="genetic-groups.html#summary"><i class="fa fa-check"></i><b>3.8</b> Summary</a></li>
<li class="chapter" data-level="3.9" data-path="genetic-groups.html"><a href="genetic-groups.html#save-files-1"><i class="fa fa-check"></i><b>3.9</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html"><i class="fa fa-check"></i><b>4</b> Select Candidate SNPs</a>
<ul>
<li class="chapter" data-level="4.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#objective-identify-candidate-snps-by"><i class="fa fa-check"></i><b>4.1</b> Objective: Identify candidate SNPs by</a></li>
<li class="chapter" data-level="4.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#read-in-data"><i class="fa fa-check"></i><b>4.2</b> Read in Data</a></li>
<li class="chapter" data-level="4.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#format-data"><i class="fa fa-check"></i><b>4.3</b> Format Data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genind-object-from-105k-snps"><i class="fa fa-check"></i><b>4.3.1</b> Make genind object from 105K SNPs</a></li>
<li class="chapter" data-level="4.3.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-a-dataframe-specifying-genetic-group-and-metapop."><i class="fa fa-check"></i><b>4.3.2</b> Make a dataframe specifying genetic group and “metapop.”</a></li>
<li class="chapter" data-level="4.3.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#assign-populations-to-genind-object"><i class="fa fa-check"></i><b>4.3.3</b> Assign populations to genind object</a></li>
<li class="chapter" data-level="4.3.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#format-data-into-the-hierfstat-format-for-analysis-of-heterozygosity-and-maf"><i class="fa fa-check"></i><b>4.3.4</b> Format data into the hierfstat format for analysis of heterozygosity and MAF</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#hwe-filtering"><i class="fa fa-check"></i><b>4.4</b> HWE filtering</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#filtering-out-the-loci-with-ho-0.5"><i class="fa fa-check"></i><b>4.4.1</b> Filtering out the loci with Ho &gt; 0.5</a></li>
<li class="chapter" data-level="4.4.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#re-analyze-the-data-for-heterozygosities"><i class="fa fa-check"></i><b>4.4.2</b> Re-analyze the data for heterozygosities</a></li>
<li class="chapter" data-level="4.4.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genind-object-without-the-loci-that-have-heterozygosities-over-0.5"><i class="fa fa-check"></i><b>4.4.3</b> Make genind object without the loci that have heterozygosities over 0.5</a></li>
<li class="chapter" data-level="4.4.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genotypes-data-frame-without-the-loci-that-have-heterozygosities-over-0.5"><i class="fa fa-check"></i><b>4.4.4</b> Make genotypes data frame without the loci that have heterozygosities over 0.5</a></li>
<li class="chapter" data-level="4.4.5" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#hw-test"><i class="fa fa-check"></i><b>4.4.5</b> HW test</a></li>
<li class="chapter" data-level="4.4.6" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-loci-that-fail-hwe-test-p-0.05"><i class="fa fa-check"></i><b>4.4.6</b> Select loci that fail HWE test (p &lt; 0.05)</a></li>
<li class="chapter" data-level="4.4.7" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#remove-loci-out-of-hwe-from-the-genind-object-and-genotypes-data-frame"><i class="fa fa-check"></i><b>4.4.7</b> Remove loci out of HWE from the genind object and genotypes data frame</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-population-diagnostic-loci"><i class="fa fa-check"></i><b>4.5</b> Select population diagnostic loci</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#south-texas-vs.-gulf"><i class="fa fa-check"></i><b>4.5.1</b> South Texas vs. Gulf</a></li>
<li class="chapter" data-level="4.5.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-25-snps-that-distinguish-stx-from-gulf"><i class="fa fa-check"></i><b>4.5.2</b> Select 25 SNPs that distinguish stx from gulf</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#ngom-vs.-fl"><i class="fa fa-check"></i><b>4.6</b> nGOM vs. FL</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-25-snps-with-highest-fst-for-differentiating-nfl-vs.-gfl-vs.-ngom"><i class="fa fa-check"></i><b>4.6.1</b> Select 25 SNPs with highest Fst for differentiating nfl vs. gfl vs. ngom</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#maf"><i class="fa fa-check"></i><b>4.7</b> MAF</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#calculate-minor-allele-frequency-for-each-locus-in-each-genetic-group"><i class="fa fa-check"></i><b>4.7.1</b> Calculate Minor Allele Frequency for each locus in each genetic group</a></li>
<li class="chapter" data-level="4.7.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#re-calculate-after-removing-loci-with-maf-of-1"><i class="fa fa-check"></i><b>4.7.2</b> Re-calculate after removing loci with MAF of 1</a></li>
<li class="chapter" data-level="4.7.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#identify-loci-with-high-maf-in-all-groups"><i class="fa fa-check"></i><b>4.7.3</b> Identify loci with high MAF in all groups</a></li>
<li class="chapter" data-level="4.7.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-maf-loci-and-exmaine-histograms-of-loci"><i class="fa fa-check"></i><b>4.7.4</b> Select MAF loci and exmaine histograms of loci</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#save-files-2"><i class="fa fa-check"></i><b>4.8</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html"><i class="fa fa-check"></i><b>5</b> Select SNPs for Panel</a>
<ul>
<li class="chapter" data-level="5.1" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#read-in-all-data-on-snps-and-relevant-data-from-section-3"><i class="fa fa-check"></i><b>5.1</b> Read in all data on SNPs and relevant data from Section 3</a></li>
<li class="chapter" data-level="5.2" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#filter-the-snp-data-file-to-only-include-maf-snps-selected-in-section3"><i class="fa fa-check"></i><b>5.2</b> Filter the SNP data file to only include MAF SNPS selected in Section3</a></li>
<li class="chapter" data-level="5.3" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#specify-number-of-markers-from-each-chromosome-you-will-select"><i class="fa fa-check"></i><b>5.3</b> Specify number of markers from each chromosome you will select</a></li>
<li class="chapter" data-level="5.4" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#use-greedy-algorithm-to-select-5200-snps-for-5k-array-200-extras"><i class="fa fa-check"></i><b>5.4</b> Use greedy algorithm to select 5200 snps for 5K array (200 extras)</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#make-snp_tbl"><i class="fa fa-check"></i><b>5.4.1</b> Make snp_tbl</a></li>
<li class="chapter" data-level="5.4.2" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#check-and-make-sure-the-algorithm-gave-sensible-results"><i class="fa fa-check"></i><b>5.4.2</b> Check and make sure the algorithm gave sensible results</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#add-pop-diagnostic-snps-to-maf-snps-for-final-list"><i class="fa fa-check"></i><b>5.5</b> Add Pop Diagnostic SNPs to MAF SNPs for final list</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="summary-plots-of-snps-for-panel.html"><a href="summary-plots-of-snps-for-panel.html"><i class="fa fa-check"></i><b>6</b> Summary Plots of SNPs for Panel</a>
<ul>
<li class="chapter" data-level="6.1" data-path="summary-plots-of-snps-for-panel.html"><a href="summary-plots-of-snps-for-panel.html#distribution-of-snps"><i class="fa fa-check"></i><b>6.1</b> Distribution of SNPs</a></li>
<li class="chapter" data-level="6.2" data-path="summary-plots-of-snps-for-panel.html"><a href="summary-plots-of-snps-for-panel.html#maf-per-population"><i class="fa fa-check"></i><b>6.2</b> MAF per population</a></li>
<li class="chapter" data-level="6.3" data-path="summary-plots-of-snps-for-panel.html"><a href="summary-plots-of-snps-for-panel.html#pca-with-selected-snps"><i class="fa fa-check"></i><b>6.3</b> PCA with selected SNPs</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Selecting SNPs for SALT array</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="input-data" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Input Data<a href="input-data.html#input-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="objective-organize-individual-and-snp-data" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Objective: Organize individual and SNP data<a href="input-data.html#objective-organize-individual-and-snp-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Organize and consolidate data from hundreds of C.virginica collected around the Gulf of Mexico genotyped on a high density SNP panel (~600K) developed by the East Coast consortium.</p>
</div>
<div id="individual-sample-information" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Individual Sample Information<a href="input-data.html#individual-sample-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>General Gulf (GG) samples (n=404) and South Texas (STX) samples (n=76) were genotyped on 600K ThermoFisher array and analyzed separately. A unique genotyping probeset list was used for each sample set. 447 samples passed QA.</p>
<div id="basic-sample-information" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Basic sample information<a href="input-data.html#basic-sample-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>“sample_info_SALT” data file provided by Eric Salliant of University of Southern Mississippi, includes data provided by ThermoFisher and metadata (e.g., geographic details of where oysters were collected).</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="input-data.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(sample_info_SALT <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;raw&quot;</span>,<span class="st">&quot;cvir_gulf_600K_ES.csv&quot;</span>)))</span>
<span id="cb1-2"><a href="input-data.html#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="input-data.html#cb1-3" aria-hidden="true" tabindex="-1"></a>sample_info_SALT<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="input-data.html#cb2-1" aria-hidden="true" tabindex="-1"></a>sample_info_SALT</span></code></pre></div>
<pre><code>## # A tibble: 480 × 16
##    Sample          Filename SubmittedPlate sample_id Geographic_code Collection_data SampleName
##    &lt;chr&gt;           &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;           &lt;chr&gt;     
##  1 a551319-444806… 9SR48340 F12            BWBFL_1   BWBFL           25.92860, -81.… Customer  
##  2 a551319-444806… 9SR48340 G08            BWBFL_10  BWBFL           25.92860, -81.… Customer  
##  3 a551319-444806… 9SR48340 G09            BWBFL_11  BWBFL           25.92860, -81.… Customer  
##  4 a551319-444806… 9SR48340 G10            BWBFL_12  BWBFL           25.92860, -81.… Customer  
##  5 a551319-444806… 9SR48340 G11            BWBFL_14  BWBFL           25.92860, -81.… Customer  
##  6 a551319-444806… 9SR48340 G12            BWBFL_15  BWBFL           25.92860, -81.… Customer  
##  7 a551319-444806… 9SR48340 H01            BWBFL_16  BWBFL           25.92860, -81.… Customer  
##  8 a551319-444806… 9SR48340 H02            BWBFL_17  BWBFL           25.92860, -81.… Customer  
##  9 a551319-444806… 9SR48340 H03            BWBFL_18  BWBFL           25.92860, -81.… Customer  
## 10 a551319-444806… 9SR48340 H04            BWBFL_19  BWBFL           25.92860, -81.… Customer  
## # ℹ 470 more rows
## # ℹ 9 more variables: SampleSource &lt;chr&gt;, SampleType &lt;chr&gt;, Species &lt;chr&gt;,
## #   Concentration &lt;dbl&gt;, PlateName &lt;chr&gt;, Rename &lt;chr&gt;, Reproducibility &lt;dbl&gt;, Gender &lt;chr&gt;,
## #   SampleOrigin &lt;chr&gt;</code></pre>
<p>“sample_info_RAD” data file provided by Amanda Barker of the Marine Genomics Lab, it is data of a TPWD and Marine Genomics Lab project using RAD</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="input-data.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(sample_info_RAD <span class="ot">&lt;-</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;raw&quot;</span>,<span class="st">&quot;oyster_genotyped_sample_info.csv&quot;</span>)))</span>
<span id="cb4-2"><a href="input-data.html#cb4-2" aria-hidden="true" tabindex="-1"></a>sample_info_RAD<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="input-data.html#cb5-1" aria-hidden="true" tabindex="-1"></a>sample_info_RAD</span></code></pre></div>
<pre><code>## # A tibble: 463 × 14
##    lib_id sample_id barcode index    site         lib region gulf_location state bay   latitude
##    &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt;
##  1 apa_01 59751     TCAGT   index_04 apalachic…     1 north… Eastern       FL    Apal…     29.7
##  2 apa_02 59752     AGCTA   index_04 apalachic…     1 north… Eastern       FL    Apal…     29.7
##  3 apa_03 59753     CGGTA   index_04 apalachic…     1 north… Eastern       FL    Apal…     29.7
##  4 apa_04 59754     GGATA   index_04 apalachic…     1 north… Eastern       FL    Apal…     29.7
##  5 apa_05 59755     TCCGG   index_04 apalachic…     1 north… Eastern       FL    Apal…     29.7
##  6 apa_06 59756     ACACA   index_04 apalachic…     1 north… Eastern       FL    Apal…     29.7
##  7 apa_07 59757     GCATG   index_03 apalachic…     1 north… Eastern       FL    Apal…     29.7
##  8 apa_08 59758     ACTGG   index_03 apalachic…     1 north… Eastern       FL    Apal…     29.7
##  9 apa_09 59759     CTGCG   index_03 apalachic…     1 north… Eastern       FL    Apal…     29.7
## 10 apa_10 59760     GTCCG   index_03 apalachic…     1 north… Eastern       FL    Apal…     29.7
## # ℹ 453 more rows
## # ℹ 3 more variables: longitude &lt;dbl&gt;, tidal.zone &lt;chr&gt;, harvest &lt;chr&gt;</code></pre>
<p>Some sample info from the RAD dataset was in a seperate file</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="input-data.html#cb7-1" aria-hidden="true" tabindex="-1"></a>missingdata <span class="ot">&lt;-</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;raw&quot;</span>,<span class="st">&quot;missing_cvir.csv&quot;</span>))</span>
<span id="cb7-2"><a href="input-data.html#cb7-2" aria-hidden="true" tabindex="-1"></a>missingdata</span></code></pre></div>
<pre><code>## # A tibble: 5 × 8
##   sample_id gulf_location state    bay                latitude longitude `tidal zone` harvest
##       &lt;dbl&gt; &lt;chr&gt;         &lt;chr&gt;    &lt;chr&gt;                 &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;        &lt;chr&gt;  
## 1     60111 Southwestern  TX-Lower Upper Laguna Madre     27.6     -97.2 Intertidal   No     
## 2     60113 Southwestern  TX-Lower Upper Laguna Madre     27.6     -97.2 Intertidal   No     
## 3     60008 Southwestern  TX-Lower Port Mansfield         26.6     -97.4 Intertidal   No     
## 4     60012 Southwestern  TX-Lower Port Mansfield         26.6     -97.4 Intertidal   No     
## 5     60013 Southwestern  TX-Lower Port Mansfield         26.6     -97.4 Intertidal   No</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="input-data.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># add missing data with RAD data</span></span>
<span id="cb9-2"><a href="input-data.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb9-3"><a href="input-data.html#cb9-3" aria-hidden="true" tabindex="-1"></a>sample_info_RADx <span class="ot">&lt;-</span> <span class="fu">rbind.fill</span>(sample_info_RAD,missingdata)</span>
<span id="cb9-4"><a href="input-data.html#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">detach</span>(<span class="st">&quot;package:plyr&quot;</span>, <span class="at">unload=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="quality-control-info-from-thermo-fisher" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Quality control info from Thermo Fisher<a href="input-data.html#quality-control-info-from-thermo-fisher" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="input-data.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in QC data for both data sets</span></span>
<span id="cb10-2"><a href="input-data.html#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="input-data.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#not sure how to use read_table to imitate read.table, so &quot;here&quot; tabled for now</span></span>
<span id="cb10-4"><a href="input-data.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#other_qc &lt;- read_table(here::here(&quot;data&quot;,&quot;raw&quot;,&quot;other_genotypingQC.report.txt&quot;), col_names = TRUE)</span></span>
<span id="cb10-5"><a href="input-data.html#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="input-data.html#cb10-6" aria-hidden="true" tabindex="-1"></a>other_qc <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_genotypingQC.report.txt&quot;</span>,<span class="at">fill=</span><span class="cn">TRUE</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-7"><a href="input-data.html#cb10-7" aria-hidden="true" tabindex="-1"></a>south_qc <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_genotypingQC.report.txt&quot;</span>, <span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb10-8"><a href="input-data.html#cb10-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-9"><a href="input-data.html#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co">#add column of data_set - &quot;other&quot; or &quot;south_texas&quot;</span></span>
<span id="cb10-10"><a href="input-data.html#cb10-10" aria-hidden="true" tabindex="-1"></a>other_qc<span class="sc">$</span>data_set <span class="ot">&lt;-</span> <span class="st">&quot;other&quot;</span></span>
<span id="cb10-11"><a href="input-data.html#cb10-11" aria-hidden="true" tabindex="-1"></a>south_qc<span class="sc">$</span>data_set <span class="ot">&lt;-</span> <span class="st">&quot;south_texas&quot;</span></span>
<span id="cb10-12"><a href="input-data.html#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="input-data.html#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co">#rbind &quot;other&quot; and &quot;south_texas&quot; data frames</span></span>
<span id="cb10-14"><a href="input-data.html#cb10-14" aria-hidden="true" tabindex="-1"></a>qc_indv <span class="ot">&lt;-</span> <span class="fu">rbind</span>(other_qc,south_qc)</span>
<span id="cb10-15"><a href="input-data.html#cb10-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-16"><a href="input-data.html#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="co">#rename in QC &quot;celfiles&quot; to &quot;Sample&quot;</span></span>
<span id="cb10-17"><a href="input-data.html#cb10-17" aria-hidden="true" tabindex="-1"></a>qc_indv <span class="ot">&lt;-</span> qc_indv <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Sample =</span> cel_files)</span>
<span id="cb10-18"><a href="input-data.html#cb10-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-19"><a href="input-data.html#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(qc_indv)</span></code></pre></div>
<pre><code>## # A tibble: 473 × 19
##    Sample            computed_gender call_rate total_call_rate het_rate total_het_rate hom_rate
##    &lt;chr&gt;                       &lt;dbl&gt;     &lt;dbl&gt;           &lt;dbl&gt;    &lt;dbl&gt;          &lt;dbl&gt;    &lt;dbl&gt;
##  1 a551319-4448066-…            99.0      99.0            21.6     21.6           77.4     77.4
##  2 a551319-4448066-…            99.3      99.3            22.0     22.0           77.2     77.2
##  3 a551319-4448066-…            99.2      99.2            21.8     21.8           77.4     77.4
##  4 a551319-4448066-…            99.4      99.4            22.0     22.0           77.4     77.4
##  5 a551319-4448066-…            99.4      99.4            22.6     22.6           76.8     76.8
##  6 a551319-4448066-…            99.4      99.4            22.8     22.8           76.6     76.6
##  7 a551319-4448066-…            99.1      99.1            20.9     20.9           78.2     78.2
##  8 a551319-4448066-…            99.4      99.4            22.7     22.7           76.7     76.7
##  9 a551319-4448066-…            99.1      99.1            20.3     20.3           78.8     78.8
## 10 a551319-4448066-…            99.2      99.2            20.8     20.8           78.5     78.5
## # ℹ 463 more rows
## # ℹ 12 more variables: total_hom_rate &lt;dbl&gt;, cluster_distance_mean &lt;dbl&gt;,
## #   cluster_distance_stdev &lt;dbl&gt;, allele_summarization_mean &lt;dbl&gt;,
## #   allele_summarization_stdev &lt;dbl&gt;, allele_deviation_mean &lt;dbl&gt;,
## #   allele_deviation_stdev &lt;dbl&gt;, allele_mad_residuals_mean &lt;dbl&gt;,
## #   allele_mad_residuals_stdev &lt;dbl&gt;, affymetrix.plate.barcode &lt;chr&gt;,
## #   affymetrix.plate.peg.wellposition &lt;lgl&gt;, data_set &lt;chr&gt;</code></pre>
</div>
<div id="list-of-oysters-that-passed-the-genotyping-according-to-thermofisher" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> List of oysters that passed the genotyping according to ThermoFisher<a href="input-data.html#list-of-oysters-that-passed-the-genotyping-according-to-thermofisher" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="input-data.html#cb12-1" aria-hidden="true" tabindex="-1"></a>gg_qc_filt <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_sample_QCFilteredCR.txt&quot;</span>,<span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-2"><a href="input-data.html#cb12-2" aria-hidden="true" tabindex="-1"></a>passed_GG <span class="ot">&lt;-</span> gg_qc_filt<span class="sc">$</span>cel_files</span>
<span id="cb12-3"><a href="input-data.html#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="input-data.html#cb12-4" aria-hidden="true" tabindex="-1"></a>st_qc_filt <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_sample_QCFilteredCR.txt&quot;</span>,<span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-5"><a href="input-data.html#cb12-5" aria-hidden="true" tabindex="-1"></a>passed_ST <span class="ot">&lt;-</span> st_qc_filt<span class="sc">$</span>cel_files</span></code></pre></div>
</div>
<div id="merge-files" class="section level3 hasAnchor" number="2.2.4">
<h3><span class="header-section-number">2.2.4</span> Merge Files<a href="input-data.html#merge-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Consolidate all identifying information on samples to “sample_info_full”</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="input-data.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#merge sample_info_SALT and sample_info_RAD</span></span>
<span id="cb13-2"><a href="input-data.html#cb13-2" aria-hidden="true" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="at">x =</span> sample_info_SALT, <span class="at">y =</span> sample_info_RADx, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-3"><a href="input-data.html#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="input-data.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co">#add quality control data, filter out only individuals that passed Thermo&#39;s quality control</span></span>
<span id="cb13-5"><a href="input-data.html#cb13-5" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="ot">&lt;-</span> sample_info <span class="sc">%&gt;%</span></span>
<span id="cb13-6"><a href="input-data.html#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(qc_indv, <span class="at">all.x =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="input-data.html#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span> (Sample <span class="sc">%in%</span> passed_GG <span class="sc">|</span> Sample <span class="sc">%in%</span> passed_ST) </span>
<span id="cb13-8"><a href="input-data.html#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="input-data.html#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co">#some individuals have multiple rows in the data set, with the only varying information the lib_id, barcode, and index columns, likely RAD information.  Below, choose only one row for these individuals: a551319-4448066-110623-383_A04.CEL and a551319-4448066-110623-383_A08.CEL</span></span>
<span id="cb13-10"><a href="input-data.html#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="input-data.html#cb13-11" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="ot">&lt;-</span> sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb13-12"><a href="input-data.html#cb13-12" aria-hidden="true" tabindex="-1"></a> <span class="fu">distinct</span>(Sample, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-13"><a href="input-data.html#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="input-data.html#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="co">#add a &quot;locaton&quot; column for the RAD data</span></span>
<span id="cb13-15"><a href="input-data.html#cb13-15" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>location <span class="ot">&lt;-</span> <span class="fu">paste</span>(sample_info_full<span class="sc">$</span>bay,sample_info_full<span class="sc">$</span>state,<span class="at">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb13-16"><a href="input-data.html#cb13-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-17"><a href="input-data.html#cb13-17" aria-hidden="true" tabindex="-1"></a><span class="co">#remove all periods and hyphens from sample name (causes issues downstream)</span></span>
<span id="cb13-18"><a href="input-data.html#cb13-18" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>Sample)</span>
<span id="cb13-19"><a href="input-data.html#cb13-19" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>Sample)</span>
<span id="cb13-20"><a href="input-data.html#cb13-20" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>Sample)</span>
<span id="cb13-21"><a href="input-data.html#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="input-data.html#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(sample_info_full)</span></code></pre></div>
<pre><code>## # A tibble: 447 × 49
##    Sample          sample_id Filename SubmittedPlate Geographic_code Collection_data SampleName
##    &lt;chr&gt;           &lt;chr&gt;     &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;           &lt;chr&gt;           &lt;chr&gt;     
##  1 a5513194448066… Auburn_1  9SR48342 A01            Auburn selecte… Auburn selecte… Customer  
##  2 a5513194448066… Auburn_2  9SR48342 A02            Auburn selecte… Auburn selecte… Customer  
##  3 a5513194448066… Auburn_3  9SR48342 A03            Auburn selecte… Auburn selecte… Customer  
##  4 a5513194448066… Auburn_4  9SR48342 A04            Auburn selecte… Auburn selecte… Customer  
##  5 a5513194448066… Auburn_6  9SR48342 A05            Auburn selecte… Auburn selecte… Customer  
##  6 a5513194448066… Auburn_7  9SR48342 A06            Auburn selecte… Auburn selecte… Customer  
##  7 a5513194448066… Auburn_8  9SR48342 A07            Auburn selecte… Auburn selecte… Customer  
##  8 a5513194448066… Auburn_9  9SR48342 A08            Auburn selecte… Auburn selecte… Customer  
##  9 a5513194448066… Auburn_10 9SR48342 A09            Auburn selecte… Auburn selecte… Customer  
## 10 a5513194448066… Auburn_11 9SR48342 A10            Auburn selecte… Auburn selecte… Customer  
## # ℹ 437 more rows
## # ℹ 42 more variables: SampleSource &lt;chr&gt;, SampleType &lt;chr&gt;, Species &lt;chr&gt;,
## #   Concentration &lt;dbl&gt;, PlateName &lt;chr&gt;, Rename &lt;chr&gt;, Reproducibility &lt;dbl&gt;, Gender &lt;chr&gt;,
## #   SampleOrigin &lt;chr&gt;, lib_id &lt;chr&gt;, barcode &lt;chr&gt;, index &lt;chr&gt;, site &lt;chr&gt;, lib &lt;dbl&gt;,
## #   region &lt;chr&gt;, gulf_location &lt;chr&gt;, state &lt;chr&gt;, bay &lt;chr&gt;, latitude &lt;dbl&gt;,
## #   longitude &lt;dbl&gt;, tidal.zone &lt;chr&gt;, harvest &lt;chr&gt;, `tidal zone` &lt;chr&gt;,
## #   computed_gender &lt;dbl&gt;, call_rate &lt;dbl&gt;, total_call_rate &lt;dbl&gt;, het_rate &lt;dbl&gt;, …</code></pre>
</div>
<div id="tidy-up-latlong-data-and-site-codes" class="section level3 hasAnchor" number="2.2.5">
<h3><span class="header-section-number">2.2.5</span> Tidy Up Lat/Long Data and Site Codes<a href="input-data.html#tidy-up-latlong-data-and-site-codes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Fix inconsistencies in geographic data and site codes</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="input-data.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#lat and long data differs in format for samples from SALT and RAD data files</span></span>
<span id="cb15-2"><a href="input-data.html#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="input-data.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 1 - tidy up lat and long from SALT data files</span></span>
<span id="cb15-4"><a href="input-data.html#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="input-data.html#cb15-5" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>lat<span class="ot">&lt;-</span><span class="fu">substr</span>(sample_info_full<span class="sc">$</span>Collection_data,<span class="dv">1</span>,<span class="dv">8</span>)</span>
<span id="cb15-6"><a href="input-data.html#cb15-6" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>long<span class="ot">&lt;-</span><span class="fu">str_sub</span>(sample_info_full<span class="sc">$</span>Collection_data,<span class="sc">-</span><span class="dv">10</span>)</span>
<span id="cb15-7"><a href="input-data.html#cb15-7" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>long <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>long)</span>
<span id="cb15-8"><a href="input-data.html#cb15-8" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>long <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>long)</span>
<span id="cb15-9"><a href="input-data.html#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="input-data.html#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 - add the SALT lat and long data to the latitude and longitude  </span></span>
<span id="cb15-11"><a href="input-data.html#cb15-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="input-data.html#cb15-12" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>latitude <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(sample_info_full<span class="sc">$</span>latitude),</span>
<span id="cb15-13"><a href="input-data.html#cb15-13" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">as.numeric</span>(sample_info_full<span class="sc">$</span>lat),sample_info_full<span class="sc">$</span>latitude)</span>
<span id="cb15-14"><a href="input-data.html#cb15-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-15"><a href="input-data.html#cb15-15" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>longitude <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(sample_info_full<span class="sc">$</span>longitude),</span>
<span id="cb15-16"><a href="input-data.html#cb15-16" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">as.numeric</span>(sample_info_full<span class="sc">$</span>long),sample_info_full<span class="sc">$</span>longitude)</span>
<span id="cb15-17"><a href="input-data.html#cb15-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-18"><a href="input-data.html#cb15-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-19"><a href="input-data.html#cb15-19" aria-hidden="true" tabindex="-1"></a><span class="co"># 3 - as numeric lat and long data</span></span>
<span id="cb15-20"><a href="input-data.html#cb15-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-21"><a href="input-data.html#cb15-21" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>latitude <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(sample_info_full<span class="sc">$</span>latitude)</span>
<span id="cb15-22"><a href="input-data.html#cb15-22" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>longitude <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(sample_info_full<span class="sc">$</span>longitude)</span>
<span id="cb15-23"><a href="input-data.html#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="input-data.html#cb15-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-25"><a href="input-data.html#cb15-25" aria-hidden="true" tabindex="-1"></a><span class="co"># - 4 add &quot;geographic_code&quot; information to those without it </span></span>
<span id="cb15-26"><a href="input-data.html#cb15-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-27"><a href="input-data.html#cb15-27" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code<span class="ot">&lt;-</span><span class="fu">as.character</span>(sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb15-28"><a href="input-data.html#cb15-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-29"><a href="input-data.html#cb15-29" aria-hidden="true" tabindex="-1"></a><span class="co">#ULTX 1/2</span></span>
<span id="cb15-30"><a href="input-data.html#cb15-30" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>bay <span class="sc">%in%</span> <span class="st">&quot;Upper Laguna Madre&quot;</span>,<span class="st">&quot;ULTX&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb15-31"><a href="input-data.html#cb15-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-32"><a href="input-data.html#cb15-32" aria-hidden="true" tabindex="-1"></a><span class="co">#ULTX 2/2</span></span>
<span id="cb15-33"><a href="input-data.html#cb15-33" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>sample_id <span class="sc">%like%</span> <span class="st">&quot;ULM&quot;</span>,<span class="st">&quot;ULTX&quot;</span>,sample_info_full<span class="sc">$</span>Geographic_code) <span class="co">#like requires data.table package</span></span>
<span id="cb15-34"><a href="input-data.html#cb15-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-35"><a href="input-data.html#cb15-35" aria-hidden="true" tabindex="-1"></a><span class="co">#PMTX Port Mansfield Texas</span></span>
<span id="cb15-36"><a href="input-data.html#cb15-36" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>bay <span class="sc">%in%</span> <span class="st">&quot;Port Mansfield&quot;</span>,<span class="st">&quot;PMTX&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb15-37"><a href="input-data.html#cb15-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-38"><a href="input-data.html#cb15-38" aria-hidden="true" tabindex="-1"></a><span class="co">#Corpus Christi Bay Texas</span></span>
<span id="cb15-39"><a href="input-data.html#cb15-39" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>site <span class="sc">%in%</span> <span class="st">&quot;corpus_christi&quot;</span>,<span class="st">&quot;CCTX&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb15-40"><a href="input-data.html#cb15-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-41"><a href="input-data.html#cb15-41" aria-hidden="true" tabindex="-1"></a><span class="co">#Tampa Bay FL</span></span>
<span id="cb15-42"><a href="input-data.html#cb15-42" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>lib_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;tam_01&quot;</span>,<span class="st">&quot;tam_04&quot;</span>,<span class="st">&quot;tam_05&quot;</span>,<span class="st">&quot;tam_06&quot;</span>),<span class="st">&quot;TBFL&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb15-43"><a href="input-data.html#cb15-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-44"><a href="input-data.html#cb15-44" aria-hidden="true" tabindex="-1"></a><span class="co">#Boca Ciega Bay Aquatic Preserve FL</span></span>
<span id="cb15-45"><a href="input-data.html#cb15-45" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>lib_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;tam_20&quot;</span>,<span class="st">&quot;tam_22&quot;</span>,<span class="st">&quot;tam_23&quot;</span>),<span class="st">&quot;BCFL&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb15-46"><a href="input-data.html#cb15-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-47"><a href="input-data.html#cb15-47" aria-hidden="true" tabindex="-1"></a><span class="co">#Madelaine Key, FL</span></span>
<span id="cb15-48"><a href="input-data.html#cb15-48" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>lib_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;tam_17&quot;</span>,<span class="st">&quot;tam_18&quot;</span>),<span class="st">&quot;MKFL&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb15-49"><a href="input-data.html#cb15-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-50"><a href="input-data.html#cb15-50" aria-hidden="true" tabindex="-1"></a><span class="co">#Caloosahatchee River, FL</span></span>
<span id="cb15-51"><a href="input-data.html#cb15-51" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>lib_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;car_01&quot;</span>,<span class="st">&quot;car_03&quot;</span>,<span class="st">&quot;car_05&quot;</span>,<span class="st">&quot;car_07&quot;</span>),<span class="st">&quot;CAFL&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb15-52"><a href="input-data.html#cb15-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-53"><a href="input-data.html#cb15-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-54"><a href="input-data.html#cb15-54" aria-hidden="true" tabindex="-1"></a><span class="co"># - 5 add lat and long to ULM obs that do not have this info</span></span>
<span id="cb15-55"><a href="input-data.html#cb15-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-56"><a href="input-data.html#cb15-56" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>latitude <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="st">&quot;ULTX&quot;</span>,<span class="fl">27.63083</span>,sample_info_full<span class="sc">$</span>latitude)</span>
<span id="cb15-57"><a href="input-data.html#cb15-57" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>longitude <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="st">&quot;ULTX&quot;</span>,<span class="sc">-</span><span class="fl">97.24018</span>,sample_info_full<span class="sc">$</span>longitude)</span></code></pre></div>
</div>
<div id="organize-oysters-into-metagroup-categories" class="section level3 hasAnchor" number="2.2.6">
<h3><span class="header-section-number">2.2.6</span> Organize oysters into “metagroup” categories:<a href="input-data.html#organize-oysters-into-metagroup-categories" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>South Texas (STX)</li>
<li>northern Gulf of Mexico (nGOM)</li>
<li>Florida (FL)</li>
<li>Selected Line - Auburn nGOM</li>
<li>Selected Line - Auburn FL</li>
<li>Selected Line - OBOY</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="input-data.html#cb16-1" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="st">&quot;selected lines&quot;</span></span>
<span id="cb16-2"><a href="input-data.html#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="input-data.html#cb16-3" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;AHFL&quot;</span>,<span class="st">&quot;BWBFL&quot;</span>,</span>
<span id="cb16-4"><a href="input-data.html#cb16-4" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;OBFL&quot;</span>,<span class="st">&quot;LRFL&quot;</span>,</span>
<span id="cb16-5"><a href="input-data.html#cb16-5" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;CRFL&quot;</span>,</span>
<span id="cb16-6"><a href="input-data.html#cb16-6" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;PBFL&quot;</span>,<span class="st">&quot;SKFL&quot;</span>,</span>
<span id="cb16-7"><a href="input-data.html#cb16-7" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;CAFL&quot;</span>,<span class="st">&quot;TBFL&quot;</span>,</span>
<span id="cb16-8"><a href="input-data.html#cb16-8" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;MKFL&quot;</span>,</span>
<span id="cb16-9"><a href="input-data.html#cb16-9" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;BCFL&quot;</span>),<span class="st">&quot;FL&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb16-10"><a href="input-data.html#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="input-data.html#cb16-11" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;ALAL&quot;</span>,<span class="st">&quot;CPAL&quot;</span>,</span>
<span id="cb16-12"><a href="input-data.html#cb16-12" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;EGTX&quot;</span>,<span class="st">&quot;EMTX&quot;</span>, </span>
<span id="cb16-13"><a href="input-data.html#cb16-13" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;LCLA&quot;</span>,<span class="st">&quot;LSTX&quot;</span>,</span>
<span id="cb16-14"><a href="input-data.html#cb16-14" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;PSMS&quot;</span>,<span class="st">&quot;SATX&quot;</span>,</span>
<span id="cb16-15"><a href="input-data.html#cb16-15" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;SLLA&quot;</span>,<span class="st">&quot;WGTX&quot;</span>,</span>
<span id="cb16-16"><a href="input-data.html#cb16-16" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;WMTX&quot;</span>),<span class="st">&quot;nGOM&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb16-17"><a href="input-data.html#cb16-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-18"><a href="input-data.html#cb16-18" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;ULTX&quot;</span>,<span class="st">&quot;PMTX&quot;</span>,</span>
<span id="cb16-19"><a href="input-data.html#cb16-19" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;CCTX&quot;</span>),<span class="st">&quot;STX&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb16-20"><a href="input-data.html#cb16-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-21"><a href="input-data.html#cb16-21" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Collection_data <span class="sc">%in%</span> <span class="st">&quot;Auburn selected line 1&quot;</span>,</span>
<span id="cb16-22"><a href="input-data.html#cb16-22" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Selected Line - Auburn nGOM&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb16-23"><a href="input-data.html#cb16-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-24"><a href="input-data.html#cb16-24" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Collection_data <span class="sc">%in%</span> <span class="st">&quot;Auburn selected line 2&quot;</span>,</span>
<span id="cb16-25"><a href="input-data.html#cb16-25" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Selected Line - Auburn FL&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb16-26"><a href="input-data.html#cb16-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-27"><a href="input-data.html#cb16-27" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Collection_data <span class="sc">%in%</span> <span class="st">&quot;O&#39;Boy selected line&quot;</span>,</span>
<span id="cb16-28"><a href="input-data.html#cb16-28" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Selected Line - OBOY&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="input-data.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#order the factor levels for geographic code based on longitude (move west to east)</span></span>
<span id="cb17-2"><a href="input-data.html#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="input-data.html#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb17-4"><a href="input-data.html#cb17-4" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="ot">&lt;-</span><span class="fu">arrange</span>(sample_info_full,longitude)</span>
<span id="cb17-5"><a href="input-data.html#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="input-data.html#cb17-6" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code<span class="ot">&lt;-</span><span class="fu">factor</span>(sample_info_full<span class="sc">$</span>Geographic_code, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;PMTX&quot;</span>,<span class="st">&quot;ULTX&quot;</span>,<span class="st">&quot;CCTX&quot;</span>,<span class="st">&quot;SATX&quot;</span>,<span class="st">&quot;WMTX&quot;</span>,<span class="st">&quot;EMTX&quot;</span>,<span class="st">&quot;WGTX&quot;</span>,<span class="st">&quot;EGTX&quot;</span>,<span class="st">&quot;LSTX&quot;</span>,<span class="st">&quot;LCLA&quot;</span>,<span class="st">&quot;SLLA&quot;</span>,<span class="st">&quot;PSMS&quot;</span>,<span class="st">&quot;CPAL&quot;</span>,<span class="st">&quot;ALAL&quot;</span>,<span class="st">&quot;PBFL&quot;</span>,<span class="st">&quot;AHFL&quot;</span>,<span class="st">&quot;OBFL&quot;</span>,<span class="st">&quot;LRFL&quot;</span>,<span class="st">&quot;SKFL&quot;</span>,<span class="st">&quot;CRFL&quot;</span>,<span class="st">&quot;MKFL&quot;</span>,<span class="st">&quot;BCFL&quot;</span>,<span class="st">&quot;TBFL&quot;</span>,<span class="st">&quot;CAFL&quot;</span>,<span class="st">&quot;BWBFL&quot;</span>))</span></code></pre></div>
</div>
<div id="map-sites-where-samples-were-collected" class="section level3 hasAnchor" number="2.2.7">
<h3><span class="header-section-number">2.2.7</span> Map sites where samples were collected<a href="input-data.html#map-sites-where-samples-were-collected" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Site codes and site names:</p>
<ul>
<li>PMTX ~ Port Mansfield</li>
<li>ULTX ~ Upper Laguna Madre</li>
<li>CCTX ~ Corpus Christi Bay</li>
<li>SATX ~ San Antonio Bay</li>
<li>WMTX ~ West Matagorda Bay</li>
<li>EMTX ~ East Matagorda Bay</li>
<li>WGTX ~ West Galveston Bay</li>
<li>EGTX ~ East Galveston Bay</li>
<li>LSTX ~ Sabine Pass</li>
<li>LCLA ~ Calcasieu Lake</li>
<li>SLLA ~ Caillou Lake</li>
<li>PSMS ~ Pascagoula River</li>
<li>CPAL ~ Cedar Point</li>
<li>ALAL ~ Alligator Lake</li>
<li>PBFL ~ Pensacola Bay</li>
<li>AHFL ~ Alligator Harbor</li>
<li>OBFL ~ Oyster Bay</li>
<li>LRFL ~ Suwannee River</li>
<li>SKFL ~ Seahorse Key</li>
<li>CRFL ~ Corrigan Reef</li>
<li>MKFL ~ Madelaine Key</li>
<li>BCFL ~ Boca Ciega Bay Aquatic Preserve</li>
<li>TBFL ~ Tampa Bay</li>
<li>CAFL ~ Caloosahatchee River</li>
<li>BWBFL ~ Backwater Bay</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="input-data.html#cb18-1" aria-hidden="true" tabindex="-1"></a>MainStates <span class="ot">&lt;-</span> <span class="fu">map_data</span>(<span class="st">&quot;state&quot;</span>)</span>
<span id="cb18-2"><a href="input-data.html#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="input-data.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb18-4"><a href="input-data.html#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;lightgray&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb18-5"><a href="input-data.html#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb18-6"><a href="input-data.html#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb18-7"><a href="input-data.html#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb18-8"><a href="input-data.html#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>)) <span class="sc">+</span> </span>
<span id="cb18-9"><a href="input-data.html#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> sample_info_full <span class="sc">%&gt;%</span>   </span>
<span id="cb18-10"><a href="input-data.html#cb18-10" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(<span class="sc">!</span>Collection_data <span class="sc">==</span> <span class="st">&quot;Auburn selected line 2&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-11"><a href="input-data.html#cb18-11" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(<span class="sc">!</span>Collection_data <span class="sc">==</span> <span class="st">&quot;O&#39;Boy selected line&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-12"><a href="input-data.html#cb18-12" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(<span class="sc">!</span>Collection_data <span class="sc">==</span> <span class="st">&quot;Auburn selected line 1&quot;</span>), <span class="co">#remove the selected lines from mapping</span></span>
<span id="cb18-13"><a href="input-data.html#cb18-13" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> latitude, <span class="at">x =</span> longitude, <span class="at">group=</span>Geographic_code,<span class="at">colour=</span>Geographic_code,<span class="at">shape =</span> metagroup),</span>
<span id="cb18-14"><a href="input-data.html#cb18-14" aria-hidden="true" tabindex="-1"></a>             <span class="at">size=</span><span class="dv">5</span>,<span class="at">stroke=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>Map representing the relative number of samples collected in each location</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="input-data.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb19-2"><a href="input-data.html#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;lightgray&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb19-3"><a href="input-data.html#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb19-4"><a href="input-data.html#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb19-5"><a href="input-data.html#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb19-6"><a href="input-data.html#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>)) <span class="sc">+</span> </span>
<span id="cb19-7"><a href="input-data.html#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="input-data.html#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">latitude2 =</span> <span class="fu">round</span>(latitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="input-data.html#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">longitude2 =</span> <span class="fu">round</span>(longitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-10"><a href="input-data.html#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Geographic_code,metagroup,longitude2,latitude2) <span class="sc">%&gt;%</span></span>
<span id="cb19-11"><a href="input-data.html#cb19-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">&#39;longitude2&#39;</span>, <span class="st">&#39;latitude2&#39;</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb19-12"><a href="input-data.html#cb19-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>metagroup <span class="sc">==</span> <span class="st">&quot;Selected Line - Auburn FL&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-13"><a href="input-data.html#cb19-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>metagroup <span class="sc">==</span> <span class="st">&quot;Selected Line - Auburn nGOM&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-14"><a href="input-data.html#cb19-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>metagroup <span class="sc">==</span> <span class="st">&quot;Selected Line - OBOY&quot;</span>),</span>
<span id="cb19-15"><a href="input-data.html#cb19-15" aria-hidden="true" tabindex="-1"></a>    <span class="co">#counts for proportional representation</span></span>
<span id="cb19-16"><a href="input-data.html#cb19-16" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> latitude2, <span class="at">x =</span> longitude2, <span class="at">group=</span>Geographic_code,<span class="at">colour=</span>Geographic_code,<span class="at">shape =</span> metagroup,<span class="at">size =</span> n),</span>
<span id="cb19-17"><a href="input-data.html#cb19-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">stroke=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-16-1.png" width="768" /></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="input-data.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#produce blank map to add points to later</span></span>
<span id="cb20-2"><a href="input-data.html#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="input-data.html#cb20-3" aria-hidden="true" tabindex="-1"></a>blank_map <span class="ot">&lt;-</span> <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb20-4"><a href="input-data.html#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;grey90&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb20-5"><a href="input-data.html#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb20-6"><a href="input-data.html#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb20-7"><a href="input-data.html#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb20-8"><a href="input-data.html#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>))</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="input-data.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#add site codes to individual data</span></span>
<span id="cb21-2"><a href="input-data.html#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="input-data.html#cb21-3" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="ot">&lt;-</span> sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb21-4"><a href="input-data.html#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Geographic_location =</span> <span class="fu">case_when</span>(</span>
<span id="cb21-5"><a href="input-data.html#cb21-5" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;PMTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Port Mansfield&quot;</span>,</span>
<span id="cb21-6"><a href="input-data.html#cb21-6" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;ULTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Upper Laguna Madre&quot;</span>,</span>
<span id="cb21-7"><a href="input-data.html#cb21-7" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;CCTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Corpus Christi Bay&quot;</span>,</span>
<span id="cb21-8"><a href="input-data.html#cb21-8" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;SATX&quot;</span> <span class="sc">~</span> <span class="st">&quot;San Antonio Bay&quot;</span>,</span>
<span id="cb21-9"><a href="input-data.html#cb21-9" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;WMTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;West Matagorda Bay&quot;</span>,</span>
<span id="cb21-10"><a href="input-data.html#cb21-10" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;EMTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;East Matagorda Bay&quot;</span>,</span>
<span id="cb21-11"><a href="input-data.html#cb21-11" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;WGTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;West Galveston Bay&quot;</span>,</span>
<span id="cb21-12"><a href="input-data.html#cb21-12" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;EGTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;East Galveston Bay&quot;</span>,</span>
<span id="cb21-13"><a href="input-data.html#cb21-13" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;LSTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Sabine Pass&quot;</span>,</span>
<span id="cb21-14"><a href="input-data.html#cb21-14" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;LCLA&quot;</span> <span class="sc">~</span> <span class="st">&quot;Calcasieu Lake&quot;</span>,</span>
<span id="cb21-15"><a href="input-data.html#cb21-15" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;SLLA&quot;</span> <span class="sc">~</span> <span class="st">&quot;Caillou Lake&quot;</span>,</span>
<span id="cb21-16"><a href="input-data.html#cb21-16" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;PSMS&quot;</span> <span class="sc">~</span> <span class="st">&quot;Pascagoula River&quot;</span>,</span>
<span id="cb21-17"><a href="input-data.html#cb21-17" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;CPAL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Cedar Point&quot;</span>,</span>
<span id="cb21-18"><a href="input-data.html#cb21-18" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;ALAL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Alligator Lake&quot;</span>,</span>
<span id="cb21-19"><a href="input-data.html#cb21-19" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;PBFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Pensacola Bay&quot;</span>,</span>
<span id="cb21-20"><a href="input-data.html#cb21-20" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;AHFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Alligator Harbor&quot;</span>,</span>
<span id="cb21-21"><a href="input-data.html#cb21-21" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;OBFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Oyster Bay&quot;</span>,</span>
<span id="cb21-22"><a href="input-data.html#cb21-22" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;LRFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Suwannee River&quot;</span>,</span>
<span id="cb21-23"><a href="input-data.html#cb21-23" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;SKFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Seahorse Key&quot;</span>,</span>
<span id="cb21-24"><a href="input-data.html#cb21-24" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;CRFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Corrigan Reef&quot;</span>,</span>
<span id="cb21-25"><a href="input-data.html#cb21-25" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;MKFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Madelaine Key&quot;</span>,</span>
<span id="cb21-26"><a href="input-data.html#cb21-26" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;BCFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Boca Ciega Bay Aquatic Preserve&quot;</span>,</span>
<span id="cb21-27"><a href="input-data.html#cb21-27" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;TBFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Tampa Bay&quot;</span>,</span>
<span id="cb21-28"><a href="input-data.html#cb21-28" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;CAFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Caloosahatchee River&quot;</span>,</span>
<span id="cb21-29"><a href="input-data.html#cb21-29" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;BWBFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Backwater Bay&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-30"><a href="input-data.html#cb21-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Geographic_state =</span> <span class="fu">case_when</span>(</span>
<span id="cb21-31"><a href="input-data.html#cb21-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;TX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Texas&quot;</span>,</span>
<span id="cb21-32"><a href="input-data.html#cb21-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;LA&quot;</span> <span class="sc">~</span> <span class="st">&quot;Louisiana&quot;</span>,</span>
<span id="cb21-33"><a href="input-data.html#cb21-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;MS&quot;</span> <span class="sc">~</span> <span class="st">&quot;Mississippi&quot;</span>,</span>
<span id="cb21-34"><a href="input-data.html#cb21-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;AL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Alabama&quot;</span>,</span>
<span id="cb21-35"><a href="input-data.html#cb21-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;FL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Florida&quot;</span>))</span></code></pre></div>
<p>Table of genotyped samples categorized by location of collection</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="input-data.html#cb22-1" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb22-2"><a href="input-data.html#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Geographic_code,Geographic_location,Geographic_state,metagroup) <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="input-data.html#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   Geographic_code Geographic_location Geographic_state metagroup  n
## 1            PMTX      Port Mansfield            Texas       STX 19
## 2            ULTX  Upper Laguna Madre            Texas       STX 42
## 3            CCTX  Corpus Christi Bay            Texas       STX 14
## 4            SATX     San Antonio Bay            Texas      nGOM 11
## 5            WMTX  West Matagorda Bay            Texas      nGOM 16
## 6            EMTX  East Matagorda Bay            Texas      nGOM 18</code></pre>
<p>Table of genotyped samples categorized by the region of collection - south Texas (STX), northern Gulf of Mexico (nGOM), Florida (FL), and individuals from selected lines (selected lines)</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="input-data.html#cb24-1" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="input-data.html#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(metagroup)</span></code></pre></div>
<pre><code>##                     metagroup   n
## 1                          FL 101
## 2                         STX  75
## 3   Selected Line - Auburn FL  22
## 4 Selected Line - Auburn nGOM  23
## 5        Selected Line - OBOY  22
## 6                        nGOM 204</code></pre>
</div>
</div>
<div id="snp-information" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> SNP Information<a href="input-data.html#snp-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Total: 566,262</p>
<p>SNPs considered ‘recommended’ for downstream analysis by ThermoFisher if fit in one of the following categories:</p>
<ul>
<li><p>PolyHighResolution: SNPs with well-separated genotype clusters and two or more alleles in the genotype calls</p></li>
<li><p>NoMinorHom: SNPs with well-separated genotype clusters; one cluster is homozygous and one is heterozygous for biallelic SNPs, only one homozygous cluster and one or more heterozygous clusters appear for multiallelic SNPs</p></li>
<li><p>MonoHighResolution: SNPs with one well-formed genotype cluster; must be homozygous</p></li>
<li><p>Recommended “other” cluster (GG): 207,534</p></li>
<li><p>Recommended “south” cluster (ST): 152,260</p></li>
<li><p>Recommended in both clusters: 105,363</p></li>
</ul>
<p>Only SNPs recommended for both GG and ST analyzed here (the 105,363)</p>
<div id="general-information-on-snps" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> General Information on SNPs<a href="input-data.html#general-information-on-snps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="input-data.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in general info</span></span>
<span id="cb26-2"><a href="input-data.html#cb26-2" aria-hidden="true" tabindex="-1"></a>snp_info <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/snp_info_cv_ECarray_header_edited.txt&quot;</span>, <span class="at">fill =</span> <span class="cn">TRUE</span> , <span class="at">header =</span> <span class="cn">TRUE</span> )</span>
<span id="cb26-3"><a href="input-data.html#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="input-data.html#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co">#rename general info column &quot;ID&quot; to &quot;probeset_id&quot; to match with genotyped files</span></span>
<span id="cb26-5"><a href="input-data.html#cb26-5" aria-hidden="true" tabindex="-1"></a>snp_info <span class="ot">&lt;-</span> snp_info <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">probeset_id =</span> ID_Name)</span>
<span id="cb26-6"><a href="input-data.html#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="input-data.html#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(snp_info)</span></code></pre></div>
<pre><code>## # A tibble: 582,726 × 31
##    ReporterKey Affx       probeset_id Sequence SNPType ProbeCount Tile  Replicates DesignNumber
##    &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;       
##  1 531501570   Affx-1248… AX-5782382… CCTGCCC… Y       2          1     2          1           
##  2 531501571   Affx-1248… AX-5782382… ATGATTG… K       2          1     2          1           
##  3 531501572   Affx-1248… AX-5782382… AGAGGGA… R       2          1     2          1           
##  4 531422636   Affx-1248… AX-5781595… CCTGATA… Y       2          1     2          1           
##  5 531421883   Affx-1248… AX-5781587… GGCGGAT… K       2          1     2          1           
##  6 531422066   Affx-1248… AX-5781589… AGCCGCG… M       2          1     2          1           
##  7 531501573   Affx-1248… AX-5782382… TGGTGCA… R       2          1     2          1           
##  8 531421986   Affx-1248… AX-5781588… ACGTT[C… Y       2          1     2          1           
##  9 531501574   Affx-1248… AX-5782382… ATTGAAG… Y       2          1     2          1           
## 10 531422050   Affx-1248… AX-5781589… GTGGCTT… K       2          1     2          1           
## # ℹ 582,716 more rows
## # ℹ 22 more variables: Rank &lt;chr&gt;, Strand &lt;chr&gt;, SetBack &lt;chr&gt;, taxonomy_key &lt;chr&gt;,
## #   organism &lt;chr&gt;, cust_id &lt;chr&gt;, cust_chr &lt;chr&gt;, cust_pos &lt;chr&gt;, cust_ref &lt;chr&gt;,
## #   cust_alt &lt;chr&gt;, cust_source &lt;chr&gt;, cust_gene_id &lt;chr&gt;, cust_gene_name &lt;chr&gt;,
## #   cust_snptype &lt;chr&gt;, cust_other_id &lt;chr&gt;, cust_validation &lt;chr&gt;, cust_cn_region &lt;chr&gt;,
## #   tiling_order &lt;chr&gt;, importance &lt;chr&gt;, pconvert &lt;chr&gt;, recommendation &lt;chr&gt;, Error &lt;chr&gt;</code></pre>
</div>
<div id="recommended-snps" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Recommended SNPs<a href="input-data.html#recommended-snps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>SNPs recommended by ThermoFisher for both GG and ST</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="input-data.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co">#list of recommended SNPs for each sample set</span></span>
<span id="cb28-2"><a href="input-data.html#cb28-2" aria-hidden="true" tabindex="-1"></a>other_rec <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_Recommended.ps&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-3"><a href="input-data.html#cb28-3" aria-hidden="true" tabindex="-1"></a>south_rec <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_Recommended.ps&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-4"><a href="input-data.html#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="input-data.html#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co">#merge recommended SNPs for each sample set</span></span>
<span id="cb28-6"><a href="input-data.html#cb28-6" aria-hidden="true" tabindex="-1"></a>combined_rec <span class="ot">&lt;-</span> <span class="fu">merge</span>(other_rec, south_rec)</span>
<span id="cb28-7"><a href="input-data.html#cb28-7" aria-hidden="true" tabindex="-1"></a>combined_rec <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##    probeset_id
## 1 AX-168593894
## 2 AX-168612286
## 3 AX-562727784
## 4 AX-562727806
## 5 AX-562727827
## 6 AX-562728138</code></pre>
</div>
<div id="quality-control-info" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Quality control info<a href="input-data.html#quality-control-info" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Quality control data on SNPs from ThermoFisher</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="input-data.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in QC data for both data sets</span></span>
<span id="cb30-2"><a href="input-data.html#cb30-2" aria-hidden="true" tabindex="-1"></a>other_qc_snp <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_Ps.performance.txt&quot;</span>, <span class="at">fill =</span> <span class="cn">TRUE</span>,<span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb30-3"><a href="input-data.html#cb30-3" aria-hidden="true" tabindex="-1"></a>south_qc_snp <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_Ps.performance.txt&quot;</span>, <span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb30-4"><a href="input-data.html#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="input-data.html#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="co">#add column of data_set - &quot;other&quot; or &quot;south_texas&quot;</span></span>
<span id="cb30-6"><a href="input-data.html#cb30-6" aria-hidden="true" tabindex="-1"></a>other_qc_snp<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="st">&quot;other&quot;</span></span>
<span id="cb30-7"><a href="input-data.html#cb30-7" aria-hidden="true" tabindex="-1"></a>south_qc_snp<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="st">&quot;south_texas&quot;</span></span>
<span id="cb30-8"><a href="input-data.html#cb30-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-9"><a href="input-data.html#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="co">#rbind qc data for both data sets</span></span>
<span id="cb30-10"><a href="input-data.html#cb30-10" aria-hidden="true" tabindex="-1"></a>qc_snp <span class="ot">&lt;-</span> <span class="fu">rbind</span>(other_qc_snp,south_qc_snp)</span>
<span id="cb30-11"><a href="input-data.html#cb30-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-12"><a href="input-data.html#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="co">#select only SNPs recommended for both clusters </span></span>
<span id="cb30-13"><a href="input-data.html#cb30-13" aria-hidden="true" tabindex="-1"></a>qc_snp_rec<span class="ot">&lt;-</span><span class="fu">merge</span>(qc_snp,combined_rec)</span>
<span id="cb30-14"><a href="input-data.html#cb30-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-15"><a href="input-data.html#cb30-15" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(qc_snp_rec)</span></code></pre></div>
<pre><code>## # A tibble: 210,726 × 25
##    probeset_id affy_snp_id    CR   FLD HomFLD  HetSO HomRO nMinorAllele Nclus  n_AA  n_AB  n_BB
##    &lt;chr&gt;       &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;        &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt;
##  1 AX-1685938… Affx-12468…  99.7  7.46   NA    0.952  3.53            2     2     0     2   369
##  2 AX-1685938… Affx-12468… 100    7.22   15.4  0.714  2.63           70     3    28    24    23
##  3 AX-1686122… Affx-12459…  98.7  5.97   12.2  0.518  2.64           86     3    12    62   293
##  4 AX-1686122… Affx-12459…  97.3  5.19   11.3  0.205  1.71            8     3     2     4    67
##  5 AX-5627277… Affx-12468…  98.7  4.94   NA   -0.166  2.76            3     2     0     3    71
##  6 AX-5627277… Affx-12468…  99.5  6.52   15.1  0.508  2.90           26     3     2    22   346
##  7 AX-5627278… Affx-12365…  97.3  8.20   17.7  0.247  2.74           14     3    61    10     2
##  8 AX-5627278… Affx-12365…  99.7  9.11   19.1  0.756  3.11          368     3   104   166   101
##  9 AX-5627278… Affx-12365… 100    4.75   NA    1.30   2.10            2     3    67     2     0
## 10 AX-5627278… Affx-12365… 100    5.63   15.6  0.540  2.36           46     3   330    38     4
## # ℹ 210,716 more rows
## # ℹ 13 more variables: n_NC &lt;int&gt;, hemizygous &lt;int&gt;, specialSNP_chr &lt;chr&gt;,
## #   gender_metrics &lt;chr&gt;, ConversionType &lt;chr&gt;, BestProbeset &lt;int&gt;, BestandRecommended &lt;int&gt;,
## #   HomHet &lt;int&gt;, MMD &lt;dbl&gt;, MinorAlleleFrequency &lt;dbl&gt;, H.W.p.Value &lt;dbl&gt;,
## #   H.W.chisquared.statistic &lt;dbl&gt;, cluster &lt;chr&gt;</code></pre>
</div>
<div id="merge-files-1" class="section level3 hasAnchor" number="2.3.4">
<h3><span class="header-section-number">2.3.4</span> Merge Files<a href="input-data.html#merge-files-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Consolidate information on SNPs into one file “snp_info_full.txt”</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="input-data.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co">#merge recommended and general info</span></span>
<span id="cb32-2"><a href="input-data.html#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="co">#note the final file has two rows for each SNP (based on whether it is in the &#39;other&#39; or &#39;south_texas&#39; cluster)</span></span>
<span id="cb32-3"><a href="input-data.html#cb32-3" aria-hidden="true" tabindex="-1"></a>snp_info_full <span class="ot">&lt;-</span> <span class="fu">merge</span>(combined_rec, snp_info)</span>
<span id="cb32-4"><a href="input-data.html#cb32-4" aria-hidden="true" tabindex="-1"></a>snp_info_full <span class="ot">&lt;-</span> <span class="fu">merge</span>(snp_info_full, qc_snp_rec)</span>
<span id="cb32-5"><a href="input-data.html#cb32-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-6"><a href="input-data.html#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="co">#remove &quot;-&quot; in probeset id</span></span>
<span id="cb32-7"><a href="input-data.html#cb32-7" aria-hidden="true" tabindex="-1"></a>snp_info_full<span class="sc">$</span>probeset_id <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>, snp_info_full<span class="sc">$</span>probeset_id)</span>
<span id="cb32-8"><a href="input-data.html#cb32-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-9"><a href="input-data.html#cb32-9" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(snp_info_full)</span></code></pre></div>
<pre><code>## # A tibble: 210,726 × 55
##    probeset_id ReporterKey Affx       Sequence SNPType ProbeCount Tile  Replicates DesignNumber
##    &lt;chr&gt;       &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;       
##  1 AX168593894 530224709   Affx-1246… TTGTCTT… Y       2          1     2          1           
##  2 AX168593894 530224709   Affx-1246… TTGTCTT… Y       2          1     2          1           
##  3 AX168612286 526588721   Affx-1245… GTTTAAT… Y       2          1     2          1           
##  4 AX168612286 526588721   Affx-1245… GTTTAAT… Y       2          1     2          1           
##  5 AX562727784 527308034   Affx-1246… GAGGAAA… Y       2          1     2          1           
##  6 AX562727784 527308034   Affx-1246… GAGGAAA… Y       2          1     2          1           
##  7 AX562727806 513178708   Affx-1236… GCTAAGT… R       2          1     2          1           
##  8 AX562727806 513178708   Affx-1236… GCTAAGT… R       2          1     2          1           
##  9 AX562727827 513178729   Affx-1236… GGTCATG… Y       2          1     2          1           
## 10 AX562727827 513178729   Affx-1236… GGTCATG… Y       2          1     2          1           
## # ℹ 210,716 more rows
## # ℹ 46 more variables: Rank &lt;chr&gt;, Strand &lt;chr&gt;, SetBack &lt;chr&gt;, taxonomy_key &lt;chr&gt;,
## #   organism &lt;chr&gt;, cust_id &lt;chr&gt;, cust_chr &lt;chr&gt;, cust_pos &lt;chr&gt;, cust_ref &lt;chr&gt;,
## #   cust_alt &lt;chr&gt;, cust_source &lt;chr&gt;, cust_gene_id &lt;chr&gt;, cust_gene_name &lt;chr&gt;,
## #   cust_snptype &lt;chr&gt;, cust_other_id &lt;chr&gt;, cust_validation &lt;chr&gt;, cust_cn_region &lt;chr&gt;,
## #   tiling_order &lt;chr&gt;, importance &lt;chr&gt;, pconvert &lt;chr&gt;, recommendation &lt;chr&gt;, Error &lt;chr&gt;,
## #   affy_snp_id &lt;chr&gt;, CR &lt;dbl&gt;, FLD &lt;dbl&gt;, HomFLD &lt;dbl&gt;, HetSO &lt;dbl&gt;, HomRO &lt;dbl&gt;, …</code></pre>
</div>
<div id="formatting-for-hierfstat-and-adegenet" class="section level3 hasAnchor" number="2.3.5">
<h3><span class="header-section-number">2.3.5</span> Formatting for hierfstat and adegenet<a href="input-data.html#formatting-for-hierfstat-and-adegenet" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>SNP data formatted so to be analyzed with the “hierfstat” and “adegenet” package</p>
<p>Formatted to loci in 1 column</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="input-data.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in .cs files (takes a minute)</span></span>
<span id="cb34-2"><a href="input-data.html#cb34-2" aria-hidden="true" tabindex="-1"></a>Ncalls_other <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_AxiomGT1.calls.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb34-3"><a href="input-data.html#cb34-3" aria-hidden="true" tabindex="-1"></a>Ncalls_south <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_AxiomGT1.calls.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb34-4"><a href="input-data.html#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="input-data.html#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="co">#filter out only the rec SNPs from the calls data</span></span>
<span id="cb34-6"><a href="input-data.html#cb34-6" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> <span class="fu">subset</span>(Ncalls_other, probeset_id <span class="sc">%in%</span> combined_rec<span class="sc">$</span>probeset_id)</span>
<span id="cb34-7"><a href="input-data.html#cb34-7" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> <span class="fu">subset</span>(Ncalls_south, probeset_id <span class="sc">%in%</span> combined_rec<span class="sc">$</span>probeset_id)</span>
<span id="cb34-8"><a href="input-data.html#cb34-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-9"><a href="input-data.html#cb34-9" aria-hidden="true" tabindex="-1"></a><span class="co">#transpose so individuals are rows and SNPs are columns</span></span>
<span id="cb34-10"><a href="input-data.html#cb34-10" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(other_filtered))</span>
<span id="cb34-11"><a href="input-data.html#cb34-11" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(south_filtered))</span>
<span id="cb34-12"><a href="input-data.html#cb34-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-13"><a href="input-data.html#cb34-13" aria-hidden="true" tabindex="-1"></a><span class="co">#make the first row of the dataframe the column headers</span></span>
<span id="cb34-14"><a href="input-data.html#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(other_filtered) <span class="ot">&lt;-</span> other_filtered[<span class="dv">1</span>,]</span>
<span id="cb34-15"><a href="input-data.html#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(south_filtered) <span class="ot">&lt;-</span> south_filtered[<span class="dv">1</span>,]</span>
<span id="cb34-16"><a href="input-data.html#cb34-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-17"><a href="input-data.html#cb34-17" aria-hidden="true" tabindex="-1"></a><span class="co">#delete the first row of the data set (only used as column headers)</span></span>
<span id="cb34-18"><a href="input-data.html#cb34-18" aria-hidden="true" tabindex="-1"></a>other_filtered<span class="ot">&lt;-</span> other_filtered[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb34-19"><a href="input-data.html#cb34-19" aria-hidden="true" tabindex="-1"></a>south_filtered<span class="ot">&lt;-</span> south_filtered[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb34-20"><a href="input-data.html#cb34-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-21"><a href="input-data.html#cb34-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Make vector of other/south individuals for later referencing</span></span>
<span id="cb34-22"><a href="input-data.html#cb34-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-23"><a href="input-data.html#cb34-23" aria-hidden="true" tabindex="-1"></a>other_inds <span class="ot">&lt;-</span> <span class="fu">rownames</span>(other_filtered)</span>
<span id="cb34-24"><a href="input-data.html#cb34-24" aria-hidden="true" tabindex="-1"></a>  other_inds <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, other_inds)</span>
<span id="cb34-25"><a href="input-data.html#cb34-25" aria-hidden="true" tabindex="-1"></a>  other_inds <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, other_inds)</span>
<span id="cb34-26"><a href="input-data.html#cb34-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-27"><a href="input-data.html#cb34-27" aria-hidden="true" tabindex="-1"></a>south_inds <span class="ot">&lt;-</span> <span class="fu">rownames</span>(south_filtered)</span>
<span id="cb34-28"><a href="input-data.html#cb34-28" aria-hidden="true" tabindex="-1"></a>  south_inds <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, south_inds)</span>
<span id="cb34-29"><a href="input-data.html#cb34-29" aria-hidden="true" tabindex="-1"></a>  south_inds <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, south_inds)</span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="input-data.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="do">##set to eval=FALSE because takes so long</span></span>
<span id="cb35-2"><a href="input-data.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="do">##Change allele codes. </span></span>
<span id="cb35-3"><a href="input-data.html#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="do">###Warning: this takes a very long time (almost 1 hour). </span></span>
<span id="cb35-4"><a href="input-data.html#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="input-data.html#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co">#For some reason there is whitespace around some of the values which causes problems when converting the values. Strip all leading/trailing white space </span></span>
<span id="cb35-6"><a href="input-data.html#cb35-6" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> other_filtered <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.character), str_trim))</span>
<span id="cb35-7"><a href="input-data.html#cb35-7" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> south_filtered <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.character), str_trim))</span>
<span id="cb35-8"><a href="input-data.html#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="input-data.html#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="co">#if value in dataframe is -1, change to NA</span></span>
<span id="cb35-10"><a href="input-data.html#cb35-10" aria-hidden="true" tabindex="-1"></a><span class="do">## -1 specifies a &#39;No Call&#39;</span></span>
<span id="cb35-11"><a href="input-data.html#cb35-11" aria-hidden="true" tabindex="-1"></a>other_filtered[other_filtered <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span>   </span>
<span id="cb35-12"><a href="input-data.html#cb35-12" aria-hidden="true" tabindex="-1"></a>south_filtered[south_filtered <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span>   </span>
<span id="cb35-13"><a href="input-data.html#cb35-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-14"><a href="input-data.html#cb35-14" aria-hidden="true" tabindex="-1"></a><span class="co"># If code is 0 (AA) change to 11</span></span>
<span id="cb35-15"><a href="input-data.html#cb35-15" aria-hidden="true" tabindex="-1"></a>other_filtered[other_filtered <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="st">&quot;11&quot;</span></span>
<span id="cb35-16"><a href="input-data.html#cb35-16" aria-hidden="true" tabindex="-1"></a>south_filtered[south_filtered <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="st">&quot;11&quot;</span></span>
<span id="cb35-17"><a href="input-data.html#cb35-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-18"><a href="input-data.html#cb35-18" aria-hidden="true" tabindex="-1"></a><span class="co">#If code is 1 (AB) change to 12</span></span>
<span id="cb35-19"><a href="input-data.html#cb35-19" aria-hidden="true" tabindex="-1"></a>other_filtered[other_filtered <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;12&quot;</span></span>
<span id="cb35-20"><a href="input-data.html#cb35-20" aria-hidden="true" tabindex="-1"></a>south_filtered[south_filtered <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;12&quot;</span></span>
<span id="cb35-21"><a href="input-data.html#cb35-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-22"><a href="input-data.html#cb35-22" aria-hidden="true" tabindex="-1"></a><span class="co"># If code is 2 (BB) change to 22</span></span>
<span id="cb35-23"><a href="input-data.html#cb35-23" aria-hidden="true" tabindex="-1"></a>other_filtered[other_filtered <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;22&quot;</span></span>
<span id="cb35-24"><a href="input-data.html#cb35-24" aria-hidden="true" tabindex="-1"></a>south_filtered[south_filtered <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;22&quot;</span></span>
<span id="cb35-25"><a href="input-data.html#cb35-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-26"><a href="input-data.html#cb35-26" aria-hidden="true" tabindex="-1"></a><span class="co">#save formatted data files for other and south</span></span>
<span id="cb35-27"><a href="input-data.html#cb35-27" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(other_filtered, <span class="st">&quot;data/derived/cluster_other_formatted.txt&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb35-28"><a href="input-data.html#cb35-28" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(south_filtered, <span class="st">&quot;data/derived/cluster_south_formatted.txt&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="input-data.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in formatted data files for other and south</span></span>
<span id="cb36-2"><a href="input-data.html#cb36-2" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> <span class="fu">read.table</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;derived&quot;</span>,<span class="st">&quot;cluster_other_formatted.txt&quot;</span>))</span>
<span id="cb36-3"><a href="input-data.html#cb36-3" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> <span class="fu">read.table</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;derived&quot;</span>,<span class="st">&quot;cluster_south_formatted.txt&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="input-data.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Consolidate to one file - &quot;combined_filtered.txt&quot;</span></span>
<span id="cb37-2"><a href="input-data.html#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="input-data.html#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co"># sort data frames to make sure columns are in the same order before combining</span></span>
<span id="cb37-4"><a href="input-data.html#cb37-4" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> other_filtered[,<span class="fu">order</span>(<span class="fu">names</span>(other_filtered))]</span>
<span id="cb37-5"><a href="input-data.html#cb37-5" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> south_filtered[,<span class="fu">order</span>(<span class="fu">names</span>(south_filtered))]</span>
<span id="cb37-6"><a href="input-data.html#cb37-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-7"><a href="input-data.html#cb37-7" aria-hidden="true" tabindex="-1"></a><span class="co"># double check because I&#39;m paranoid </span></span>
<span id="cb37-8"><a href="input-data.html#cb37-8" aria-hidden="true" tabindex="-1"></a>cols1 <span class="ot">&lt;-</span> <span class="fu">names</span>(other_filtered)</span>
<span id="cb37-9"><a href="input-data.html#cb37-9" aria-hidden="true" tabindex="-1"></a>cols2 <span class="ot">&lt;-</span> <span class="fu">names</span>(south_filtered)</span>
<span id="cb37-10"><a href="input-data.html#cb37-10" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(cols1, cols2)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="input-data.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co">#combine datasets</span></span>
<span id="cb39-2"><a href="input-data.html#cb39-2" aria-hidden="true" tabindex="-1"></a>combined_filtered <span class="ot">&lt;-</span> <span class="fu">rbind</span>(other_filtered, south_filtered)</span>
<span id="cb39-3"><a href="input-data.html#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="input-data.html#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(combined_filtered))</span>
<span id="cb39-5"><a href="input-data.html#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(combined_filtered))</span>
<span id="cb39-6"><a href="input-data.html#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(combined_filtered))</span>
<span id="cb39-7"><a href="input-data.html#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(combined_filtered))</span>
<span id="cb39-8"><a href="input-data.html#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(combined_filtered))</span></code></pre></div>
</div>
</div>
<div id="save-files" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Save files<a href="input-data.html#save-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="input-data.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">#save sample_info_Full</span></span>
<span id="cb40-2"><a href="input-data.html#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(sample_info_full, <span class="at">file =</span> here <span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;derived&quot;</span>,<span class="st">&quot;sample_info_full&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="input-data.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co">#save_snp_info_full</span></span>
<span id="cb41-2"><a href="input-data.html#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_info_full, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;derived&quot;</span>,<span class="st">&quot;snp_info_full&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="input-data.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co">#save combined_filtered data file</span></span>
<span id="cb42-2"><a href="input-data.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(combined_filtered, <span class="st">&quot;data/derived/combined_filtered.txt&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="genetic-groups.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="book_files/gitbook-2.6.7/js/app.min.js"></script>
<script src="book_files/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="book_files/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"toolbar": {
"position": "static"
}
});
});
</script>

</body>

</html>
