<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1 Input Data | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="1 Input Data | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1 Input Data | _main.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="genetic-groups.html"/>
<script src="book_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="book_files/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="book_files/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="book_files/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="book_files/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="book_files/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>1</b> Input Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="input-data.html"><a href="input-data.html#objective-organize-individual-and-snp-data"><i class="fa fa-check"></i><b>1.1</b> Objective: Organize individual and SNP data</a></li>
<li class="chapter" data-level="1.2" data-path="input-data.html"><a href="input-data.html#individual-sample-information"><i class="fa fa-check"></i><b>1.2</b> Individual Sample Information</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="input-data.html"><a href="input-data.html#basic-sample-information"><i class="fa fa-check"></i><b>1.2.1</b> Basic sample information</a></li>
<li class="chapter" data-level="1.2.2" data-path="input-data.html"><a href="input-data.html#quality-control-info-from-thermo-fisher"><i class="fa fa-check"></i><b>1.2.2</b> Quality control info from Thermo Fisher</a></li>
<li class="chapter" data-level="1.2.3" data-path="input-data.html"><a href="input-data.html#list-of-oysters-that-passed-the-genotyping-according-to-thermofisher"><i class="fa fa-check"></i><b>1.2.3</b> List of oysters that passed the genotyping according to ThermoFisher</a></li>
<li class="chapter" data-level="1.2.4" data-path="input-data.html"><a href="input-data.html#merge-files"><i class="fa fa-check"></i><b>1.2.4</b> Merge Files</a></li>
<li class="chapter" data-level="1.2.5" data-path="input-data.html"><a href="input-data.html#tidy-up-latlong-data-and-site-codes"><i class="fa fa-check"></i><b>1.2.5</b> Tidy Up Lat/Long Data and Site Codes</a></li>
<li class="chapter" data-level="1.2.6" data-path="input-data.html"><a href="input-data.html#organize-oysters-into-metagroup-categories"><i class="fa fa-check"></i><b>1.2.6</b> Organize oysters into “metagroup” categories:</a></li>
<li class="chapter" data-level="1.2.7" data-path="input-data.html"><a href="input-data.html#map-sites-where-samples-were-collected"><i class="fa fa-check"></i><b>1.2.7</b> Map sites where samples were collected</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="input-data.html"><a href="input-data.html#snp-information"><i class="fa fa-check"></i><b>1.3</b> SNP Information</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="input-data.html"><a href="input-data.html#general-information-on-snps"><i class="fa fa-check"></i><b>1.3.1</b> General Information on SNPs</a></li>
<li class="chapter" data-level="1.3.2" data-path="input-data.html"><a href="input-data.html#recommended-snps"><i class="fa fa-check"></i><b>1.3.2</b> Recommended SNPs</a></li>
<li class="chapter" data-level="1.3.3" data-path="input-data.html"><a href="input-data.html#quality-control-info"><i class="fa fa-check"></i><b>1.3.3</b> Quality control info</a></li>
<li class="chapter" data-level="1.3.4" data-path="input-data.html"><a href="input-data.html#merge-files-1"><i class="fa fa-check"></i><b>1.3.4</b> Merge Files</a></li>
<li class="chapter" data-level="1.3.5" data-path="input-data.html"><a href="input-data.html#formatting-for-hierfstat-and-adegenet"><i class="fa fa-check"></i><b>1.3.5</b> Formatting for hierfstat and adegenet</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="input-data.html"><a href="input-data.html#save-files"><i class="fa fa-check"></i><b>1.4</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="genetic-groups.html"><a href="genetic-groups.html"><i class="fa fa-check"></i><b>2</b> Genetic Groups</a>
<ul>
<li class="chapter" data-level="2.1" data-path="genetic-groups.html"><a href="genetic-groups.html#objective-assign-individuals-to-genetic-groups"><i class="fa fa-check"></i><b>2.1</b> Objective: Assign individuals to genetic groups</a></li>
<li class="chapter" data-level="2.2" data-path="genetic-groups.html"><a href="genetic-groups.html#data-subsetting"><i class="fa fa-check"></i><b>2.2</b> Data Subsetting</a></li>
<li class="chapter" data-level="2.3" data-path="genetic-groups.html"><a href="genetic-groups.html#kmeans---all-wild-individuals"><i class="fa fa-check"></i><b>2.3</b> Kmeans - All Wild Individuals</a></li>
<li class="chapter" data-level="2.4" data-path="genetic-groups.html"><a href="genetic-groups.html#pca---k-groups"><i class="fa fa-check"></i><b>2.4</b> PCA - K groups</a></li>
<li class="chapter" data-level="2.5" data-path="genetic-groups.html"><a href="genetic-groups.html#kmeans-and-pca-with-four-groups"><i class="fa fa-check"></i><b>2.5</b> Kmeans and PCA with four groups</a></li>
<li class="chapter" data-level="2.6" data-path="genetic-groups.html"><a href="genetic-groups.html#pca---four-k-groups"><i class="fa fa-check"></i><b>2.6</b> PCA - Four K groups</a></li>
<li class="chapter" data-level="2.7" data-path="genetic-groups.html"><a href="genetic-groups.html#assign-genetic-groups"><i class="fa fa-check"></i><b>2.7</b> Assign Genetic Groups</a></li>
<li class="chapter" data-level="2.8" data-path="genetic-groups.html"><a href="genetic-groups.html#summary"><i class="fa fa-check"></i><b>2.8</b> Summary</a></li>
<li class="chapter" data-level="2.9" data-path="genetic-groups.html"><a href="genetic-groups.html#save-files-1"><i class="fa fa-check"></i><b>2.9</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html"><i class="fa fa-check"></i><b>3</b> Select Candidate SNPs</a>
<ul>
<li class="chapter" data-level="3.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#objective-identify-candidate-snps-by"><i class="fa fa-check"></i><b>3.1</b> Objective: Identify candidate SNPs by</a></li>
<li class="chapter" data-level="3.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#read-in-data"><i class="fa fa-check"></i><b>3.2</b> Read in Data</a></li>
<li class="chapter" data-level="3.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#format-data"><i class="fa fa-check"></i><b>3.3</b> Format Data</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genind-object-from-105k-snps"><i class="fa fa-check"></i><b>3.3.1</b> Make genind object from 105K SNPs</a></li>
<li class="chapter" data-level="3.3.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-a-dataframe-specifying-genetic-group-and-metapop."><i class="fa fa-check"></i><b>3.3.2</b> Make a dataframe specifying genetic group and “metapop.”</a></li>
<li class="chapter" data-level="3.3.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#assign-populations-to-genind-object"><i class="fa fa-check"></i><b>3.3.3</b> Assign populations to genind object</a></li>
<li class="chapter" data-level="3.3.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#format-data-into-the-hierfstat-format-for-analysis-of-heterozygosity-and-maf"><i class="fa fa-check"></i><b>3.3.4</b> Format data into the hierfstat format for analysis of heterozygosity and MAF</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#hwe-filtering"><i class="fa fa-check"></i><b>3.4</b> HWE filtering</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#filtering-out-the-loci-with-ho-0.5"><i class="fa fa-check"></i><b>3.4.1</b> Filtering out the loci with Ho &gt; 0.5</a></li>
<li class="chapter" data-level="3.4.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#re-analyze-the-data-for-heterozygosities"><i class="fa fa-check"></i><b>3.4.2</b> Re-analyze the data for heterozygosities</a></li>
<li class="chapter" data-level="3.4.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genind-object-without-the-loci-that-have-heterozygosities-over-0.5"><i class="fa fa-check"></i><b>3.4.3</b> Make genind object without the loci that have heterozygosities over 0.5</a></li>
<li class="chapter" data-level="3.4.4" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#make-genotypes-data-frame-without-the-loci-that-have-heterozygosities-over-0.5"><i class="fa fa-check"></i><b>3.4.4</b> Make genotypes data frame without the loci that have heterozygosities over 0.5</a></li>
<li class="chapter" data-level="3.4.5" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#hw-test"><i class="fa fa-check"></i><b>3.4.5</b> HW test</a></li>
<li class="chapter" data-level="3.4.6" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-loci-that-are-fail-hwe-test-p-0.05"><i class="fa fa-check"></i><b>3.4.6</b> Select loci that are fail HWE test (p &lt; 0.05)</a></li>
<li class="chapter" data-level="3.4.7" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#remove-the-loci-out-of-hwe-from-the-genind-object-and-genotypes-data-frame"><i class="fa fa-check"></i><b>3.4.7</b> Remove the loci out of HWE from the genind object and genotypes data frame</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-population-diagnostic-loci"><i class="fa fa-check"></i><b>3.5</b> Select population diagnostic loci</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#south-texas-vs.-gulf"><i class="fa fa-check"></i><b>3.5.1</b> South Texas vs. Gulf</a></li>
<li class="chapter" data-level="3.5.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-25-snps-that-distinguish-stx-from-gulf"><i class="fa fa-check"></i><b>3.5.2</b> Select 25 SNPs that distinguish stx from gulf</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#ngom-vs.-fl"><i class="fa fa-check"></i><b>3.6</b> nGOM vs. FL</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#select-25-snps-with-highest-fst-for-differentiating-nfl-vs.-gfl-vs.-ngom"><i class="fa fa-check"></i><b>3.6.1</b> Select 25 SNPs with highest Fst for differentiating nfl vs. gfl vs. ngom</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#maf"><i class="fa fa-check"></i><b>3.7</b> MAF</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#calculate-minor-allele-frequency-for-each-locus-in-each-genetic-group"><i class="fa fa-check"></i><b>3.7.1</b> Calculate Minor Allele Frequency for each locus in each genetic group</a></li>
<li class="chapter" data-level="3.7.2" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#re-calculate-after-removing-loci-with-maf-of-1"><i class="fa fa-check"></i><b>3.7.2</b> Re-calculate after removing loci with MAF of 1</a></li>
<li class="chapter" data-level="3.7.3" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#identify-loci-with-high-maf-in-all-groups"><i class="fa fa-check"></i><b>3.7.3</b> Identify loci with high MAF in all groups</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="select-candidate-snps.html"><a href="select-candidate-snps.html#save-files-2"><i class="fa fa-check"></i><b>3.8</b> Save files</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html"><i class="fa fa-check"></i><b>4</b> Select SNPs for Panel</a>
<ul>
<li class="chapter" data-level="4.1" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#read-in-all-data-on-snps-and-relevant-data-from-section-3"><i class="fa fa-check"></i><b>4.1</b> Read in all data on SNPs and relevant data from Section 3</a></li>
<li class="chapter" data-level="4.2" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#filter-the-combined_filtered-data-file-to-only-include-candidate-snps"><i class="fa fa-check"></i><b>4.2</b> Filter the combined_filtered data file to only include candidate SNPs</a></li>
<li class="chapter" data-level="4.3" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#specify-number-of-markers-from-each-chromosome-you-will-select"><i class="fa fa-check"></i><b>4.3</b> Specify number of markers from each chromosome you will select</a></li>
<li class="chapter" data-level="4.4" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#use-greedy-algorithm-to-select-5200-snps-for-5k-array-200-extras"><i class="fa fa-check"></i><b>4.4</b> Use greedy algorithm to select 5200 snps for 5K array (200 extras)</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#make-snp_tbl"><i class="fa fa-check"></i><b>4.4.1</b> Make snp_tbl</a></li>
<li class="chapter" data-level="4.4.2" data-path="select-snps-for-panel.html"><a href="select-snps-for-panel.html#check-and-make-sure-the-algorithm-gave-sensible-results"><i class="fa fa-check"></i><b>4.4.2</b> Check and make sure the algorithm gave sensible results</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:title:end-->
<!--bookdown:title:start-->
<div id="input-data" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">1</span> Input Data<a href="input-data.html#input-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="objective-organize-individual-and-snp-data" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Objective: Organize individual and SNP data<a href="input-data.html#objective-organize-individual-and-snp-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Organize and consolidate data from hundreds of C.virginica collected around the Gulf of Mexico genotyped on a high density SNP panel (~600K) developed by the East Coast consortium.</p>
</div>
<div id="individual-sample-information" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Individual Sample Information<a href="input-data.html#individual-sample-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>General Gulf (GG) samples (n=404) and South Texas (STX) samples (n=76) were genotyped on 600K ThermoFisher array and analyzed separately. A unique genotyping probeset list was used for each sample set. 447 samples passed QA.</p>
<div id="basic-sample-information" class="section level3 hasAnchor" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Basic sample information<a href="input-data.html#basic-sample-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>“sample_info_SALT” data file provided by Eric Salliant of University of Southern Mississippi, includes data provided by ThermoFisher and metadata (e.g., geographic details of where oysters were collected).</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="input-data.html#cb1-1" aria-hidden="true" tabindex="-1"></a>sample_info_SALT <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;raw&quot;</span>,<span class="st">&quot;cvir_gulf_600K_ES.csv&quot;</span>))</span></code></pre></div>
<pre><code>## New names:
## Rows: 480 Columns: 17
## ── Column specification
## ────────────────────────────────────────────────────────────────────────────── Delimiter: &quot;,&quot; chr
## (14): Sample, Filename, SubmittedPlate, sample_id, Geographic_code, Collection_data, SampleNam... dbl
## (3): ...1, Concentration, Reproducibility
## ℹ Use `spec()` to retrieve the full column specification for this data. ℹ Specify the column types or
## set `show_col_types = FALSE` to quiet this message.
## • `` -&gt; `...1`</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="input-data.html#cb3-1" aria-hidden="true" tabindex="-1"></a>sample_info_SALT<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="input-data.html#cb4-1" aria-hidden="true" tabindex="-1"></a>sample_info_SALT</span></code></pre></div>
<pre><code>## # A tibble: 480 × 16
##    Sample    Filename SubmittedPlate sample_id Geographic_code Collection_data SampleName SampleSource
##    &lt;chr&gt;     &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;           &lt;chr&gt;      &lt;chr&gt;       
##  1 a551319-… 9SR48340 F12            BWBFL_1   BWBFL           25.92860, -81.… Customer   Genomic     
##  2 a551319-… 9SR48340 G08            BWBFL_10  BWBFL           25.92860, -81.… Customer   Genomic     
##  3 a551319-… 9SR48340 G09            BWBFL_11  BWBFL           25.92860, -81.… Customer   Genomic     
##  4 a551319-… 9SR48340 G10            BWBFL_12  BWBFL           25.92860, -81.… Customer   Genomic     
##  5 a551319-… 9SR48340 G11            BWBFL_14  BWBFL           25.92860, -81.… Customer   Genomic     
##  6 a551319-… 9SR48340 G12            BWBFL_15  BWBFL           25.92860, -81.… Customer   Genomic     
##  7 a551319-… 9SR48340 H01            BWBFL_16  BWBFL           25.92860, -81.… Customer   Genomic     
##  8 a551319-… 9SR48340 H02            BWBFL_17  BWBFL           25.92860, -81.… Customer   Genomic     
##  9 a551319-… 9SR48340 H03            BWBFL_18  BWBFL           25.92860, -81.… Customer   Genomic     
## 10 a551319-… 9SR48340 H04            BWBFL_19  BWBFL           25.92860, -81.… Customer   Genomic     
## # ℹ 470 more rows
## # ℹ 8 more variables: SampleType &lt;chr&gt;, Species &lt;chr&gt;, Concentration &lt;dbl&gt;, PlateName &lt;chr&gt;,
## #   Rename &lt;chr&gt;, Reproducibility &lt;dbl&gt;, Gender &lt;chr&gt;, SampleOrigin &lt;chr&gt;</code></pre>
<p>“sample_info_RAD” data file provided by Amanda Barker of the Marine Genomics Lab, it is data of a TPWD and Marine Genomics Lab project using RAD</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="input-data.html#cb6-1" aria-hidden="true" tabindex="-1"></a>sample_info_RAD <span class="ot">&lt;-</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;raw&quot;</span>,<span class="st">&quot;oyster_genotyped_sample_info.csv&quot;</span>))</span></code></pre></div>
<pre><code>## New names:
## Rows: 463 Columns: 15
## ── Column specification
## ────────────────────────────────────────────────────────────────────────────── Delimiter: &quot;,&quot; chr
## (11): lib_id, sample_id, barcode, index, site, region, gulf_location, state, bay, tidal.zone, ... dbl
## (4): ...1, lib, latitude, longitude
## ℹ Use `spec()` to retrieve the full column specification for this data. ℹ Specify the column types or
## set `show_col_types = FALSE` to quiet this message.
## • `` -&gt; `...1`</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="input-data.html#cb8-1" aria-hidden="true" tabindex="-1"></a>sample_info_RAD<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="input-data.html#cb9-1" aria-hidden="true" tabindex="-1"></a>sample_info_RAD</span></code></pre></div>
<pre><code>## # A tibble: 463 × 14
##    lib_id sample_id barcode index    site      lib region gulf_location state bay   latitude longitude
##    &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt;     &lt;dbl&gt;
##  1 apa_01 59751     TCAGT   index_04 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
##  2 apa_02 59752     AGCTA   index_04 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
##  3 apa_03 59753     CGGTA   index_04 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
##  4 apa_04 59754     GGATA   index_04 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
##  5 apa_05 59755     TCCGG   index_04 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
##  6 apa_06 59756     ACACA   index_04 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
##  7 apa_07 59757     GCATG   index_03 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
##  8 apa_08 59758     ACTGG   index_03 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
##  9 apa_09 59759     CTGCG   index_03 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
## 10 apa_10 59760     GTCCG   index_03 apalac…     1 north… Eastern       FL    Apal…     29.7     -84.9
## # ℹ 453 more rows
## # ℹ 2 more variables: tidal.zone &lt;chr&gt;, harvest &lt;chr&gt;</code></pre>
<p>Some sample info from the RAD dataset was in a seperate file</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="input-data.html#cb11-1" aria-hidden="true" tabindex="-1"></a>missingdata <span class="ot">&lt;-</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;raw&quot;</span>,<span class="st">&quot;missing_cvir.csv&quot;</span>))</span>
<span id="cb11-2"><a href="input-data.html#cb11-2" aria-hidden="true" tabindex="-1"></a>missingdata</span></code></pre></div>
<pre><code>## # A tibble: 5 × 8
##   sample_id gulf_location state    bay                latitude longitude `tidal zone` harvest
##       &lt;dbl&gt; &lt;chr&gt;         &lt;chr&gt;    &lt;chr&gt;                 &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;        &lt;chr&gt;  
## 1     60111 Southwestern  TX-Lower Upper Laguna Madre     27.6     -97.2 Intertidal   No     
## 2     60113 Southwestern  TX-Lower Upper Laguna Madre     27.6     -97.2 Intertidal   No     
## 3     60008 Southwestern  TX-Lower Port Mansfield         26.6     -97.4 Intertidal   No     
## 4     60012 Southwestern  TX-Lower Port Mansfield         26.6     -97.4 Intertidal   No     
## 5     60013 Southwestern  TX-Lower Port Mansfield         26.6     -97.4 Intertidal   No</code></pre>
</div>
<div id="quality-control-info-from-thermo-fisher" class="section level3 hasAnchor" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Quality control info from Thermo Fisher<a href="input-data.html#quality-control-info-from-thermo-fisher" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="input-data.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in QC data for both data sets</span></span>
<span id="cb13-2"><a href="input-data.html#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="input-data.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co">#not sure how to use read_table to imitate read.table, so &quot;here&quot; tabled for now</span></span>
<span id="cb13-4"><a href="input-data.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co">#other_qc &lt;- read_table(here::here(&quot;data&quot;,&quot;raw&quot;,&quot;other_genotypingQC.report.txt&quot;), col_names = TRUE)</span></span>
<span id="cb13-5"><a href="input-data.html#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="input-data.html#cb13-6" aria-hidden="true" tabindex="-1"></a>other_qc <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_genotypingQC.report.txt&quot;</span>,<span class="at">fill=</span><span class="cn">TRUE</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-7"><a href="input-data.html#cb13-7" aria-hidden="true" tabindex="-1"></a>south_qc <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_genotypingQC.report.txt&quot;</span>, <span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb13-8"><a href="input-data.html#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="input-data.html#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co">#add column of data_set - &quot;other&quot; or &quot;south_texas&quot;</span></span>
<span id="cb13-10"><a href="input-data.html#cb13-10" aria-hidden="true" tabindex="-1"></a>other_qc<span class="sc">$</span>data_set <span class="ot">&lt;-</span> <span class="st">&quot;other&quot;</span></span>
<span id="cb13-11"><a href="input-data.html#cb13-11" aria-hidden="true" tabindex="-1"></a>south_qc<span class="sc">$</span>data_set <span class="ot">&lt;-</span> <span class="st">&quot;south_texas&quot;</span></span>
<span id="cb13-12"><a href="input-data.html#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="input-data.html#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co">#rbind &quot;other&quot; and &quot;south_texas&quot; data frames</span></span>
<span id="cb13-14"><a href="input-data.html#cb13-14" aria-hidden="true" tabindex="-1"></a>qc_indv <span class="ot">&lt;-</span> <span class="fu">rbind</span>(other_qc,south_qc)</span>
<span id="cb13-15"><a href="input-data.html#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="input-data.html#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co">#rename in QC &quot;celfiles&quot; to &quot;Sample&quot;</span></span>
<span id="cb13-17"><a href="input-data.html#cb13-17" aria-hidden="true" tabindex="-1"></a>qc_indv <span class="ot">&lt;-</span> qc_indv <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Sample =</span> cel_files)</span>
<span id="cb13-18"><a href="input-data.html#cb13-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-19"><a href="input-data.html#cb13-19" aria-hidden="true" tabindex="-1"></a>qc_indv <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##                               Sample computed_gender call_rate total_call_rate het_rate
## 1 a551319-4448066-110623-379_A01.CEL          99.020    99.020          21.630   21.630
## 2 a551319-4448066-110623-379_A02.CEL          99.280    99.280          22.040   22.040
## 3 a551319-4448066-110623-379_A03.CEL          99.245    99.245          21.835   21.835
## 4 a551319-4448066-110623-379_A04.CEL          99.360    99.360          22.005   22.005
## 5 a551319-4448066-110623-379_A05.CEL          99.380    99.380          22.610   22.610
## 6 a551319-4448066-110623-379_A06.CEL          99.435    99.435          22.800   22.800
##   total_het_rate hom_rate total_hom_rate cluster_distance_mean cluster_distance_stdev
## 1         77.390   77.390        0.71195               0.57517               11.54653
## 2         77.240   77.240        0.75682               0.57020               11.53059
## 3         77.410   77.410        0.69493               0.54648               11.53760
## 4         77.355   77.355        0.71897               0.54597               11.54199
## 5         76.770   76.770        0.70479               0.54023               11.54943
## 6         76.635   76.635        0.71709               0.53633               11.55852
##   allele_summarization_mean allele_summarization_stdev allele_deviation_mean allele_deviation_stdev
## 1                   1.59285                    0.65691               0.71869                0.14285
## 2                   1.59604                    0.64291               0.69209                0.09128
## 3                   1.58928                    0.61509               0.70068                0.13653
## 4                   1.58284                    0.62758               0.69696                0.13401
## 5                   1.58350                    0.62034               0.70294                0.12122
## 6                   1.57922                    0.61985               0.70110                0.10657
##   allele_mad_residuals_mean allele_mad_residuals_stdev affymetrix.plate.barcode
## 1                   0.12355               5.513194e+21                      A01
## 2                   0.08666               5.513194e+21                      A02
## 3                   0.11354               5.513194e+21                      A03
## 4                   0.10643               5.513194e+21                      A04
## 5                   0.10181               5.513194e+21                      A05
## 6                   0.09364               5.513194e+21                      A06
##   affymetrix.plate.peg.wellposition data_set
## 1                                NA    other
## 2                                NA    other
## 3                                NA    other
## 4                                NA    other
## 5                                NA    other
## 6                                NA    other</code></pre>
</div>
<div id="list-of-oysters-that-passed-the-genotyping-according-to-thermofisher" class="section level3 hasAnchor" number="1.2.3">
<h3><span class="header-section-number">1.2.3</span> List of oysters that passed the genotyping according to ThermoFisher<a href="input-data.html#list-of-oysters-that-passed-the-genotyping-according-to-thermofisher" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="input-data.html#cb15-1" aria-hidden="true" tabindex="-1"></a>gg_qc_filt <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_sample_QCFilteredCR.txt&quot;</span>,<span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-2"><a href="input-data.html#cb15-2" aria-hidden="true" tabindex="-1"></a>passed_GG <span class="ot">&lt;-</span> gg_qc_filt<span class="sc">$</span>cel_files</span>
<span id="cb15-3"><a href="input-data.html#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="input-data.html#cb15-4" aria-hidden="true" tabindex="-1"></a>st_qc_filt <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_sample_QCFilteredCR.txt&quot;</span>,<span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-5"><a href="input-data.html#cb15-5" aria-hidden="true" tabindex="-1"></a>passed_ST <span class="ot">&lt;-</span> st_qc_filt<span class="sc">$</span>cel_files</span></code></pre></div>
</div>
<div id="merge-files" class="section level3 hasAnchor" number="1.2.4">
<h3><span class="header-section-number">1.2.4</span> Merge Files<a href="input-data.html#merge-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Consolidate all identifying information on samples to “sample_info_full”</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="input-data.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#merge sample_info_SALT and sample_info_RAD</span></span>
<span id="cb16-2"><a href="input-data.html#cb16-2" aria-hidden="true" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="at">x =</span> sample_info_SALT, <span class="at">y =</span> sample_info_RADx, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-3"><a href="input-data.html#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="input-data.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co">#add quality control data, filter out only individuals that passed Thermo&#39;s quality control</span></span>
<span id="cb16-5"><a href="input-data.html#cb16-5" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="ot">&lt;-</span> sample_info <span class="sc">%&gt;%</span></span>
<span id="cb16-6"><a href="input-data.html#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(qc_indv, <span class="at">all.x =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-7"><a href="input-data.html#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span> (Sample <span class="sc">%in%</span> passed_GG <span class="sc">|</span> Sample <span class="sc">%in%</span> passed_ST) </span>
<span id="cb16-8"><a href="input-data.html#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="input-data.html#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="co">#some individuals have multiple rows in the data set, with the only varying information the lib_id, barcode, and index columns, likely RAD information.  Below, choose only one row for these individuals: a551319-4448066-110623-383_A04.CEL and a551319-4448066-110623-383_A08.CEL</span></span>
<span id="cb16-10"><a href="input-data.html#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="input-data.html#cb16-11" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="ot">&lt;-</span> sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb16-12"><a href="input-data.html#cb16-12" aria-hidden="true" tabindex="-1"></a> <span class="fu">distinct</span>(Sample, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-13"><a href="input-data.html#cb16-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-14"><a href="input-data.html#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="co">#add a &quot;locaton&quot; column for the RAD data</span></span>
<span id="cb16-15"><a href="input-data.html#cb16-15" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>location <span class="ot">&lt;-</span> <span class="fu">paste</span>(sample_info_full<span class="sc">$</span>bay,sample_info_full<span class="sc">$</span>state,<span class="at">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb16-16"><a href="input-data.html#cb16-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-17"><a href="input-data.html#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="co">#remove all periods and hyphens from sample name (causes issues downstream)</span></span>
<span id="cb16-18"><a href="input-data.html#cb16-18" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>Sample)</span>
<span id="cb16-19"><a href="input-data.html#cb16-19" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>Sample)</span>
<span id="cb16-20"><a href="input-data.html#cb16-20" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>Sample)</span>
<span id="cb16-21"><a href="input-data.html#cb16-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-22"><a href="input-data.html#cb16-22" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##                          Sample sample_id Filename SubmittedPlate        Geographic_code
## 1 a5513194448066110623379A01CEL  Auburn_1 9SR48342            A01 Auburn selected line 1
## 2 a5513194448066110623379A02CEL  Auburn_2 9SR48342            A02 Auburn selected line 1
## 3 a5513194448066110623379A03CEL  Auburn_3 9SR48342            A03 Auburn selected line 1
## 4 a5513194448066110623379A04CEL  Auburn_4 9SR48342            A04 Auburn selected line 1
## 5 a5513194448066110623379A05CEL  Auburn_6 9SR48342            A05 Auburn selected line 1
## 6 a5513194448066110623379A06CEL  Auburn_7 9SR48342            A06 Auburn selected line 1
##          Collection_data SampleName SampleSource SampleType               Species Concentration
## 1 Auburn selected line 1   Customer      Genomic        DNA Crassostrea_virginica        21.867
## 2 Auburn selected line 1   Customer      Genomic        DNA Crassostrea_virginica        80.802
## 3 Auburn selected line 1   Customer      Genomic        DNA Crassostrea_virginica        49.900
## 4 Auburn selected line 1   Customer      Genomic        DNA Crassostrea_virginica        28.870
## 5 Auburn selected line 1   Customer      Genomic        DNA Crassostrea_virginica        28.704
## 6 Auburn selected line 1   Customer      Genomic        DNA Crassostrea_virginica       103.718
##     PlateName   Rename Reproducibility Gender SampleOrigin lib_id barcode index site lib region
## 1 379_4448066 Auburn_1              NA      U       Auburn   &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;  NA   &lt;NA&gt;
## 2 379_4448066 Auburn_2              NA      U       Auburn   &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;  NA   &lt;NA&gt;
## 3 379_4448066 Auburn_3              NA      U       Auburn   &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;  NA   &lt;NA&gt;
## 4 379_4448066 Auburn_4              NA      U       Auburn   &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;  NA   &lt;NA&gt;
## 5 379_4448066 Auburn_6              NA      U       Auburn   &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;  NA   &lt;NA&gt;
## 6 379_4448066 Auburn_7              NA      U       Auburn   &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;  NA   &lt;NA&gt;
##   gulf_location state  bay latitude longitude tidal.zone harvest tidal zone computed_gender call_rate
## 1          &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;       NA        NA       &lt;NA&gt;    &lt;NA&gt;       &lt;NA&gt;          99.020    99.020
## 2          &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;       NA        NA       &lt;NA&gt;    &lt;NA&gt;       &lt;NA&gt;          99.280    99.280
## 3          &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;       NA        NA       &lt;NA&gt;    &lt;NA&gt;       &lt;NA&gt;          99.245    99.245
## 4          &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;       NA        NA       &lt;NA&gt;    &lt;NA&gt;       &lt;NA&gt;          99.360    99.360
## 5          &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;       NA        NA       &lt;NA&gt;    &lt;NA&gt;       &lt;NA&gt;          99.380    99.380
## 6          &lt;NA&gt;  &lt;NA&gt; &lt;NA&gt;       NA        NA       &lt;NA&gt;    &lt;NA&gt;       &lt;NA&gt;          99.435    99.435
##   total_call_rate het_rate total_het_rate hom_rate total_hom_rate cluster_distance_mean
## 1          21.630   21.630         77.390   77.390        0.71195               0.57517
## 2          22.040   22.040         77.240   77.240        0.75682               0.57020
## 3          21.835   21.835         77.410   77.410        0.69493               0.54648
## 4          22.005   22.005         77.355   77.355        0.71897               0.54597
## 5          22.610   22.610         76.770   76.770        0.70479               0.54023
## 6          22.800   22.800         76.635   76.635        0.71709               0.53633
##   cluster_distance_stdev allele_summarization_mean allele_summarization_stdev allele_deviation_mean
## 1               11.54653                   1.59285                    0.65691               0.71869
## 2               11.53059                   1.59604                    0.64291               0.69209
## 3               11.53760                   1.58928                    0.61509               0.70068
## 4               11.54199                   1.58284                    0.62758               0.69696
## 5               11.54943                   1.58350                    0.62034               0.70294
## 6               11.55852                   1.57922                    0.61985               0.70110
##   allele_deviation_stdev allele_mad_residuals_mean allele_mad_residuals_stdev
## 1                0.14285                   0.12355               5.513194e+21
## 2                0.09128                   0.08666               5.513194e+21
## 3                0.13653                   0.11354               5.513194e+21
## 4                0.13401                   0.10643               5.513194e+21
## 5                0.12122                   0.10181               5.513194e+21
## 6                0.10657                   0.09364               5.513194e+21
##   affymetrix.plate.barcode affymetrix.plate.peg.wellposition data_set location
## 1                      A01                                NA    other    NA,NA
## 2                      A02                                NA    other    NA,NA
## 3                      A03                                NA    other    NA,NA
## 4                      A04                                NA    other    NA,NA
## 5                      A05                                NA    other    NA,NA
## 6                      A06                                NA    other    NA,NA</code></pre>
</div>
<div id="tidy-up-latlong-data-and-site-codes" class="section level3 hasAnchor" number="1.2.5">
<h3><span class="header-section-number">1.2.5</span> Tidy Up Lat/Long Data and Site Codes<a href="input-data.html#tidy-up-latlong-data-and-site-codes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Fix inconsistencies in geographic data and site codes</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="input-data.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#lat and long data differs in format for samples from SALT and RAD data files</span></span>
<span id="cb18-2"><a href="input-data.html#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="input-data.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 1 - tidy up lat and long from SALT data files</span></span>
<span id="cb18-4"><a href="input-data.html#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="input-data.html#cb18-5" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>lat<span class="ot">&lt;-</span><span class="fu">substr</span>(sample_info_full<span class="sc">$</span>Collection_data,<span class="dv">1</span>,<span class="dv">8</span>)</span>
<span id="cb18-6"><a href="input-data.html#cb18-6" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>long<span class="ot">&lt;-</span><span class="fu">str_sub</span>(sample_info_full<span class="sc">$</span>Collection_data,<span class="sc">-</span><span class="dv">10</span>)</span>
<span id="cb18-7"><a href="input-data.html#cb18-7" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>long <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>long)</span>
<span id="cb18-8"><a href="input-data.html#cb18-8" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>long <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>, sample_info_full<span class="sc">$</span>long)</span>
<span id="cb18-9"><a href="input-data.html#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="input-data.html#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 - add the SALT lat and long data to the latitude and longitude  </span></span>
<span id="cb18-11"><a href="input-data.html#cb18-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-12"><a href="input-data.html#cb18-12" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>latitude <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(sample_info_full<span class="sc">$</span>latitude),</span>
<span id="cb18-13"><a href="input-data.html#cb18-13" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">as.numeric</span>(sample_info_full<span class="sc">$</span>lat),sample_info_full<span class="sc">$</span>latitude)</span>
<span id="cb18-14"><a href="input-data.html#cb18-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-15"><a href="input-data.html#cb18-15" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>longitude <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(sample_info_full<span class="sc">$</span>longitude),</span>
<span id="cb18-16"><a href="input-data.html#cb18-16" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">as.numeric</span>(sample_info_full<span class="sc">$</span>long),sample_info_full<span class="sc">$</span>longitude)</span>
<span id="cb18-17"><a href="input-data.html#cb18-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-18"><a href="input-data.html#cb18-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-19"><a href="input-data.html#cb18-19" aria-hidden="true" tabindex="-1"></a><span class="co"># 3 - as numeric lat and long data</span></span>
<span id="cb18-20"><a href="input-data.html#cb18-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-21"><a href="input-data.html#cb18-21" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>latitude <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(sample_info_full<span class="sc">$</span>latitude)</span>
<span id="cb18-22"><a href="input-data.html#cb18-22" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>longitude <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(sample_info_full<span class="sc">$</span>longitude)</span>
<span id="cb18-23"><a href="input-data.html#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="input-data.html#cb18-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-25"><a href="input-data.html#cb18-25" aria-hidden="true" tabindex="-1"></a><span class="co"># - 4 add &quot;geographic_code&quot; information to those without it </span></span>
<span id="cb18-26"><a href="input-data.html#cb18-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-27"><a href="input-data.html#cb18-27" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code<span class="ot">&lt;-</span><span class="fu">as.character</span>(sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb18-28"><a href="input-data.html#cb18-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-29"><a href="input-data.html#cb18-29" aria-hidden="true" tabindex="-1"></a><span class="co">#ULTX 1/2</span></span>
<span id="cb18-30"><a href="input-data.html#cb18-30" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>bay <span class="sc">%in%</span> <span class="st">&quot;Upper Laguna Madre&quot;</span>,<span class="st">&quot;ULTX&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb18-31"><a href="input-data.html#cb18-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-32"><a href="input-data.html#cb18-32" aria-hidden="true" tabindex="-1"></a><span class="co">#ULTX 2/2</span></span>
<span id="cb18-33"><a href="input-data.html#cb18-33" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>sample_id <span class="sc">%like%</span> <span class="st">&quot;ULM&quot;</span>,<span class="st">&quot;ULTX&quot;</span>,sample_info_full<span class="sc">$</span>Geographic_code) <span class="co">#like requires data.table package</span></span>
<span id="cb18-34"><a href="input-data.html#cb18-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-35"><a href="input-data.html#cb18-35" aria-hidden="true" tabindex="-1"></a><span class="co">#PMTX Port Mansfield Texas</span></span>
<span id="cb18-36"><a href="input-data.html#cb18-36" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>bay <span class="sc">%in%</span> <span class="st">&quot;Port Mansfield&quot;</span>,<span class="st">&quot;PMTX&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb18-37"><a href="input-data.html#cb18-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-38"><a href="input-data.html#cb18-38" aria-hidden="true" tabindex="-1"></a><span class="co">#Corpus Christi Bay Texas</span></span>
<span id="cb18-39"><a href="input-data.html#cb18-39" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>site <span class="sc">%in%</span> <span class="st">&quot;corpus_christi&quot;</span>,<span class="st">&quot;CCTX&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb18-40"><a href="input-data.html#cb18-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-41"><a href="input-data.html#cb18-41" aria-hidden="true" tabindex="-1"></a><span class="co">#Tampa Bay FL</span></span>
<span id="cb18-42"><a href="input-data.html#cb18-42" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>lib_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;tam_01&quot;</span>,<span class="st">&quot;tam_04&quot;</span>,<span class="st">&quot;tam_05&quot;</span>,<span class="st">&quot;tam_06&quot;</span>),<span class="st">&quot;TBFL&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb18-43"><a href="input-data.html#cb18-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-44"><a href="input-data.html#cb18-44" aria-hidden="true" tabindex="-1"></a><span class="co">#Boca Ciega Bay Aquatic Preserve FL</span></span>
<span id="cb18-45"><a href="input-data.html#cb18-45" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>lib_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;tam_20&quot;</span>,<span class="st">&quot;tam_22&quot;</span>,<span class="st">&quot;tam_23&quot;</span>),<span class="st">&quot;BCFL&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb18-46"><a href="input-data.html#cb18-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-47"><a href="input-data.html#cb18-47" aria-hidden="true" tabindex="-1"></a><span class="co">#Madelaine Key, FL</span></span>
<span id="cb18-48"><a href="input-data.html#cb18-48" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>lib_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;tam_17&quot;</span>,<span class="st">&quot;tam_18&quot;</span>),<span class="st">&quot;MKFL&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb18-49"><a href="input-data.html#cb18-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-50"><a href="input-data.html#cb18-50" aria-hidden="true" tabindex="-1"></a><span class="co">#Caloosahatchee River, FL</span></span>
<span id="cb18-51"><a href="input-data.html#cb18-51" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>lib_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;car_01&quot;</span>,<span class="st">&quot;car_03&quot;</span>,<span class="st">&quot;car_05&quot;</span>,<span class="st">&quot;car_07&quot;</span>),<span class="st">&quot;CAFL&quot;</span>, sample_info_full<span class="sc">$</span>Geographic_code)</span>
<span id="cb18-52"><a href="input-data.html#cb18-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-53"><a href="input-data.html#cb18-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-54"><a href="input-data.html#cb18-54" aria-hidden="true" tabindex="-1"></a><span class="co"># - 5 add lat and long to ULM obs that do not have this info</span></span>
<span id="cb18-55"><a href="input-data.html#cb18-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-56"><a href="input-data.html#cb18-56" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>latitude <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="st">&quot;ULTX&quot;</span>,<span class="fl">27.63083</span>,sample_info_full<span class="sc">$</span>latitude)</span>
<span id="cb18-57"><a href="input-data.html#cb18-57" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>longitude <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="st">&quot;ULTX&quot;</span>,<span class="sc">-</span><span class="fl">97.24018</span>,sample_info_full<span class="sc">$</span>longitude)</span></code></pre></div>
</div>
<div id="organize-oysters-into-metagroup-categories" class="section level3 hasAnchor" number="1.2.6">
<h3><span class="header-section-number">1.2.6</span> Organize oysters into “metagroup” categories:<a href="input-data.html#organize-oysters-into-metagroup-categories" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>South Texas (STX)</li>
<li>northern Gulf of Mexico (nGOM)</li>
<li>Florida (FL)</li>
<li>Selected Line - Auburn nGOM</li>
<li>Selected Line - Auburn FL</li>
<li>Selected Line - OBOY</li>
</ul>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="input-data.html#cb19-1" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="st">&quot;selected lines&quot;</span></span>
<span id="cb19-2"><a href="input-data.html#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="input-data.html#cb19-3" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;AHFL&quot;</span>,<span class="st">&quot;BWBFL&quot;</span>,</span>
<span id="cb19-4"><a href="input-data.html#cb19-4" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;OBFL&quot;</span>,<span class="st">&quot;LRFL&quot;</span>,</span>
<span id="cb19-5"><a href="input-data.html#cb19-5" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;CRFL&quot;</span>,</span>
<span id="cb19-6"><a href="input-data.html#cb19-6" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;PBFL&quot;</span>,<span class="st">&quot;SKFL&quot;</span>,</span>
<span id="cb19-7"><a href="input-data.html#cb19-7" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;CAFL&quot;</span>,<span class="st">&quot;TBFL&quot;</span>,</span>
<span id="cb19-8"><a href="input-data.html#cb19-8" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;MKFL&quot;</span>,</span>
<span id="cb19-9"><a href="input-data.html#cb19-9" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;BCFL&quot;</span>),<span class="st">&quot;FL&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb19-10"><a href="input-data.html#cb19-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-11"><a href="input-data.html#cb19-11" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;ALAL&quot;</span>,<span class="st">&quot;CPAL&quot;</span>,</span>
<span id="cb19-12"><a href="input-data.html#cb19-12" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;EGTX&quot;</span>,<span class="st">&quot;EMTX&quot;</span>, </span>
<span id="cb19-13"><a href="input-data.html#cb19-13" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;LCLA&quot;</span>,<span class="st">&quot;LSTX&quot;</span>,</span>
<span id="cb19-14"><a href="input-data.html#cb19-14" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;PSMS&quot;</span>,<span class="st">&quot;SATX&quot;</span>,</span>
<span id="cb19-15"><a href="input-data.html#cb19-15" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;SLLA&quot;</span>,<span class="st">&quot;WGTX&quot;</span>,</span>
<span id="cb19-16"><a href="input-data.html#cb19-16" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;WMTX&quot;</span>),<span class="st">&quot;nGOM&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb19-17"><a href="input-data.html#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="input-data.html#cb19-18" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Geographic_code <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;ULTX&quot;</span>,<span class="st">&quot;PMTX&quot;</span>,</span>
<span id="cb19-19"><a href="input-data.html#cb19-19" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">&quot;CCTX&quot;</span>),<span class="st">&quot;STX&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb19-20"><a href="input-data.html#cb19-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-21"><a href="input-data.html#cb19-21" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Collection_data <span class="sc">%in%</span> <span class="st">&quot;Auburn selected line 1&quot;</span>,</span>
<span id="cb19-22"><a href="input-data.html#cb19-22" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Selected Line - Auburn nGOM&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb19-23"><a href="input-data.html#cb19-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-24"><a href="input-data.html#cb19-24" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Collection_data <span class="sc">%in%</span> <span class="st">&quot;Auburn selected line 2&quot;</span>,</span>
<span id="cb19-25"><a href="input-data.html#cb19-25" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Selected Line - Auburn FL&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span>
<span id="cb19-26"><a href="input-data.html#cb19-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-27"><a href="input-data.html#cb19-27" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>metagroup <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_info_full<span class="sc">$</span>Collection_data <span class="sc">%in%</span> <span class="st">&quot;O&#39;Boy selected line&quot;</span>,</span>
<span id="cb19-28"><a href="input-data.html#cb19-28" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Selected Line - OBOY&quot;</span>,sample_info_full<span class="sc">$</span>metagroup)</span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="input-data.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#order the factor levels for geographic code based on longitude (move west to east)</span></span>
<span id="cb20-2"><a href="input-data.html#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="input-data.html#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb20-4"><a href="input-data.html#cb20-4" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="ot">&lt;-</span><span class="fu">arrange</span>(sample_info_full,longitude)</span>
<span id="cb20-5"><a href="input-data.html#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="input-data.html#cb20-6" aria-hidden="true" tabindex="-1"></a>sample_info_full<span class="sc">$</span>Geographic_code<span class="ot">&lt;-</span><span class="fu">factor</span>(sample_info_full<span class="sc">$</span>Geographic_code, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;PMTX&quot;</span>,<span class="st">&quot;ULTX&quot;</span>,<span class="st">&quot;CCTX&quot;</span>,<span class="st">&quot;SATX&quot;</span>,<span class="st">&quot;WMTX&quot;</span>,<span class="st">&quot;EMTX&quot;</span>,<span class="st">&quot;WGTX&quot;</span>,<span class="st">&quot;EGTX&quot;</span>,<span class="st">&quot;LSTX&quot;</span>,<span class="st">&quot;LCLA&quot;</span>,<span class="st">&quot;SLLA&quot;</span>,<span class="st">&quot;PSMS&quot;</span>,<span class="st">&quot;CPAL&quot;</span>,<span class="st">&quot;ALAL&quot;</span>,<span class="st">&quot;PBFL&quot;</span>,<span class="st">&quot;AHFL&quot;</span>,<span class="st">&quot;OBFL&quot;</span>,<span class="st">&quot;LRFL&quot;</span>,<span class="st">&quot;SKFL&quot;</span>,<span class="st">&quot;CRFL&quot;</span>,<span class="st">&quot;MKFL&quot;</span>,<span class="st">&quot;BCFL&quot;</span>,<span class="st">&quot;TBFL&quot;</span>,<span class="st">&quot;CAFL&quot;</span>,<span class="st">&quot;BWBFL&quot;</span>))</span></code></pre></div>
</div>
<div id="map-sites-where-samples-were-collected" class="section level3 hasAnchor" number="1.2.7">
<h3><span class="header-section-number">1.2.7</span> Map sites where samples were collected<a href="input-data.html#map-sites-where-samples-were-collected" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Site codes and site names:</p>
<ul>
<li>PMTX ~ Port Mansfield</li>
<li>ULTX ~ Upper Laguna Madre</li>
<li>CCTX ~ Corpus Christi Bay</li>
<li>SATX ~ San Antonio Bay</li>
<li>WMTX ~ West Matagorda Bay</li>
<li>EMTX ~ East Matagorda Bay</li>
<li>WGTX ~ West Galveston Bay</li>
<li>EGTX ~ East Galveston Bay</li>
<li>LSTX ~ Sabine Pass</li>
<li>LCLA ~ Calcasieu Lake</li>
<li>SLLA ~ Caillou Lake</li>
<li>PSMS ~ Pascagoula River</li>
<li>CPAL ~ Cedar Point</li>
<li>ALAL ~ Alligator Lake</li>
<li>PBFL ~ Pensacola Bay</li>
<li>AHFL ~ Alligator Harbor</li>
<li>OBFL ~ Oyster Bay</li>
<li>LRFL ~ Suwannee River</li>
<li>SKFL ~ Seahorse Key</li>
<li>CRFL ~ Corrigan Reef</li>
<li>MKFL ~ Madelaine Key</li>
<li>BCFL ~ Boca Ciega Bay Aquatic Preserve</li>
<li>TBFL ~ Tampa Bay</li>
<li>CAFL ~ Caloosahatchee River</li>
<li>BWBFL ~ Backwater Bay</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="input-data.html#cb21-1" aria-hidden="true" tabindex="-1"></a>MainStates <span class="ot">&lt;-</span> <span class="fu">map_data</span>(<span class="st">&quot;state&quot;</span>)</span>
<span id="cb21-2"><a href="input-data.html#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="input-data.html#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb21-4"><a href="input-data.html#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;lightgray&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb21-5"><a href="input-data.html#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb21-6"><a href="input-data.html#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb21-7"><a href="input-data.html#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb21-8"><a href="input-data.html#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>)) <span class="sc">+</span> </span>
<span id="cb21-9"><a href="input-data.html#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> sample_info_full <span class="sc">%&gt;%</span>   </span>
<span id="cb21-10"><a href="input-data.html#cb21-10" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(<span class="sc">!</span>Collection_data <span class="sc">==</span> <span class="st">&quot;Auburn selected line 2&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-11"><a href="input-data.html#cb21-11" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(<span class="sc">!</span>Collection_data <span class="sc">==</span> <span class="st">&quot;O&#39;Boy selected line&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-12"><a href="input-data.html#cb21-12" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(<span class="sc">!</span>Collection_data <span class="sc">==</span> <span class="st">&quot;Auburn selected line 1&quot;</span>), <span class="co">#remove the selected lines from mapping</span></span>
<span id="cb21-13"><a href="input-data.html#cb21-13" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> latitude, <span class="at">x =</span> longitude, <span class="at">group=</span>Geographic_code,<span class="at">colour=</span>Geographic_code,<span class="at">shape =</span> metagroup),</span>
<span id="cb21-14"><a href="input-data.html#cb21-14" aria-hidden="true" tabindex="-1"></a>             <span class="at">size=</span><span class="dv">5</span>,<span class="at">stroke=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>Map representing the relative number of samples collected in each location</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="input-data.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb22-2"><a href="input-data.html#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;lightgray&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb22-3"><a href="input-data.html#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb22-4"><a href="input-data.html#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb22-5"><a href="input-data.html#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb22-6"><a href="input-data.html#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>)) <span class="sc">+</span> </span>
<span id="cb22-7"><a href="input-data.html#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb22-8"><a href="input-data.html#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">latitude2 =</span> <span class="fu">round</span>(latitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-9"><a href="input-data.html#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">longitude2 =</span> <span class="fu">round</span>(longitude,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-10"><a href="input-data.html#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Geographic_code,metagroup,longitude2,latitude2) <span class="sc">%&gt;%</span></span>
<span id="cb22-11"><a href="input-data.html#cb22-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">&#39;longitude2&#39;</span>, <span class="st">&#39;latitude2&#39;</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb22-12"><a href="input-data.html#cb22-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>metagroup <span class="sc">==</span> <span class="st">&quot;Selected Line - Auburn FL&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-13"><a href="input-data.html#cb22-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>metagroup <span class="sc">==</span> <span class="st">&quot;Selected Line - Auburn nGOM&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-14"><a href="input-data.html#cb22-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>metagroup <span class="sc">==</span> <span class="st">&quot;Selected Line - OBOY&quot;</span>),</span>
<span id="cb22-15"><a href="input-data.html#cb22-15" aria-hidden="true" tabindex="-1"></a>    <span class="co">#counts for proportional representation</span></span>
<span id="cb22-16"><a href="input-data.html#cb22-16" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> latitude2, <span class="at">x =</span> longitude2, <span class="at">group=</span>Geographic_code,<span class="at">colour=</span>Geographic_code,<span class="at">shape =</span> metagroup,<span class="at">size =</span> n),</span>
<span id="cb22-17"><a href="input-data.html#cb22-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">stroke=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-16-1.png" width="768" /></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="input-data.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">#produce blank map to add points to later</span></span>
<span id="cb23-2"><a href="input-data.html#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="input-data.html#cb23-3" aria-hidden="true" tabindex="-1"></a>blank_map <span class="ot">&lt;-</span> <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb23-4"><a href="input-data.html#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data=</span>MainStates,<span class="fu">aes</span>(<span class="at">x=</span>long,<span class="at">y=</span>lat,<span class="at">group=</span>group),<span class="at">fill=</span><span class="st">&quot;grey90&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb23-5"><a href="input-data.html#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb23-6"><a href="input-data.html#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb23-7"><a href="input-data.html#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb23-8"><a href="input-data.html#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">99.830000</span>,<span class="sc">-</span><span class="fl">79.783711</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">24.783218</span>,<span class="fl">30.858945</span>))</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="input-data.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">#add site codes to individual data</span></span>
<span id="cb24-2"><a href="input-data.html#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="input-data.html#cb24-3" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="ot">&lt;-</span> sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb24-4"><a href="input-data.html#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Geographic_location =</span> <span class="fu">case_when</span>(</span>
<span id="cb24-5"><a href="input-data.html#cb24-5" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;PMTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Port Mansfield&quot;</span>,</span>
<span id="cb24-6"><a href="input-data.html#cb24-6" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;ULTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Upper Laguna Madre&quot;</span>,</span>
<span id="cb24-7"><a href="input-data.html#cb24-7" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;CCTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Corpus Christi Bay&quot;</span>,</span>
<span id="cb24-8"><a href="input-data.html#cb24-8" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;SATX&quot;</span> <span class="sc">~</span> <span class="st">&quot;San Antonio Bay&quot;</span>,</span>
<span id="cb24-9"><a href="input-data.html#cb24-9" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;WMTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;West Matagorda Bay&quot;</span>,</span>
<span id="cb24-10"><a href="input-data.html#cb24-10" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;EMTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;East Matagorda Bay&quot;</span>,</span>
<span id="cb24-11"><a href="input-data.html#cb24-11" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;WGTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;West Galveston Bay&quot;</span>,</span>
<span id="cb24-12"><a href="input-data.html#cb24-12" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;EGTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;East Galveston Bay&quot;</span>,</span>
<span id="cb24-13"><a href="input-data.html#cb24-13" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;LSTX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Sabine Pass&quot;</span>,</span>
<span id="cb24-14"><a href="input-data.html#cb24-14" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;LCLA&quot;</span> <span class="sc">~</span> <span class="st">&quot;Calcasieu Lake&quot;</span>,</span>
<span id="cb24-15"><a href="input-data.html#cb24-15" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;SLLA&quot;</span> <span class="sc">~</span> <span class="st">&quot;Caillou Lake&quot;</span>,</span>
<span id="cb24-16"><a href="input-data.html#cb24-16" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;PSMS&quot;</span> <span class="sc">~</span> <span class="st">&quot;Pascagoula River&quot;</span>,</span>
<span id="cb24-17"><a href="input-data.html#cb24-17" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;CPAL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Cedar Point&quot;</span>,</span>
<span id="cb24-18"><a href="input-data.html#cb24-18" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;ALAL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Alligator Lake&quot;</span>,</span>
<span id="cb24-19"><a href="input-data.html#cb24-19" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;PBFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Pensacola Bay&quot;</span>,</span>
<span id="cb24-20"><a href="input-data.html#cb24-20" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;AHFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Alligator Harbor&quot;</span>,</span>
<span id="cb24-21"><a href="input-data.html#cb24-21" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;OBFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Oyster Bay&quot;</span>,</span>
<span id="cb24-22"><a href="input-data.html#cb24-22" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;LRFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Suwannee River&quot;</span>,</span>
<span id="cb24-23"><a href="input-data.html#cb24-23" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;SKFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Seahorse Key&quot;</span>,</span>
<span id="cb24-24"><a href="input-data.html#cb24-24" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;CRFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Corrigan Reef&quot;</span>,</span>
<span id="cb24-25"><a href="input-data.html#cb24-25" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;MKFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Madelaine Key&quot;</span>,</span>
<span id="cb24-26"><a href="input-data.html#cb24-26" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;BCFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Boca Ciega Bay Aquatic Preserve&quot;</span>,</span>
<span id="cb24-27"><a href="input-data.html#cb24-27" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;TBFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Tampa Bay&quot;</span>,</span>
<span id="cb24-28"><a href="input-data.html#cb24-28" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;CAFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Caloosahatchee River&quot;</span>,</span>
<span id="cb24-29"><a href="input-data.html#cb24-29" aria-hidden="true" tabindex="-1"></a>    Geographic_code <span class="sc">%in%</span> <span class="st">&quot;BWBFL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Backwater Bay&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-30"><a href="input-data.html#cb24-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Geographic_state =</span> <span class="fu">case_when</span>(</span>
<span id="cb24-31"><a href="input-data.html#cb24-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;TX&quot;</span> <span class="sc">~</span> <span class="st">&quot;Texas&quot;</span>,</span>
<span id="cb24-32"><a href="input-data.html#cb24-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;LA&quot;</span> <span class="sc">~</span> <span class="st">&quot;Louisiana&quot;</span>,</span>
<span id="cb24-33"><a href="input-data.html#cb24-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;MS&quot;</span> <span class="sc">~</span> <span class="st">&quot;Mississippi&quot;</span>,</span>
<span id="cb24-34"><a href="input-data.html#cb24-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;AL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Alabama&quot;</span>,</span>
<span id="cb24-35"><a href="input-data.html#cb24-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_sub</span>(Geographic_code,<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%in%</span> <span class="st">&quot;FL&quot;</span> <span class="sc">~</span> <span class="st">&quot;Florida&quot;</span>))</span></code></pre></div>
<p>Table of genotyped samples categorized by location of collection</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="input-data.html#cb25-1" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="input-data.html#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Geographic_code,Geographic_location,Geographic_state,metagroup) <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="input-data.html#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   Geographic_code Geographic_location Geographic_state metagroup  n
## 1            PMTX      Port Mansfield            Texas       STX 19
## 2            ULTX  Upper Laguna Madre            Texas       STX 42
## 3            CCTX  Corpus Christi Bay            Texas       STX 14
## 4            SATX     San Antonio Bay            Texas      nGOM 11
## 5            WMTX  West Matagorda Bay            Texas      nGOM 16
## 6            EMTX  East Matagorda Bay            Texas      nGOM 18</code></pre>
<p>Table of genotyped samples categorized by the region of collection - south Texas (STX), northern Gulf of Mexico (nGOM), Florida (FL), and individuals from selected lines (selected lines)</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="input-data.html#cb27-1" aria-hidden="true" tabindex="-1"></a>sample_info_full <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="input-data.html#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(metagroup)</span></code></pre></div>
<pre><code>##                     metagroup   n
## 1                          FL 101
## 2                         STX  75
## 3   Selected Line - Auburn FL  22
## 4 Selected Line - Auburn nGOM  23
## 5        Selected Line - OBOY  22
## 6                        nGOM 204</code></pre>
</div>
</div>
<div id="snp-information" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> SNP Information<a href="input-data.html#snp-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Total: 566,262</p>
<p>SNPs considered ‘recommended’ for downstream analysis by ThermoFisher if fit in one of the following categories:</p>
<ul>
<li><p>PolyHighResolution: SNPs with well-separated genotype clusters and two or more alleles in the genotype calls</p></li>
<li><p>NoMinorHom: SNPs with well-separated genotype clusters; one cluster is homozygous and one is heterozygous for biallelic SNPs, only one homozygous cluster and one or more heterozygous clusters appear for multiallelic SNPs</p></li>
<li><p>MonoHighResolution: SNPs with one well-formed genotype cluster; must be homozygous</p></li>
<li><p>Recommended “other” cluster (GG): 207,534</p></li>
<li><p>Recommended “south” cluster (ST): 152,260</p></li>
<li><p>Recommended in both clusters: 105,363</p></li>
</ul>
<p>Only SNPs recommended for both GG and ST analyzed here (the 105,363)</p>
<div id="general-information-on-snps" class="section level3 hasAnchor" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> General Information on SNPs<a href="input-data.html#general-information-on-snps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="input-data.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in general info</span></span>
<span id="cb29-2"><a href="input-data.html#cb29-2" aria-hidden="true" tabindex="-1"></a>snp_info <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/snp_info_cv_ECarray_header_edited.txt&quot;</span>, <span class="at">fill =</span> <span class="cn">TRUE</span> , <span class="at">header =</span> <span class="cn">TRUE</span> )</span>
<span id="cb29-3"><a href="input-data.html#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="input-data.html#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co">#rename general info column &quot;ID&quot; to &quot;probeset_id&quot; to match with genotyped files</span></span>
<span id="cb29-5"><a href="input-data.html#cb29-5" aria-hidden="true" tabindex="-1"></a>snp_info <span class="ot">&lt;-</span> snp_info <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">probeset_id =</span> ID_Name)</span>
<span id="cb29-6"><a href="input-data.html#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="input-data.html#cb29-7" aria-hidden="true" tabindex="-1"></a>snp_info <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   ReporterKey            Affx  probeset_id
## 1   531501570 Affx-1248201711 AX-578238217
## 2   531501571 Affx-1248201712 AX-578238218
## 3   531501572 Affx-1248201713 AX-578238219
## 4   531422636 Affx-1248201567 AX-578159512
## 5   531421883 Affx-1248200807 AX-578158779
## 6   531422066 Affx-1248200990 AX-578158953
##                                                                 Sequence SNPType ProbeCount Tile
## 1 CCTGCCCTTTGTACACACCGCCCGTCGCTT[C/T]TACCGATTGAATAATGAGGTGAATTAGGTGGATAA       Y          2    1
## 2                              ATGATTGCAATTGTTCATCTTGAACAAGGAATATC[G/T]A       K          2    1
## 3                        AGAGGGACTATCTGTGATGCAGCAGATGGAAGAAT[A/G]AGGCAAT       R          2    1
## 4                              CCTGATAACCAACACCCTTTTACAAAACCGTGAG[C/T]TT       Y          2    1
## 5                     GGCGGATCGA[G/T]TTTGTCTTCTCTGTCGACGTTCCAGCAGATGTGGA       K          2    1
## 6                   AGCCGCGTAATACCAGCTCCACTAGCGTATACTAA[A/C]GTTGCTGCTGTT       M          2    1
##   Replicates DesignNumber Rank Strand SetBack   taxonomy_key   organism        cust_id cust_chr
## 1          2            1  100      r     213        Bonamia    ostreae  BO_22_031_rev    BO_22
## 2          2            1  100      f     213 Haplosporidium    nelsoni MSX_17_070_fwd   MSX_17
## 3          2            1  100      f     213 Haplosporidium    nelsoni MSX_45_064_fwd   MSX_45
## 4          2            1  100      f     213         Vibrio vulnificus   VV_7_035_fwd     VV_7
## 5          2            1  100      r     213        Bonamia   exitiosa  BE_10_011_rev    BE_10
## 6          2            1  100      f     213 Haplosporidium    costale  HC_14_059_fwd    HC_14
##   cust_pos                  cust_ref cust_alt    cust_source   cust_gene_id cust_gene_name
## 1       31                AF262995.1        0        Bonamia        ostreae              1
## 2       70                    MSX_17        0 Haplosporidium        nelsoni              1
## 3       64                    MSX_45        0 Haplosporidium        nelsoni              1
## 4       35 Vibrio_vulnificus_83_4067        0         Vibrio     vulnificus              1
## 5       11                KM073107.1        0        Bonamia       exitiosa              1
## 6       59                  U20858.1      SSU           rDNA Haplosporidium        costale
##   cust_snptype cust_other_id cust_validation cust_cn_region tiling_order importance pconvert
## 1    Important          0.74     recommended                                                
## 2    Important         0.748     recommended                                                
## 3    Important         0.744     recommended                                                
## 4    Important          0.73     recommended                                                
## 5    Important         0.784     recommended                                                
## 6            1     Important           0.796    recommended                                 
##   recommendation Error
## 1                     
## 2                     
## 3                     
## 4                     
## 5                     
## 6</code></pre>
</div>
<div id="recommended-snps" class="section level3 hasAnchor" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> Recommended SNPs<a href="input-data.html#recommended-snps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>SNPs recommended by ThermoFisher for both GG and ST</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="input-data.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co">#list of recommended SNPs for each sample set</span></span>
<span id="cb31-2"><a href="input-data.html#cb31-2" aria-hidden="true" tabindex="-1"></a>other_rec <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_Recommended.ps&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-3"><a href="input-data.html#cb31-3" aria-hidden="true" tabindex="-1"></a>south_rec <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_Recommended.ps&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-4"><a href="input-data.html#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="input-data.html#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co">#merge recommended SNPs for each sample set</span></span>
<span id="cb31-6"><a href="input-data.html#cb31-6" aria-hidden="true" tabindex="-1"></a>combined_rec <span class="ot">&lt;-</span> <span class="fu">merge</span>(other_rec, south_rec)</span>
<span id="cb31-7"><a href="input-data.html#cb31-7" aria-hidden="true" tabindex="-1"></a>combined_rec <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##    probeset_id
## 1 AX-168593894
## 2 AX-168612286
## 3 AX-562727784
## 4 AX-562727806
## 5 AX-562727827
## 6 AX-562728138</code></pre>
</div>
<div id="quality-control-info" class="section level3 hasAnchor" number="1.3.3">
<h3><span class="header-section-number">1.3.3</span> Quality control info<a href="input-data.html#quality-control-info" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Quality control data on SNPs from ThermoFisher</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="input-data.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in QC data for both data sets</span></span>
<span id="cb33-2"><a href="input-data.html#cb33-2" aria-hidden="true" tabindex="-1"></a>other_qc_snp <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_Ps.performance.txt&quot;</span>, <span class="at">fill =</span> <span class="cn">TRUE</span>,<span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb33-3"><a href="input-data.html#cb33-3" aria-hidden="true" tabindex="-1"></a>south_qc_snp <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_Ps.performance.txt&quot;</span>, <span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb33-4"><a href="input-data.html#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="input-data.html#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="co">#add column of data_set - &quot;other&quot; or &quot;south_texas&quot;</span></span>
<span id="cb33-6"><a href="input-data.html#cb33-6" aria-hidden="true" tabindex="-1"></a>other_qc_snp<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="st">&quot;other&quot;</span></span>
<span id="cb33-7"><a href="input-data.html#cb33-7" aria-hidden="true" tabindex="-1"></a>south_qc_snp<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="st">&quot;south_texas&quot;</span></span>
<span id="cb33-8"><a href="input-data.html#cb33-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-9"><a href="input-data.html#cb33-9" aria-hidden="true" tabindex="-1"></a><span class="co">#rbind qc data for both data sets</span></span>
<span id="cb33-10"><a href="input-data.html#cb33-10" aria-hidden="true" tabindex="-1"></a>qc_snp <span class="ot">&lt;-</span> <span class="fu">rbind</span>(other_qc_snp,south_qc_snp)</span>
<span id="cb33-11"><a href="input-data.html#cb33-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-12"><a href="input-data.html#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="co">#select only SNPs recommended for both clusters </span></span>
<span id="cb33-13"><a href="input-data.html#cb33-13" aria-hidden="true" tabindex="-1"></a>qc_snp_rec<span class="ot">&lt;-</span><span class="fu">merge</span>(qc_snp,combined_rec)</span>
<span id="cb33-14"><a href="input-data.html#cb33-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-15"><a href="input-data.html#cb33-15" aria-hidden="true" tabindex="-1"></a>qc_snp_rec <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##    probeset_id     affy_snp_id       CR    FLD  HomFLD   HetSO  HomRO nMinorAllele Nclus n_AA n_AB
## 1 AX-168593894 Affx-1246881070  99.7312 7.4621      NA  0.9517 3.5271            2     2    0    2
## 2 AX-168593894 Affx-1246881070 100.0000 7.2167 15.3502  0.7137 2.6279           70     3   28   24
## 3 AX-168612286 Affx-1245953414  98.6559 5.9743 12.1686  0.5181 2.6369           86     3   12   62
## 4 AX-168612286 Affx-1245953414  97.3333 5.1921 11.3054  0.2048 1.7069            8     3    2    4
## 5 AX-562727784 Affx-1246856859  98.6667 4.9399      NA -0.1660 2.7572            3     2    0    3
## 6 AX-562727784 Affx-1246856859  99.4624 6.5163 15.1036  0.5076 2.8980           26     3    2   22
##   n_BB n_NC hemizygous specialSNP_chr gender_metrics     ConversionType BestProbeset
## 1  369    1          0      autosomal            all         NoMinorHom            1
## 2   23    0          0      autosomal            all PolyHighResolution            1
## 3  293    5          0      autosomal            all PolyHighResolution            1
## 4   67    2          0      autosomal            all PolyHighResolution            1
## 5   71    1          0      autosomal            all         NoMinorHom            1
## 6  346    2          0      autosomal            all PolyHighResolution            1
##   BestandRecommended HomHet     MMD MinorAlleleFrequency  H.W.p.Value H.W.chisquared.statistic
## 1                  1      1      NA               0.0027 1.0000000000                       NA
## 2                  1      0 37.1764               0.4667 0.0019817894                   9.5663
## 3                  1      0 35.6548               0.1172 0.0004426417                  12.3430
## 4                  1      0 24.7433               0.0548 0.0098412648                       NA
## 5                  1      1      NA               0.0203 1.0000000000                       NA
## 6                  1      0 41.2467               0.0351 0.0666804113                       NA
##       cluster
## 1       other
## 2 south_texas
## 3       other
## 4 south_texas
## 5 south_texas
## 6       other</code></pre>
</div>
<div id="merge-files-1" class="section level3 hasAnchor" number="1.3.4">
<h3><span class="header-section-number">1.3.4</span> Merge Files<a href="input-data.html#merge-files-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Consolidate information on SNPs into one file “snp_info_full.txt”</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="input-data.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#merge recommended and general info</span></span>
<span id="cb35-2"><a href="input-data.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co">#note the final file has two rows for each SNP (based on whether it is in the &#39;other&#39; or &#39;south_texas&#39; cluster)</span></span>
<span id="cb35-3"><a href="input-data.html#cb35-3" aria-hidden="true" tabindex="-1"></a>snp_info_full <span class="ot">&lt;-</span> <span class="fu">merge</span>(combined_rec, snp_info)</span>
<span id="cb35-4"><a href="input-data.html#cb35-4" aria-hidden="true" tabindex="-1"></a>snp_info_full <span class="ot">&lt;-</span> <span class="fu">merge</span>(snp_info_full, qc_snp_rec)</span>
<span id="cb35-5"><a href="input-data.html#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="input-data.html#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="co">#remove &quot;-&quot; in probeset id</span></span>
<span id="cb35-7"><a href="input-data.html#cb35-7" aria-hidden="true" tabindex="-1"></a>snp_info_full<span class="sc">$</span>probeset_id <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>, snp_info_full<span class="sc">$</span>probeset_id)</span>
<span id="cb35-8"><a href="input-data.html#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="input-data.html#cb35-9" aria-hidden="true" tabindex="-1"></a>snp_info_full <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   probeset_id ReporterKey            Affx
## 1 AX168593894   530224709 Affx-1246881070
## 2 AX168593894   530224709 Affx-1246881070
## 3 AX168612286   526588721 Affx-1245953414
## 4 AX168612286   526588721 Affx-1245953414
## 5 AX562727784   527308034 Affx-1246856859
## 6 AX562727784   527308034 Affx-1246856859
##                                                                      Sequence SNPType ProbeCount Tile
## 1 TTGTCTTAAGGTTGAGAGTACTGGAGACAACAGCC[C/T]CGATTAAAATCTCCCGAGCACACTTCAGTTTTCAA       Y          2    1
## 2 TTGTCTTAAGGTTGAGAGTACTGGAGACAACAGCC[C/T]CGATTAAAATCTCCCGAGCACACTTCAGTTTTCAA       Y          2    1
## 3 GTTTAATTTCTTTGCAGCTTTTTCCAGATCGATTA[C/T]CCAAACGTGAATATATCTGCCGAGCCTGGCGGAGA       Y          2    1
## 4 GTTTAATTTCTTTGCAGCTTTTTCCAGATCGATTA[C/T]CCAAACGTGAATATATCTGCCGAGCCTGGCGGAGA       Y          2    1
## 5        GAGGAAATGTATGATATATTTGGAAAATATGGAGC[C/T]ATTCGACAAATAAGGGTGTAAGTTGAAA       Y          2    1
## 6        GAGGAAATGTATGATATATTTGGAAAATATGGAGC[C/T]ATTCGACAAATAAGGGTGTAAGTTGAAA       Y          2    1
##   Replicates DesignNumber Rank Strand SetBack taxonomy_key  organism     cust_id cust_chr cust_pos
## 1          2            1 5000      r     205  Crassostrea virginica C1_61744352        1 61744352
## 2          2            1 5000      r     205  Crassostrea virginica C1_61744352        1 61744352
## 3          2            1 5000      f     205  Crassostrea virginica C7_49307452        7 49307452
## 4          2            1 5000      f     205  Crassostrea virginica C7_49307452        7 49307452
## 5          2            1 3000      f     205  Crassostrea virginica C1_58957385        1 58957385
## 6          2            1 3000      f     205  Crassostrea virginica C1_58957385        1 58957385
##   cust_ref cust_alt cust_source cust_gene_id cust_gene_name cust_snptype cust_other_id
## 1        C        T        EOBC      genomic              0     standard         0.726
## 2        C        T        EOBC      genomic              0     standard         0.726
## 3        C        T        EOBC      genomic              0     standard         0.676
## 4        C        T        EOBC      genomic              0     standard         0.676
## 5        C        T         Guo        genic   LOC111120794     splicing        factor
## 6        C        T         Guo        genic   LOC111120794     splicing        factor
##   cust_validation cust_cn_region tiling_order importance pconvert recommendation Error
## 1     recommended                                                                     
## 2     recommended                                                                     
## 3     recommended                                                                     
## 4     recommended                                                                     
## 5              3B        subunit       6-like    protein        1       standard  0.66
## 6              3B        subunit       6-like    protein        1       standard  0.66
##       affy_snp_id       CR    FLD  HomFLD   HetSO  HomRO nMinorAllele Nclus n_AA n_AB n_BB n_NC
## 1 Affx-1246881070  99.7312 7.4621      NA  0.9517 3.5271            2     2    0    2  369    1
## 2 Affx-1246881070 100.0000 7.2167 15.3502  0.7137 2.6279           70     3   28   24   23    0
## 3 Affx-1245953414  98.6559 5.9743 12.1686  0.5181 2.6369           86     3   12   62  293    5
## 4 Affx-1245953414  97.3333 5.1921 11.3054  0.2048 1.7069            8     3    2    4   67    2
## 5 Affx-1246856859  98.6667 4.9399      NA -0.1660 2.7572            3     2    0    3   71    1
## 6 Affx-1246856859  99.4624 6.5163 15.1036  0.5076 2.8980           26     3    2   22  346    2
##   hemizygous specialSNP_chr gender_metrics     ConversionType BestProbeset BestandRecommended HomHet
## 1          0      autosomal            all         NoMinorHom            1                  1      1
## 2          0      autosomal            all PolyHighResolution            1                  1      0
## 3          0      autosomal            all PolyHighResolution            1                  1      0
## 4          0      autosomal            all PolyHighResolution            1                  1      0
## 5          0      autosomal            all         NoMinorHom            1                  1      1
## 6          0      autosomal            all PolyHighResolution            1                  1      0
##       MMD MinorAlleleFrequency  H.W.p.Value H.W.chisquared.statistic     cluster
## 1      NA               0.0027 1.0000000000                       NA       other
## 2 37.1764               0.4667 0.0019817894                   9.5663 south_texas
## 3 35.6548               0.1172 0.0004426417                  12.3430       other
## 4 24.7433               0.0548 0.0098412648                       NA south_texas
## 5      NA               0.0203 1.0000000000                       NA south_texas
## 6 41.2467               0.0351 0.0666804113                       NA       other</code></pre>
</div>
<div id="formatting-for-hierfstat-and-adegenet" class="section level3 hasAnchor" number="1.3.5">
<h3><span class="header-section-number">1.3.5</span> Formatting for hierfstat and adegenet<a href="input-data.html#formatting-for-hierfstat-and-adegenet" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>SNP data formatted so to be analyzed with the “hierfstat” and “adegenet” package</p>
<p>Formatted to loci in 1 column</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="input-data.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in .cs files (takes a minute)</span></span>
<span id="cb37-2"><a href="input-data.html#cb37-2" aria-hidden="true" tabindex="-1"></a>Ncalls_other <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/other_AxiomGT1.calls.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb37-3"><a href="input-data.html#cb37-3" aria-hidden="true" tabindex="-1"></a>Ncalls_south <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/raw/south_AxiomGT1.calls.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb37-4"><a href="input-data.html#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="input-data.html#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co">#filter out only the rec SNPs from the calls data</span></span>
<span id="cb37-6"><a href="input-data.html#cb37-6" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> <span class="fu">subset</span>(Ncalls_other, probeset_id <span class="sc">%in%</span> combined_rec<span class="sc">$</span>probeset_id)</span>
<span id="cb37-7"><a href="input-data.html#cb37-7" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> <span class="fu">subset</span>(Ncalls_south, probeset_id <span class="sc">%in%</span> combined_rec<span class="sc">$</span>probeset_id)</span>
<span id="cb37-8"><a href="input-data.html#cb37-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-9"><a href="input-data.html#cb37-9" aria-hidden="true" tabindex="-1"></a><span class="co">#transpose so individuals are rows and SNPs are columns</span></span>
<span id="cb37-10"><a href="input-data.html#cb37-10" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(other_filtered))</span>
<span id="cb37-11"><a href="input-data.html#cb37-11" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(south_filtered))</span>
<span id="cb37-12"><a href="input-data.html#cb37-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-13"><a href="input-data.html#cb37-13" aria-hidden="true" tabindex="-1"></a><span class="co">#make the first row of the dataframe the column headers</span></span>
<span id="cb37-14"><a href="input-data.html#cb37-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(other_filtered) <span class="ot">&lt;-</span> other_filtered[<span class="dv">1</span>,]</span>
<span id="cb37-15"><a href="input-data.html#cb37-15" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(south_filtered) <span class="ot">&lt;-</span> south_filtered[<span class="dv">1</span>,]</span>
<span id="cb37-16"><a href="input-data.html#cb37-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-17"><a href="input-data.html#cb37-17" aria-hidden="true" tabindex="-1"></a><span class="co">#delete the first row of the data set (only used as column headers)</span></span>
<span id="cb37-18"><a href="input-data.html#cb37-18" aria-hidden="true" tabindex="-1"></a>other_filtered<span class="ot">&lt;-</span> other_filtered[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb37-19"><a href="input-data.html#cb37-19" aria-hidden="true" tabindex="-1"></a>south_filtered<span class="ot">&lt;-</span> south_filtered[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb37-20"><a href="input-data.html#cb37-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-21"><a href="input-data.html#cb37-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Make vector of other/south individuals for later referencing</span></span>
<span id="cb37-22"><a href="input-data.html#cb37-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-23"><a href="input-data.html#cb37-23" aria-hidden="true" tabindex="-1"></a>other_inds <span class="ot">&lt;-</span> <span class="fu">rownames</span>(other_filtered)</span>
<span id="cb37-24"><a href="input-data.html#cb37-24" aria-hidden="true" tabindex="-1"></a>  other_inds <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, other_inds)</span>
<span id="cb37-25"><a href="input-data.html#cb37-25" aria-hidden="true" tabindex="-1"></a>  other_inds <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, other_inds)</span>
<span id="cb37-26"><a href="input-data.html#cb37-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-27"><a href="input-data.html#cb37-27" aria-hidden="true" tabindex="-1"></a>south_inds <span class="ot">&lt;-</span> <span class="fu">rownames</span>(south_filtered)</span>
<span id="cb37-28"><a href="input-data.html#cb37-28" aria-hidden="true" tabindex="-1"></a>  south_inds <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, south_inds)</span>
<span id="cb37-29"><a href="input-data.html#cb37-29" aria-hidden="true" tabindex="-1"></a>  south_inds <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, south_inds)</span></code></pre></div>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="input-data.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="do">##set to eval=FALSE because takes so long</span></span>
<span id="cb38-2"><a href="input-data.html#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="do">##Change allele codes. </span></span>
<span id="cb38-3"><a href="input-data.html#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="do">###Warning: this takes a very long time (almost 1 hour). </span></span>
<span id="cb38-4"><a href="input-data.html#cb38-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-5"><a href="input-data.html#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co">#For some reason there is whitespace around some of the values which causes problems when converting the values. Strip all leading/trailing white space </span></span>
<span id="cb38-6"><a href="input-data.html#cb38-6" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> other_filtered <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.character), str_trim))</span>
<span id="cb38-7"><a href="input-data.html#cb38-7" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> south_filtered <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.character), str_trim))</span>
<span id="cb38-8"><a href="input-data.html#cb38-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-9"><a href="input-data.html#cb38-9" aria-hidden="true" tabindex="-1"></a><span class="co">#if value in dataframe is -1, change to NA</span></span>
<span id="cb38-10"><a href="input-data.html#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="do">## -1 specifies a &#39;No Call&#39;</span></span>
<span id="cb38-11"><a href="input-data.html#cb38-11" aria-hidden="true" tabindex="-1"></a>other_filtered[other_filtered <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span>   </span>
<span id="cb38-12"><a href="input-data.html#cb38-12" aria-hidden="true" tabindex="-1"></a>south_filtered[south_filtered <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span>   </span>
<span id="cb38-13"><a href="input-data.html#cb38-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-14"><a href="input-data.html#cb38-14" aria-hidden="true" tabindex="-1"></a><span class="co"># If code is 0 (AA) change to 11</span></span>
<span id="cb38-15"><a href="input-data.html#cb38-15" aria-hidden="true" tabindex="-1"></a>other_filtered[other_filtered <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="st">&quot;11&quot;</span></span>
<span id="cb38-16"><a href="input-data.html#cb38-16" aria-hidden="true" tabindex="-1"></a>south_filtered[south_filtered <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="st">&quot;11&quot;</span></span>
<span id="cb38-17"><a href="input-data.html#cb38-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-18"><a href="input-data.html#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="co">#If code is 1 (AB) change to 12</span></span>
<span id="cb38-19"><a href="input-data.html#cb38-19" aria-hidden="true" tabindex="-1"></a>other_filtered[other_filtered <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;12&quot;</span></span>
<span id="cb38-20"><a href="input-data.html#cb38-20" aria-hidden="true" tabindex="-1"></a>south_filtered[south_filtered <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;12&quot;</span></span>
<span id="cb38-21"><a href="input-data.html#cb38-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-22"><a href="input-data.html#cb38-22" aria-hidden="true" tabindex="-1"></a><span class="co"># If code is 2 (BB) change to 22</span></span>
<span id="cb38-23"><a href="input-data.html#cb38-23" aria-hidden="true" tabindex="-1"></a>other_filtered[other_filtered <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;22&quot;</span></span>
<span id="cb38-24"><a href="input-data.html#cb38-24" aria-hidden="true" tabindex="-1"></a>south_filtered[south_filtered <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;22&quot;</span></span>
<span id="cb38-25"><a href="input-data.html#cb38-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-26"><a href="input-data.html#cb38-26" aria-hidden="true" tabindex="-1"></a><span class="co">#save formatted data files for other and south</span></span>
<span id="cb38-27"><a href="input-data.html#cb38-27" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(other_filtered, <span class="st">&quot;data/derived/cluster_other_formatted.txt&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb38-28"><a href="input-data.html#cb38-28" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(south_filtered, <span class="st">&quot;data/derived/cluster_south_formatted.txt&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="input-data.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in formatted data files for other and south</span></span>
<span id="cb39-2"><a href="input-data.html#cb39-2" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> <span class="fu">read.table</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;derived&quot;</span>,<span class="st">&quot;cluster_other_formatted.txt&quot;</span>))</span>
<span id="cb39-3"><a href="input-data.html#cb39-3" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> <span class="fu">read.table</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;derived&quot;</span>,<span class="st">&quot;cluster_south_formatted.txt&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="input-data.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Consolidate to one file - &quot;combined_filtered.txt&quot;</span></span>
<span id="cb40-2"><a href="input-data.html#cb40-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-3"><a href="input-data.html#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="co"># sort data frames to make sure columns are in the same order before combining</span></span>
<span id="cb40-4"><a href="input-data.html#cb40-4" aria-hidden="true" tabindex="-1"></a>other_filtered <span class="ot">&lt;-</span> other_filtered[,<span class="fu">order</span>(<span class="fu">names</span>(other_filtered))]</span>
<span id="cb40-5"><a href="input-data.html#cb40-5" aria-hidden="true" tabindex="-1"></a>south_filtered <span class="ot">&lt;-</span> south_filtered[,<span class="fu">order</span>(<span class="fu">names</span>(south_filtered))]</span>
<span id="cb40-6"><a href="input-data.html#cb40-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-7"><a href="input-data.html#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co"># double check because I&#39;m paranoid </span></span>
<span id="cb40-8"><a href="input-data.html#cb40-8" aria-hidden="true" tabindex="-1"></a>cols1 <span class="ot">&lt;-</span> <span class="fu">names</span>(other_filtered)</span>
<span id="cb40-9"><a href="input-data.html#cb40-9" aria-hidden="true" tabindex="-1"></a>cols2 <span class="ot">&lt;-</span> <span class="fu">names</span>(south_filtered)</span>
<span id="cb40-10"><a href="input-data.html#cb40-10" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(cols1, cols2)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="input-data.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co">#combine datasets</span></span>
<span id="cb42-2"><a href="input-data.html#cb42-2" aria-hidden="true" tabindex="-1"></a>combined_filtered <span class="ot">&lt;-</span> <span class="fu">rbind</span>(other_filtered, south_filtered)</span>
<span id="cb42-3"><a href="input-data.html#cb42-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-4"><a href="input-data.html#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(combined_filtered))</span>
<span id="cb42-5"><a href="input-data.html#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(combined_filtered))</span>
<span id="cb42-6"><a href="input-data.html#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(combined_filtered))</span>
<span id="cb42-7"><a href="input-data.html#cb42-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(combined_filtered))</span>
<span id="cb42-8"><a href="input-data.html#cb42-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(combined_filtered) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(combined_filtered))</span></code></pre></div>
</div>
</div>
<div id="save-files" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Save files<a href="input-data.html#save-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="input-data.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co">#save sample_info_Full</span></span>
<span id="cb43-2"><a href="input-data.html#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(sample_info_full, <span class="at">file =</span> here <span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;derived&quot;</span>,<span class="st">&quot;sample_info_full&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="input-data.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co">#save_snp_info_full</span></span>
<span id="cb44-2"><a href="input-data.html#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_info_full, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;derived&quot;</span>,<span class="st">&quot;snp_info_full&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="input-data.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co">#save combined_filtered data file</span></span>
<span id="cb45-2"><a href="input-data.html#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(combined_filtered, <span class="st">&quot;data/derived/combined_filtered.txt&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="genetic-groups.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="book_files/gitbook-2.6.7/js/app.min.js"></script>
<script src="book_files/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="book_files/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="book_files/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"toolbar": {
"position": "static"
}
});
});
</script>

</body>

</html>
